// intl-tel-input/vue/build/intl-tel-input-BMFKRhx-.mjs
var ue = (a) => {
  throw TypeError(a);
};
var $t = (a, t, s) => t.has(a) || ue("Cannot " + s);
var e = (a, t, s) => ($t(a, t, "read from private field"), s ? s.call(a) : t.get(a)), I = (a, t, s) => t.has(a) ? ue("Cannot add the same private member more than once") : t instanceof WeakSet ? t.add(a) : t.set(a, s), p = (a, t, s, i) => ($t(a, t, "write to private field"), i ? i.call(a, s) : t.set(a, s), s), r = (a, t, s) => ($t(a, t, "access private method"), s);
const ds = [
  [
    "af",
    // Afghanistan
    "93",
    0,
    null,
    "0"
  ],
  [
    "ax",
    // Åland Islands
    "358",
    1,
    ["18", "4"],
    // (4 is a mobile range shared with FI)
    "0"
  ],
  [
    "al",
    // Albania
    "355",
    0,
    null,
    "0"
  ],
  [
    "dz",
    // Algeria
    "213",
    0,
    null,
    "0"
  ],
  [
    "as",
    // American Samoa
    "1",
    5,
    ["684"],
    "1"
  ],
  [
    "ad",
    // Andorra
    "376"
  ],
  [
    "ao",
    // Angola
    "244"
  ],
  [
    "ai",
    // Anguilla
    "1",
    6,
    ["264"],
    "1"
  ],
  [
    "ag",
    // Antigua and Barbuda
    "1",
    7,
    ["268"],
    "1"
  ],
  [
    "ar",
    // Argentina
    "54",
    0,
    null,
    "0"
  ],
  [
    "am",
    // Armenia
    "374",
    0,
    null,
    "0"
  ],
  [
    "aw",
    // Aruba
    "297"
  ],
  [
    "ac",
    // Ascension Island
    "247"
  ],
  [
    "au",
    // Australia
    "61",
    0,
    ["4"],
    // (mobile range shared with CX and CC)
    "0"
  ],
  [
    "at",
    // Austria
    "43",
    0,
    null,
    "0"
  ],
  [
    "az",
    // Azerbaijan
    "994",
    0,
    null,
    "0"
  ],
  [
    "bs",
    // Bahamas
    "1",
    8,
    ["242"],
    "1"
  ],
  [
    "bh",
    // Bahrain
    "973"
  ],
  [
    "bd",
    // Bangladesh
    "880",
    0,
    null,
    "0"
  ],
  [
    "bb",
    // Barbados
    "1",
    9,
    ["246"],
    "1"
  ],
  [
    "by",
    // Belarus
    "375",
    0,
    null,
    "8"
  ],
  [
    "be",
    // Belgium
    "32",
    0,
    null,
    "0"
  ],
  [
    "bz",
    // Belize
    "501"
  ],
  [
    "bj",
    // Benin
    "229"
  ],
  [
    "bm",
    // Bermuda
    "1",
    10,
    ["441"],
    "1"
  ],
  [
    "bt",
    // Bhutan
    "975"
  ],
  [
    "bo",
    // Bolivia
    "591",
    0,
    null,
    "0"
  ],
  [
    "ba",
    // Bosnia and Herzegovina
    "387",
    0,
    null,
    "0"
  ],
  [
    "bw",
    // Botswana
    "267"
  ],
  [
    "br",
    // Brazil
    "55",
    0,
    null,
    "0"
  ],
  [
    "io",
    // British Indian Ocean Territory
    "246"
  ],
  [
    "vg",
    // British Virgin Islands
    "1",
    11,
    ["284"],
    "1"
  ],
  [
    "bn",
    // Brunei
    "673"
  ],
  [
    "bg",
    // Bulgaria
    "359",
    0,
    null,
    "0"
  ],
  [
    "bf",
    // Burkina Faso
    "226"
  ],
  [
    "bi",
    // Burundi
    "257"
  ],
  [
    "kh",
    // Cambodia
    "855",
    0,
    null,
    "0"
  ],
  [
    "cm",
    // Cameroon
    "237"
  ],
  [
    "ca",
    // Canada
    "1",
    1,
    ["204", "226", "236", "249", "250", "257", "263", "289", "306", "343", "354", "365", "367", "368", "382", "403", "416", "418", "428", "431", "437", "438", "450", "468", "474", "506", "514", "519", "548", "579", "581", "584", "587", "604", "613", "639", "647", "672", "683", "705", "709", "742", "753", "778", "780", "782", "807", "819", "825", "867", "873", "879", "902", "905", "942"],
    "1"
  ],
  [
    "cv",
    // Cape Verde
    "238"
  ],
  [
    "bq",
    // Caribbean Netherlands
    "599",
    1,
    ["3", "4", "7"]
  ],
  [
    "ky",
    // Cayman Islands
    "1",
    12,
    ["345"],
    "1"
  ],
  [
    "cf",
    // Central African Republic
    "236"
  ],
  [
    "td",
    // Chad
    "235"
  ],
  [
    "cl",
    // Chile
    "56"
  ],
  [
    "cn",
    // China
    "86",
    0,
    null,
    "0"
  ],
  [
    "cx",
    // Christmas Island
    "61",
    2,
    ["4", "89164"],
    // (4 is a mobile range shared with AU and CC)
    "0"
  ],
  [
    "cc",
    // Cocos (Keeling) Islands
    "61",
    1,
    ["4", "89162"],
    // (4 is a mobile range shared with AU and CX)
    "0"
  ],
  [
    "co",
    // Colombia
    "57",
    0,
    null,
    "0"
  ],
  [
    "km",
    // Comoros
    "269"
  ],
  [
    "cg",
    // Congo (Brazzaville)
    "242"
  ],
  [
    "cd",
    // Congo (Kinshasa)
    "243",
    0,
    null,
    "0"
  ],
  [
    "ck",
    // Cook Islands
    "682"
  ],
  [
    "cr",
    // Costa Rica
    "506"
  ],
  [
    "ci",
    // Côte d'Ivoire
    "225"
  ],
  [
    "hr",
    // Croatia
    "385",
    0,
    null,
    "0"
  ],
  [
    "cu",
    // Cuba
    "53",
    0,
    null,
    "0"
  ],
  [
    "cw",
    // Curaçao
    "599",
    0
  ],
  [
    "cy",
    // Cyprus
    "357"
  ],
  [
    "cz",
    // Czech Republic
    "420"
  ],
  [
    "dk",
    // Denmark
    "45"
  ],
  [
    "dj",
    // Djibouti
    "253"
  ],
  [
    "dm",
    // Dominica
    "1",
    13,
    ["767"],
    "1"
  ],
  [
    "do",
    // Dominican Republic
    "1",
    2,
    ["809", "829", "849"],
    "1"
  ],
  [
    "ec",
    // Ecuador
    "593",
    0,
    null,
    "0"
  ],
  [
    "eg",
    // Egypt
    "20",
    0,
    null,
    "0"
  ],
  [
    "sv",
    // El Salvador
    "503"
  ],
  [
    "gq",
    // Equatorial Guinea
    "240"
  ],
  [
    "er",
    // Eritrea
    "291",
    0,
    null,
    "0"
  ],
  [
    "ee",
    // Estonia
    "372"
  ],
  [
    "sz",
    // Eswatini
    "268"
  ],
  [
    "et",
    // Ethiopia
    "251",
    0,
    null,
    "0"
  ],
  [
    "fk",
    // Falkland Islands (Malvinas)
    "500"
  ],
  [
    "fo",
    // Faroe Islands
    "298"
  ],
  [
    "fj",
    // Fiji
    "679"
  ],
  [
    "fi",
    // Finland
    "358",
    0,
    ["4"],
    // (mobile range shared with AX)
    "0"
  ],
  [
    "fr",
    // France
    "33",
    0,
    null,
    "0"
  ],
  [
    "gf",
    // French Guiana
    "594",
    0,
    null,
    "0"
  ],
  [
    "pf",
    // French Polynesia
    "689"
  ],
  [
    "ga",
    // Gabon
    "241"
  ],
  [
    "gm",
    // Gambia
    "220"
  ],
  [
    "ge",
    // Georgia
    "995",
    0,
    null,
    "0"
  ],
  [
    "de",
    // Germany
    "49",
    0,
    null,
    "0"
  ],
  [
    "gh",
    // Ghana
    "233",
    0,
    null,
    "0"
  ],
  [
    "gi",
    // Gibraltar
    "350"
  ],
  [
    "gr",
    // Greece
    "30"
  ],
  [
    "gl",
    // Greenland
    "299"
  ],
  [
    "gd",
    // Grenada
    "1",
    14,
    ["473"],
    "1"
  ],
  [
    "gp",
    // Guadeloupe
    "590",
    0,
    null,
    "0"
  ],
  [
    "gu",
    // Guam
    "1",
    15,
    ["671"],
    "1"
  ],
  [
    "gt",
    // Guatemala
    "502"
  ],
  [
    "gg",
    // Guernsey
    "44",
    1,
    ["1481", "7781", "7839", "7911"],
    "0"
  ],
  [
    "gn",
    // Guinea
    "224"
  ],
  [
    "gw",
    // Guinea-Bissau
    "245"
  ],
  [
    "gy",
    // Guyana
    "592"
  ],
  [
    "ht",
    // Haiti
    "509"
  ],
  [
    "hn",
    // Honduras
    "504"
  ],
  [
    "hk",
    // Hong Kong SAR China
    "852"
  ],
  [
    "hu",
    // Hungary
    "36",
    0,
    null,
    "06"
  ],
  [
    "is",
    // Iceland
    "354"
  ],
  [
    "in",
    // India
    "91",
    0,
    null,
    "0"
  ],
  [
    "id",
    // Indonesia
    "62",
    0,
    null,
    "0"
  ],
  [
    "ir",
    // Iran
    "98",
    0,
    null,
    "0"
  ],
  [
    "iq",
    // Iraq
    "964",
    0,
    null,
    "0"
  ],
  [
    "ie",
    // Ireland
    "353",
    0,
    null,
    "0"
  ],
  [
    "im",
    // Isle of Man
    "44",
    2,
    ["1624", "74576", "7524", "7624", "7924"],
    "0"
  ],
  [
    "il",
    // Israel
    "972",
    0,
    null,
    "0"
  ],
  [
    "it",
    // Italy
    "39",
    0,
    ["3"]
    // (mobile range shared with VA)
  ],
  [
    "jm",
    // Jamaica
    "1",
    4,
    ["658", "876"],
    "1"
  ],
  [
    "jp",
    // Japan
    "81",
    0,
    null,
    "0"
  ],
  [
    "je",
    // Jersey
    "44",
    3,
    ["1534", "7509", "7700", "7797", "7829", "7937"],
    "0"
  ],
  [
    "jo",
    // Jordan
    "962",
    0,
    null,
    "0"
  ],
  [
    "kz",
    // Kazakhstan
    "7",
    1,
    ["33", "7"],
    // (33 is shared with RU)
    "8"
  ],
  [
    "ke",
    // Kenya
    "254",
    0,
    null,
    "0"
  ],
  [
    "ki",
    // Kiribati
    "686",
    0,
    null,
    "0"
  ],
  [
    "xk",
    // Kosovo
    "383",
    0,
    null,
    "0"
  ],
  [
    "kw",
    // Kuwait
    "965"
  ],
  [
    "kg",
    // Kyrgyzstan
    "996",
    0,
    null,
    "0"
  ],
  [
    "la",
    // Laos
    "856",
    0,
    null,
    "0"
  ],
  [
    "lv",
    // Latvia
    "371"
  ],
  [
    "lb",
    // Lebanon
    "961",
    0,
    null,
    "0"
  ],
  [
    "ls",
    // Lesotho
    "266"
  ],
  [
    "lr",
    // Liberia
    "231",
    0,
    null,
    "0"
  ],
  [
    "ly",
    // Libya
    "218",
    0,
    null,
    "0"
  ],
  [
    "li",
    // Liechtenstein
    "423",
    0,
    null,
    "0"
  ],
  [
    "lt",
    // Lithuania
    "370",
    0,
    null,
    "0"
  ],
  [
    "lu",
    // Luxembourg
    "352"
  ],
  [
    "mo",
    // Macao SAR China
    "853"
  ],
  [
    "mg",
    // Madagascar
    "261",
    0,
    null,
    "0"
  ],
  [
    "mw",
    // Malawi
    "265",
    0,
    null,
    "0"
  ],
  [
    "my",
    // Malaysia
    "60",
    0,
    null,
    "0"
  ],
  [
    "mv",
    // Maldives
    "960"
  ],
  [
    "ml",
    // Mali
    "223"
  ],
  [
    "mt",
    // Malta
    "356"
  ],
  [
    "mh",
    // Marshall Islands
    "692",
    0,
    null,
    "1"
  ],
  [
    "mq",
    // Martinique
    "596",
    0,
    null,
    "0"
  ],
  [
    "mr",
    // Mauritania
    "222"
  ],
  [
    "mu",
    // Mauritius
    "230"
  ],
  [
    "yt",
    // Mayotte
    "262",
    1,
    ["269", "639"],
    "0"
  ],
  [
    "mx",
    // Mexico
    "52"
  ],
  [
    "fm",
    // Micronesia
    "691"
  ],
  [
    "md",
    // Moldova
    "373",
    0,
    null,
    "0"
  ],
  [
    "mc",
    // Monaco
    "377",
    0,
    null,
    "0"
  ],
  [
    "mn",
    // Mongolia
    "976",
    0,
    null,
    "0"
  ],
  [
    "me",
    // Montenegro
    "382",
    0,
    null,
    "0"
  ],
  [
    "ms",
    // Montserrat
    "1",
    16,
    ["664"],
    "1"
  ],
  [
    "ma",
    // Morocco
    "212",
    0,
    ["6", "7"],
    // (mobile ranges shared with EH)
    "0"
  ],
  [
    "mz",
    // Mozambique
    "258"
  ],
  [
    "mm",
    // Myanmar (Burma)
    "95",
    0,
    null,
    "0"
  ],
  [
    "na",
    // Namibia
    "264",
    0,
    null,
    "0"
  ],
  [
    "nr",
    // Nauru
    "674"
  ],
  [
    "np",
    // Nepal
    "977",
    0,
    null,
    "0"
  ],
  [
    "nl",
    // Netherlands
    "31",
    0,
    null,
    "0"
  ],
  [
    "nc",
    // New Caledonia
    "687"
  ],
  [
    "nz",
    // New Zealand
    "64",
    0,
    null,
    "0"
  ],
  [
    "ni",
    // Nicaragua
    "505"
  ],
  [
    "ne",
    // Niger
    "227"
  ],
  [
    "ng",
    // Nigeria
    "234",
    0,
    null,
    "0"
  ],
  [
    "nu",
    // Niue
    "683"
  ],
  [
    "nf",
    // Norfolk Island
    "672"
  ],
  [
    "kp",
    // North Korea
    "850",
    0,
    null,
    "0"
  ],
  [
    "mk",
    // North Macedonia
    "389",
    0,
    null,
    "0"
  ],
  [
    "mp",
    // Northern Mariana Islands
    "1",
    17,
    ["670"],
    "1"
  ],
  [
    "no",
    // Norway
    "47",
    0,
    ["4", "9"]
    // (mobile ranges shared with SJ)
  ],
  [
    "om",
    // Oman
    "968"
  ],
  [
    "pk",
    // Pakistan
    "92",
    0,
    null,
    "0"
  ],
  [
    "pw",
    // Palau
    "680"
  ],
  [
    "ps",
    // Palestinian Territories
    "970",
    0,
    null,
    "0"
  ],
  [
    "pa",
    // Panama
    "507"
  ],
  [
    "pg",
    // Papua New Guinea
    "675"
  ],
  [
    "py",
    // Paraguay
    "595",
    0,
    null,
    "0"
  ],
  [
    "pe",
    // Peru
    "51",
    0,
    null,
    "0"
  ],
  [
    "ph",
    // Philippines
    "63",
    0,
    null,
    "0"
  ],
  [
    "pl",
    // Poland
    "48"
  ],
  [
    "pt",
    // Portugal
    "351"
  ],
  [
    "pr",
    // Puerto Rico
    "1",
    3,
    ["787", "939"],
    "1"
  ],
  [
    "qa",
    // Qatar
    "974"
  ],
  [
    "re",
    // Réunion
    "262",
    0,
    null,
    "0"
  ],
  [
    "ro",
    // Romania
    "40",
    0,
    null,
    "0"
  ],
  [
    "ru",
    // Russia
    "7",
    0,
    ["33"],
    // (shared with KZ)
    "8"
  ],
  [
    "rw",
    // Rwanda
    "250",
    0,
    null,
    "0"
  ],
  [
    "ws",
    // Samoa
    "685"
  ],
  [
    "sm",
    // San Marino
    "378"
  ],
  [
    "st",
    // São Tomé & Príncipe
    "239"
  ],
  [
    "sa",
    // Saudi Arabia
    "966",
    0,
    null,
    "0"
  ],
  [
    "sn",
    // Senegal
    "221"
  ],
  [
    "rs",
    // Serbia
    "381",
    0,
    null,
    "0"
  ],
  [
    "sc",
    // Seychelles
    "248"
  ],
  [
    "sl",
    // Sierra Leone
    "232",
    0,
    null,
    "0"
  ],
  [
    "sg",
    // Singapore
    "65"
  ],
  [
    "sx",
    // Sint Maarten
    "1",
    21,
    ["721"],
    "1"
  ],
  [
    "sk",
    // Slovakia
    "421",
    0,
    null,
    "0"
  ],
  [
    "si",
    // Slovenia
    "386",
    0,
    null,
    "0"
  ],
  [
    "sb",
    // Solomon Islands
    "677"
  ],
  [
    "so",
    // Somalia
    "252",
    0,
    null,
    "0"
  ],
  [
    "za",
    // South Africa
    "27",
    0,
    null,
    "0"
  ],
  [
    "kr",
    // South Korea
    "82",
    0,
    null,
    "0"
  ],
  [
    "ss",
    // South Sudan
    "211",
    0,
    null,
    "0"
  ],
  [
    "es",
    // Spain
    "34"
  ],
  [
    "lk",
    // Sri Lanka
    "94",
    0,
    null,
    "0"
  ],
  [
    "bl",
    // St. Barthélemy
    "590",
    1,
    null,
    "0"
  ],
  [
    "sh",
    // St. Helena
    "290"
  ],
  [
    "kn",
    // St. Kitts & Nevis
    "1",
    18,
    ["869"],
    "1"
  ],
  [
    "lc",
    // St. Lucia
    "1",
    19,
    ["758"],
    "1"
  ],
  [
    "mf",
    // St. Martin
    "590",
    2,
    null,
    "0"
  ],
  [
    "pm",
    // St. Pierre & Miquelon
    "508",
    0,
    null,
    "0"
  ],
  [
    "vc",
    // St. Vincent & Grenadines
    "1",
    20,
    ["784"],
    "1"
  ],
  [
    "sd",
    // Sudan
    "249",
    0,
    null,
    "0"
  ],
  [
    "sr",
    // Suriname
    "597"
  ],
  [
    "sj",
    // Svalbard & Jan Mayen
    "47",
    1,
    ["4", "79", "9"]
    // (4 and 9 are mobile ranges shared with NO)
  ],
  [
    "se",
    // Sweden
    "46",
    0,
    null,
    "0"
  ],
  [
    "ch",
    // Switzerland
    "41",
    0,
    null,
    "0"
  ],
  [
    "sy",
    // Syria
    "963",
    0,
    null,
    "0"
  ],
  [
    "tw",
    // Taiwan
    "886",
    0,
    null,
    "0"
  ],
  [
    "tj",
    // Tajikistan
    "992"
  ],
  [
    "tz",
    // Tanzania
    "255",
    0,
    null,
    "0"
  ],
  [
    "th",
    // Thailand
    "66",
    0,
    null,
    "0"
  ],
  [
    "tl",
    // Timor-Leste
    "670"
  ],
  [
    "tg",
    // Togo
    "228"
  ],
  [
    "tk",
    // Tokelau
    "690"
  ],
  [
    "to",
    // Tonga
    "676"
  ],
  [
    "tt",
    // Trinidad & Tobago
    "1",
    22,
    ["868"],
    "1"
  ],
  [
    "tn",
    // Tunisia
    "216"
  ],
  [
    "tr",
    // Turkey
    "90",
    0,
    null,
    "0"
  ],
  [
    "tm",
    // Turkmenistan
    "993",
    0,
    null,
    "8"
  ],
  [
    "tc",
    // Turks & Caicos Islands
    "1",
    23,
    ["649"],
    "1"
  ],
  [
    "tv",
    // Tuvalu
    "688"
  ],
  [
    "vi",
    // U.S. Virgin Islands
    "1",
    24,
    ["340"],
    "1"
  ],
  [
    "ug",
    // Uganda
    "256",
    0,
    null,
    "0"
  ],
  [
    "ua",
    // Ukraine
    "380",
    0,
    null,
    "0"
  ],
  [
    "ae",
    // United Arab Emirates
    "971",
    0,
    null,
    "0"
  ],
  [
    "gb",
    // United Kingdom
    "44",
    0,
    null,
    "0"
  ],
  [
    "us",
    // United States
    "1",
    0,
    null,
    "1"
  ],
  [
    "uy",
    // Uruguay
    "598",
    0,
    null,
    "0"
  ],
  [
    "uz",
    // Uzbekistan
    "998"
  ],
  [
    "vu",
    // Vanuatu
    "678"
  ],
  [
    "va",
    // Vatican City
    "39",
    1,
    ["06698", "3"]
    // (3 is a mobile range shared with IT)
  ],
  [
    "ve",
    // Venezuela
    "58",
    0,
    null,
    "0"
  ],
  [
    "vn",
    // Vietnam
    "84",
    0,
    null,
    "0"
  ],
  [
    "wf",
    // Wallis & Futuna
    "681"
  ],
  [
    "eh",
    // Western Sahara
    "212",
    1,
    ["5288", "5289", "6", "7"],
    // (6 and 7 are mobile ranges shared with MA)
    "0"
  ],
  [
    "ye",
    // Yemen
    "967",
    0,
    null,
    "0"
  ],
  [
    "zm",
    // Zambia
    "260",
    0,
    null,
    "0"
  ],
  [
    "zw",
    // Zimbabwe
    "263",
    0,
    null,
    "0"
  ]
], ht = [];
for (const a of ds)
  ht.push({
    name: "",
    // populated in the plugin
    iso2: a[0],
    dialCode: a[1],
    priority: a[2] || 0,
    areaCodes: a[3] || null,
    nodeById: {},
    // populated by the plugin
    nationalPrefix: a[4] || null,
    normalisedName: "",
    // populated in the plugin
    initials: "",
    // populated in the plugin
    dialCodePlus: ""
    // populated in the plugin
  });
const Rt = {
  OPEN_COUNTRY_DROPDOWN: "open:countrydropdown",
  CLOSE_COUNTRY_DROPDOWN: "close:countrydropdown",
  COUNTRY_CHANGE: "countrychange",
  INPUT: "input"
  // used for synthetic input trigger
}, b = {
  HIDE: "iti__hide",
  V_HIDE: "iti__v-hide",
  ARROW_UP: "iti__arrow--up",
  GLOBE: "iti__globe",
  FLAG: "iti__flag",
  LOADING: "iti__loading",
  COUNTRY_ITEM: "iti__country",
  HIGHLIGHT: "iti__highlight"
}, S = {
  ARROW_UP: "ArrowUp",
  ARROW_DOWN: "ArrowDown",
  SPACE: " ",
  ENTER: "Enter",
  ESC: "Escape",
  TAB: "Tab"
}, ce = {
  PASTE: "insertFromPaste",
  DELETE_FWD: "deleteContentForward"
}, Q = {
  ALPHA_UNICODE: new RegExp("\\p{L}", "u"),
  // any kind of letter from any language
  NON_PLUS_NUMERIC: /[^+0-9]/,
  // chars that are NOT + or digit
  NON_PLUS_NUMERIC_GLOBAL: /[^+0-9]/g,
  // chars that are NOT + or digit (global)
  HIDDEN_SEARCH_CHAR: /^[a-zA-ZÀ-ÿа-яА-Я ]$/
  // single acceptable hidden-search char
}, Ee = {
  SEARCH_DEBOUNCE_MS: 100,
  HIDDEN_SEARCH_RESET_MS: 1e3
}, he = {
  UNKNOWN_NUMBER_TYPE: -99,
  UNKNOWN_VALIDATION_ERROR: -99
}, J = {
  NARROW_VIEWPORT_WIDTH: 500,
  // keep in sync with .iti__country-list CSS media query
  SANE_SELECTED_WITH_DIAL_WIDTH: 78,
  // px width fallback when separateDialCode enabled
  SANE_SELECTED_NO_DIAL_WIDTH: 42,
  // px width fallback when no separate dial code
  INPUT_PADDING_EXTRA_LEFT: 6,
  // px gap between selected country container and input text
  DROPDOWN_MARGIN: 3,
  // px margin between dropdown and tel input
  SANE_DROPDOWN_HEIGHT: 200
  // px height fallback for dropdown
}, be = {
  NANP: "1"
  // North American Numbering Plan
}, Ft = {
  DIAL_CODE: "44",
  // +44 United Kingdom
  MOBILE_PREFIX: "7",
  // UK mobile numbers start with 7 after national trunk (0) or core section
  MOBILE_CORE_LENGTH: 10
  // core number length (excluding dial code / national prefix) for mobiles
}, ps = {
  ISO2: "us"
}, xt = {
  AGGRESSIVE: "aggressive",
  POLITE: "polite",
  OFF: "off"
}, It = {
  AUTO: "auto"
}, fs = [
  "FIXED_LINE",
  "MOBILE",
  "FIXED_LINE_OR_MOBILE",
  "TOLL_FREE",
  "PREMIUM_RATE",
  "SHARED_COST",
  "VOIP",
  "PERSONAL_NUMBER",
  "PAGER",
  "UAN",
  "VOICEMAIL",
  "UNKNOWN"
], kt = new Set(fs), de = {
  COUNTRY_CODE: "countryCode",
  DIAL_CODE: "dialCode"
}, w = {
  EXPANDED: "aria-expanded",
  LABEL: "aria-label",
  SELECTED: "aria-selected",
  ACTIVE_DESCENDANT: "aria-activedescendant",
  HASPOPUP: "aria-haspopup",
  CONTROLS: "aria-controls",
  HIDDEN: "aria-hidden",
  AUTOCOMPLETE: "aria-autocomplete",
  MODAL: "aria-modal"
}, Cs = {
  selectedCountryAriaLabel: "Change country, selected ${countryName} (${dialCode})",
  noCountrySelected: "Select country",
  countryListAriaLabel: "List of countries",
  searchPlaceholder: "Search",
  clearSearchAriaLabel: "Clear search",
  searchEmptyState: "No results found",
  searchSummaryAria(a) {
    return a === 0 ? "No results found" : a === 1 ? "1 result found" : `${a} results found`;
  }
}, jt = (a) => typeof window < "u" && typeof window.matchMedia == "function" && window.matchMedia(a).matches, we = () => jt(`(max-width: ${J.NARROW_VIEWPORT_WIDTH}px)`), ys = () => {
  if (typeof navigator < "u" && typeof window < "u") {
    const a = jt("(max-height: 600px)"), t = jt("(pointer: coarse)");
    return we() || t && a;
  }
  return !1;
}, ae = {
  //* Whether or not to allow the dropdown.
  allowDropdown: !0,
  //* The number type to enforce during validation.
  allowedNumberTypes: ["MOBILE", "FIXED_LINE"],
  //* Whether or not to allow extensions after the main number.
  allowNumberExtensions: !1,
  // Allow alphanumeric "phonewords" (e.g. +1 800 FLOWERS) as valid numbers
  allowPhonewords: !1,
  //* Add a placeholder in the input with an example number for the selected country.
  autoPlaceholder: xt.POLITE,
  //* Add a custom class to the (injected) container element.
  containerClass: "",
  //* Locale for localising country names via Intl.DisplayNames.
  countryNameLocale: "en",
  //* The order of the countries in the dropdown. Defaults to alphabetical.
  countryOrder: null,
  //* Add a country search input at the top of the dropdown.
  countrySearch: !0,
  //* Modify the auto placeholder.
  customPlaceholder: null,
  //* Always show the dropdown
  dropdownAlwaysOpen: !1,
  //* Append menu to specified element.
  dropdownContainer: null,
  //* Don't display these countries.
  excludeCountries: [],
  //* Fix the dropdown width to the input width (rather than being as wide as the longest country name).
  fixDropdownWidth: !0,
  //* Format the number as the user types
  formatAsYouType: !0,
  //* Format the input value during initialisation and on setNumber.
  formatOnDisplay: !0,
  //* geoIp lookup function.
  geoIpLookup: null,
  //* Inject a hidden input with the name returned from this function, and on submit, populate it with the result of getNumber.
  hiddenInput: null,
  //* Internationalise the plugin text e.g. search input placeholder, country names.
  i18n: {},
  //* Initial country.
  initialCountry: "",
  //* A function to load the utils script.
  loadUtils: null,
  //* National vs international formatting for numbers e.g. placeholders and displaying existing numbers.
  nationalMode: !0,
  //* Display only these countries.
  onlyCountries: [],
  //* Number type to use for placeholders.
  placeholderNumberType: "MOBILE",
  //* Add custom classes to the search input element.
  searchInputClass: "",
  //* Display the international dial code next to the selected flag.
  separateDialCode: !1,
  //* Show flags - for both the selected country, and in the country dropdown
  showFlags: !0,
  //* Only allow certain chars e.g. a plus followed by numeric digits, and cap at max valid length.
  strictMode: !1,
  //* Use full screen popup instead of dropdown for country list.
  useFullscreenPopup: ys()
}, Ne = (a) => JSON.stringify(a), pe = (a) => !!a && typeof a == "object" && !Array.isArray(a), ms = (a) => typeof a == "function", gs = (a) => {
  if (!a || typeof a != "object") return !1;
  const t = a;
  return t.nodeType === 1 && typeof t.tagName == "string" && typeof t.appendChild == "function";
}, Is = new Set(ht.map((a) => a.iso2)), Le = (a) => Is.has(a), fe = new Set(Object.values(xt)), Mt = (a) => {
  console.warn(`[intl-tel-input] ${a}`);
}, R = (a, t, s) => {
  Mt(`Option '${a}' must be ${t}; got ${Ne(s)}. Ignoring.`);
}, Es = (a, t) => Object.prototype.hasOwnProperty.call(a, t), Ce = (a, t) => {
  const s = "an array of ISO2 country code strings";
  if (!Array.isArray(t))
    return R(a, s, t), !1;
  for (const i of t) {
    if (typeof i != "string")
      return R(a, s, t), !1;
    const n = i.toLowerCase();
    if (!Le(n))
      return Mt(`Invalid country code in '${a}': '${i}'. Ignoring.`), !1;
  }
  return !0;
}, bs = (a) => {
  if (a === void 0)
    return {};
  if (!pe(a)) {
    const s = `The second argument must be an options object; got ${Ne(a)}. Using defaults.`;
    return Mt(s), {};
  }
  const t = {};
  for (const [s, i] of Object.entries(a)) {
    if (!Es(ae, s)) {
      Mt(`Unknown option '${s}'. Ignoring.`);
      continue;
    }
    switch (s) {
      case "allowDropdown":
      case "allowNumberExtensions":
      case "allowPhonewords":
      case "countrySearch":
      case "dropdownAlwaysOpen":
      case "fixDropdownWidth":
      case "formatAsYouType":
      case "formatOnDisplay":
      case "nationalMode":
      case "showFlags":
      case "separateDialCode":
      case "strictMode":
      case "useFullscreenPopup":
        if (typeof i != "boolean") {
          R(s, "a boolean", i);
          break;
        }
        t[s] = i;
        break;
      case "autoPlaceholder":
        if (typeof i != "string" || !fe.has(i)) {
          const n = Array.from(fe).join(", ");
          R("autoPlaceholder", `one of ${n}`, i);
          break;
        }
        t[s] = i;
        break;
      case "containerClass":
      case "searchInputClass":
      case "countryNameLocale":
        if (typeof i != "string") {
          R(s, "a string", i);
          break;
        }
        t[s] = i;
        break;
      case "countryOrder":
        (i === null || Ce(s, i)) && (t[s] = i);
        break;
      case "customPlaceholder":
      case "geoIpLookup":
      case "hiddenInput":
      case "loadUtils":
        if (i !== null && !ms(i)) {
          R(s, "a function or null", i);
          break;
        }
        t[s] = i;
        break;
      case "dropdownContainer":
        if (i !== null && !gs(i)) {
          R("dropdownContainer", "an HTMLElement or null", i);
          break;
        }
        t[s] = i;
        break;
      case "excludeCountries":
      case "onlyCountries":
        Ce(s, i) && (t[s] = i);
        break;
      case "i18n":
        if (i && !pe(i)) {
          R("i18n", "an object", i);
          break;
        }
        t[s] = i;
        break;
      case "initialCountry": {
        if (typeof i != "string") {
          R("initialCountry", "a string", i);
          break;
        }
        const n = i.toLowerCase();
        if (n && n !== It.AUTO && !Le(n)) {
          R("initialCountry", "a valid ISO2 country code or 'auto'", i);
          break;
        }
        t[s] = i;
        break;
      }
      case "placeholderNumberType":
        if (typeof i != "string" || !kt.has(i)) {
          const n = Array.from(kt).join(", ");
          R("placeholderNumberType", `one of ${n}`, i);
          break;
        }
        t[s] = i;
        break;
      case "allowedNumberTypes":
        if (i !== null) {
          if (!Array.isArray(i)) {
            R("allowedNumberTypes", "an array of number types or null", i);
            break;
          }
          let n = !0;
          for (const l of i)
            if (typeof l != "string" || !kt.has(l)) {
              const u = Array.from(kt).join(", ");
              R("allowedNumberTypes", `an array of valid number types (${u})`, l), n = !1;
              break;
            }
          n && (t[s] = i);
        } else
          t[s] = null;
        break;
    }
  }
  return t;
}, ws = (a) => {
  a.dropdownAlwaysOpen && (a.useFullscreenPopup = !1, a.allowDropdown = !0), a.useFullscreenPopup ? a.fixDropdownWidth = !1 : we() && (a.fixDropdownWidth = !0), a.onlyCountries.length === 1 && (a.initialCountry = a.onlyCountries[0]), a.separateDialCode && (a.nationalMode = !1), a.allowDropdown && !a.showFlags && !a.separateDialCode && (a.nationalMode = !1), a.useFullscreenPopup && !a.dropdownContainer && (a.dropdownContainer = document.body), a.i18n = { ...Cs, ...a.i18n };
}, pt = (a) => a.replace(/\D/g, ""), De = (a = "") => a.normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase(), Ns = () => typeof navigator < "u" ? /Android/i.test(navigator.userAgent) : !1, Ls = (a, t) => {
  const s = De(t), i = [], n = [], l = [], u = [], c = [], d = [];
  for (const m of a)
    m.iso2 === s ? i.push(m) : m.normalisedName.startsWith(s) ? n.push(m) : m.normalisedName.includes(s) ? l.push(m) : s === m.dialCode || s === m.dialCodePlus ? u.push(m) : m.dialCodePlus.includes(s) ? c.push(m) : m.initials.includes(s) && d.push(m);
  const f = (m, D) => m.priority - D.priority;
  return [
    ...i.sort(f),
    ...n.sort(f),
    ...l.sort(f),
    ...u.sort(f),
    ...c.sort(f),
    ...d.sort(f)
  ];
}, Ds = (a, t) => {
  const s = t.toLowerCase();
  for (const i of a)
    if (i.name.toLowerCase().startsWith(s))
      return i;
  return null;
}, Gt = (a) => Object.keys(a).filter((t) => !!a[t]).join(" "), L = (a, t, s) => {
  const i = document.createElement(a);
  return t && Object.entries(t).forEach(
    ([n, l]) => i.setAttribute(n, l)
  ), s && s.appendChild(i), i;
}, As = () => `
  <svg class="iti__search-icon-svg" width="14" height="14" viewBox="0 0 24 24" focusable="false" ${w.HIDDEN}="true">
    <circle cx="11" cy="11" r="7" />
    <line x1="21" y1="21" x2="16.65" y2="16.65" />
  </svg>`, _s = (a) => {
  const t = `iti-${a}-clear-mask`;
  return `
    <svg class="iti__search-clear-svg" width="12" height="12" viewBox="0 0 16 16" ${w.HIDDEN}="true" focusable="false">
      <mask id="${t}" maskUnits="userSpaceOnUse">
        <rect width="16" height="16" fill="white" />
        <path d="M5.2 5.2 L10.8 10.8 M10.8 5.2 L5.2 10.8" stroke="black" stroke-linecap="round" class="iti__search-clear-x" />
      </mask>
      <circle cx="8" cy="8" r="8" class="iti__search-clear-bg" mask="url(#${t})" />
    </svg>`;
}, Os = () => `
  <svg class="iti__country-check-svg" width="14" height="14" viewBox="0 0 16 16" fill="currentColor" focusable="false" ${w.HIDDEN}="true">
    <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0m-3.97-3.03a.75.75 0 0 0-1.08.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-.01-1.05z"/>
  </svg>`, ye = () => `
  <svg width="256" height="256" viewBox="0 0 512 512" class="iti__globe-svg">
    <path d="M508 213a240 240 0 0 0-449-87l-2 5-2 5c-8 14-13 30-17 46a65 65 0 0 1 56 4c16-10 35-19 56-27l9-3c-6 23-10 48-10 74h-16l4 6c3 4 5 8 6 13h6c0 22 3 44 8 65l2 10-25-10-4 5 12 18 9 3 6 2 8 3 9 26 1 2 16-7h1l-5-13-1-2c24 6 49 9 75 10v26l11 10 7 7v-30l1-13c22 0 44-3 65-8l10-2-21 48-1 1a317 317 0 0 1-14 23l-21 5h-2c6 16 7 33 1 50a240 240 0 0 0 211-265m-401-56-11 6c19-44 54-79 98-98-11 20-21 44-29 69-21 6-40 15-58 23m154 182v4c-29-1-57-6-81-13-7-25-12-52-13-81h94zm0-109h-94c1-29 6-56 13-81 24-7 52-12 81-13zm0-112c-22 1-44 4-65 8l-10 2 12-30 9-17 1-2a332 332 0 0 1 13-23c13-4 26-6 40-7zm187 69 6 4c4 12 6 25 6 38v1h-68c-1-26-4-51-10-74l48 20 1 1 14 8zm-14-44 10 20c-20-11-43-21-68-29-8-25-18-49-29-69 37 16 67 44 87 78M279 49h1c13 1 27 3 39 7l14 23 1 2a343 343 0 0 1 12 26l2 5 6 16c-23-6-48-9-74-10h-1zm0 87h1c29 1 56 6 81 13 7 24 12 51 12 80v1h-94zm2 207h-2v-94h95c-1 29-6 56-13 81-24 7-51 12-80 13m86 60-20 10c11-20 21-43 29-68 25-8 48-18 68-29-16 37-43 67-77 87m87-115-7 5-16 9-2 1a337 337 0 0 1-47 21c6-24 9-49 10-75h68c0 13-2 27-6 39"/>
    <path d="m261 428-2-2-22-21a40 40 0 0 0-32-11h-1a37 37 0 0 0-18 8l-1 1-4 2-2 2-5 4c-9-3-36-31-47-44s-32-45-34-55l3-2a151 151 0 0 0 11-9v-1a39 39 0 0 0 5-48l-3-3-11-19-3-4-5-7h-1l-3-3-4-3-5-2a35 35 0 0 0-16-3h-5c-4 1-14 5-24 11l-4 2-4 3-4 2c-9 8-17 17-18 27a380 380 0 0 0 212 259h3c12 0 25-10 36-21l10-12 6-11a39 39 0 0 0-8-40"/>
  </svg>`;
var A, v, lt, At, j, tt, rt, Ct, at, N, yt, Y, mt, _, H, g, Ae, _e, Yt, Oe, Te, Se, Pe, ve, Kt, _t, zt, Re, ke, Xt, qt, Ue, He, Qt, xe;
const Lt = class Lt {
  constructor(t, s, i) {
    I(this, g);
    // private
    I(this, A);
    I(this, v);
    I(this, lt);
    I(this, At);
    I(this, j);
    I(this, tt);
    I(this, rt);
    I(this, Ct);
    I(this, at);
    I(this, N);
    I(this, yt);
    I(this, Y);
    I(this, mt);
    I(this, _);
    I(this, H);
    p(this, tt, null), p(this, rt, null), p(this, _, null), p(this, H, null), this.highlightedItem = null, t.dataset.intlTelInputId = i.toString(), this.telInput = t, p(this, A, s), p(this, v, i), this.hadInitialPlaceholder = !!t.getAttribute("placeholder"), p(this, lt, !!this.telInput.closest("[dir=rtl]")), e(this, A).separateDialCode && p(this, At, this.telInput.style.paddingLeft);
  }
  // Validate that the provided element is an HTMLInputElement.
  static validateInput(t) {
    const s = t == null ? void 0 : t.tagName;
    if (!(!!t && typeof t == "object" && s === "INPUT" && typeof t.setAttribute == "function")) {
      const n = Object.prototype.toString.call(t);
      throw new TypeError(
        `The first argument must be an HTMLInputElement, not ${n}`
      );
    }
  }
  //* Generate all of the markup for the plugin: the selected country overlay, and the dropdown.
  generateMarkup(t) {
    p(this, j, t), this.telInput.classList.add("iti__tel-input"), this.telInput.hasAttribute("autocomplete") || this.telInput.setAttribute("autocomplete", "tel"), this.telInput.hasAttribute("inputmode") || this.telInput.setAttribute("inputmode", "tel");
    const s = r(this, g, Ae).call(this);
    r(this, g, _e).call(this, s), s.appendChild(this.telInput), r(this, g, Se).call(this), r(this, g, Pe).call(this, s);
  }
  //* Country search: Filter the countries according to the search query.
  filterCountriesByQuery(t) {
    let s;
    t === "" ? s = e(this, j) : s = Ls(e(this, j), t), r(this, g, He).call(this, s);
  }
  handleSearchChange() {
    e(this, tt) && clearTimeout(e(this, tt)), p(this, tt, setTimeout(() => {
      r(this, g, qt).call(this), p(this, tt, null);
    }, Ee.SEARCH_DEBOUNCE_MS));
  }
  handleSearchClear() {
    this.searchInput.value = "", this.searchInput.focus(), r(this, g, qt).call(this);
  }
  //* Check if an element is visible within it's container, else scroll until it is.
  scrollTo(t) {
    const s = this.countryList, i = document.documentElement.scrollTop, n = s.offsetHeight, l = s.getBoundingClientRect().top + i, u = l + n, c = t.offsetHeight, d = t.getBoundingClientRect().top + i, f = d + c, m = d - l + s.scrollTop;
    if (d < l)
      s.scrollTop = m;
    else if (f > u) {
      const D = n - c;
      s.scrollTop = m - D;
    }
  }
  //* Remove highlighting from the previous list item and highlight the new one.
  highlightListItem(t, s) {
    const i = this.highlightedItem;
    if (i && i.classList.remove(b.HIGHLIGHT), this.highlightedItem = t, this.highlightedItem && (this.highlightedItem.classList.add(b.HIGHLIGHT), e(this, A).countrySearch)) {
      const n = this.highlightedItem.getAttribute("id") || "";
      this.searchInput.setAttribute(w.ACTIVE_DESCENDANT, n);
    }
    s && this.highlightedItem.focus();
  }
  //* Highlight the next/prev item in the list (and ensure it is visible).
  handleUpDownKey(t) {
    var i, n;
    let s = t === S.ARROW_UP ? (i = this.highlightedItem) == null ? void 0 : i.previousElementSibling : (n = this.highlightedItem) == null ? void 0 : n.nextElementSibling;
    !s && this.countryList.childElementCount > 1 && (s = t === S.ARROW_UP ? this.countryList.lastElementChild : this.countryList.firstElementChild), s && (this.scrollTo(s), this.highlightListItem(s, !1));
  }
  destroy() {
    this.telInput.iti = void 0, delete this.telInput.dataset.intlTelInputId, e(this, A).separateDialCode && (this.telInput.style.paddingLeft = e(this, At));
    const t = this.telInput.parentNode;
    t.before(this.telInput), t.remove(), this.telInput = null, this.countryContainer = null, this.selectedCountry = null, this.selectedCountryInner = null, this.searchInput = null, this.searchClearButton = null, this.countryList = null, this.hiddenInput = null, this.hiddenInputCountry = null, this.highlightedItem = null, p(this, Ct, null), p(this, at, null), p(this, N, null), p(this, yt, null), p(this, Y, null), p(this, mt, null), p(this, _, null), p(this, H, null);
    for (const s of e(this, j))
      delete s.nodeById[e(this, v)];
    p(this, j, null);
  }
  // UI: Open the dropdown (DOM only).
  openDropdown() {
    const {
      countrySearch: t,
      dropdownAlwaysOpen: s,
      dropdownContainer: i
    } = e(this, A);
    if (r(this, g, Yt).call(this), i)
      r(this, g, xe).call(this);
    else {
      const n = r(this, g, Qt).call(this), l = this.telInput.offsetHeight + J.DROPDOWN_MARGIN;
      n ? e(this, N).style.top = `${l}px` : e(this, N).style.bottom = `${l}px`;
    }
    if (e(this, N).classList.remove(b.HIDE), this.selectedCountry.setAttribute(w.EXPANDED, "true"), t) {
      const n = this.countryList.firstElementChild;
      n && (this.highlightListItem(n, !1), this.countryList.scrollTop = 0), s || this.searchInput.focus();
    }
    e(this, at).classList.add(b.ARROW_UP);
  }
  // UI: Close the dropdown (DOM only).
  closeDropdown() {
    const { countrySearch: t, dropdownContainer: s } = e(this, A);
    e(this, N).classList.add(b.HIDE), this.selectedCountry.setAttribute(w.EXPANDED, "false"), t && (this.searchInput.removeAttribute(w.ACTIVE_DESCENDANT), this.highlightedItem && (this.highlightedItem.classList.remove(b.HIGHLIGHT), this.highlightedItem = null)), e(this, at).classList.remove(b.ARROW_UP), s ? (e(this, _).remove(), e(this, _).style.top = "", e(this, _).style.bottom = "", e(this, _).style.paddingLeft = "", e(this, _).style.paddingRight = "") : (e(this, N).style.top = "", e(this, N).style.bottom = "");
  }
  // UI: Whether the dropdown is currently closed (hidden).
  isDropdownClosed() {
    return e(this, N).classList.contains(b.HIDE);
  }
  setCountry(t) {
    const { allowDropdown: s, showFlags: i, separateDialCode: n, i18n: l } = e(this, A), { name: u, dialCode: c, iso2: d = "" } = t;
    if (s && r(this, g, Ue).call(this, d), this.selectedCountry) {
      const f = d && i ? `${b.FLAG} iti__${d}` : `${b.FLAG} ${b.GLOBE}`;
      let m, D, P;
      d ? (D = u, m = l.selectedCountryAriaLabel.replace("${countryName}", u).replace("${dialCode}", `+${c}`), P = i ? "" : ye()) : (D = l.noCountrySelected, m = l.noCountrySelected, P = ye()), this.selectedCountryInner.className = f, this.selectedCountry.setAttribute("title", D), this.selectedCountry.setAttribute(w.LABEL, m), this.selectedCountryInner.innerHTML = P;
    }
    if (n) {
      const f = c ? `+${c}` : "";
      e(this, Ct).textContent = f, r(this, g, Kt).call(this);
    }
  }
};
A = new WeakMap(), v = new WeakMap(), lt = new WeakMap(), At = new WeakMap(), j = new WeakMap(), tt = new WeakMap(), rt = new WeakMap(), Ct = new WeakMap(), at = new WeakMap(), N = new WeakMap(), yt = new WeakMap(), Y = new WeakMap(), mt = new WeakMap(), _ = new WeakMap(), H = new WeakMap(), g = new WeakSet(), Ae = function() {
  const {
    allowDropdown: t,
    showFlags: s,
    containerClass: i,
    useFullscreenPopup: n
  } = e(this, A), l = Gt({
    iti: !0,
    "iti--allow-dropdown": t,
    "iti--show-flags": s,
    "iti--inline-dropdown": !n,
    [i]: !!i
  }), u = L("div", { class: l });
  return e(this, lt) && u.setAttribute("dir", "ltr"), this.telInput.before(u), u;
}, _e = function(t) {
  const { allowDropdown: s, separateDialCode: i, showFlags: n } = e(this, A);
  if (s || n || i) {
    this.countryContainer = L(
      "div",
      // visibly hidden until we measure it's width to set the input padding correctly
      { class: `iti__country-container ${b.V_HIDE}` },
      t
    ), s ? (this.selectedCountry = L(
      "button",
      {
        type: "button",
        class: "iti__selected-country",
        [w.EXPANDED]: "false",
        [w.LABEL]: e(this, A).i18n.noCountrySelected,
        [w.HASPOPUP]: "dialog",
        [w.CONTROLS]: `iti-${e(this, v)}__dropdown-content`
      },
      this.countryContainer
    ), this.telInput.disabled && this.selectedCountry.setAttribute("disabled", "true")) : this.selectedCountry = L(
      "div",
      { class: "iti__selected-country" },
      this.countryContainer
    );
    const l = L(
      "div",
      { class: "iti__selected-country-primary" },
      this.selectedCountry
    );
    this.selectedCountryInner = L(
      "div",
      { class: b.FLAG },
      l
    ), s && p(this, at, L(
      "div",
      { class: "iti__arrow", [w.HIDDEN]: "true" },
      l
    )), i && p(this, Ct, L(
      "div",
      { class: "iti__selected-dial-code" },
      this.selectedCountry
    )), s && r(this, g, Oe).call(this);
  }
}, Yt = function() {
  const { fixDropdownWidth: t } = e(this, A);
  if (t && !e(this, N).style.width) {
    const s = this.telInput.offsetWidth;
    s > 0 && (e(this, N).style.width = `${s}px`);
  }
}, Oe = function() {
  const {
    fixDropdownWidth: t,
    useFullscreenPopup: s,
    countrySearch: i,
    i18n: n,
    dropdownContainer: l,
    containerClass: u
  } = e(this, A), c = t ? "" : "iti--flexible-dropdown-width";
  if (p(this, N, L("div", {
    id: `iti-${e(this, v)}__dropdown-content`,
    class: `iti__dropdown-content ${b.HIDE} ${c}`,
    role: "dialog",
    [w.MODAL]: "true"
  })), e(this, lt) && e(this, N).setAttribute("dir", "rtl"), i && r(this, g, Te).call(this), this.countryList = L(
    "ul",
    {
      class: "iti__country-list",
      id: `iti-${e(this, v)}__country-listbox`,
      role: "listbox",
      [w.LABEL]: n.countryListAriaLabel
    },
    e(this, N)
  ), r(this, g, ve).call(this), i && r(this, g, Xt).call(this), s || (r(this, g, Yt).call(this), p(this, rt, r(this, g, ke).call(this)), i && (e(this, N).style.height = `${e(this, rt)}px`)), l) {
    const d = Gt({
      iti: !0,
      "iti--container": !0,
      "iti--fullscreen-popup": s,
      "iti--inline-dropdown": !s,
      [u]: !!u
    });
    p(this, _, L("div", { class: d })), e(this, _).appendChild(e(this, N));
  } else
    this.countryContainer.appendChild(e(this, N));
}, Te = function() {
  const { i18n: t, searchInputClass: s } = e(this, A), i = L(
    "div",
    { class: "iti__search-input-wrapper" },
    e(this, N)
  );
  p(this, yt, L(
    "span",
    {
      class: "iti__search-icon",
      [w.HIDDEN]: "true"
    },
    i
  )), e(this, yt).innerHTML = As(), this.searchInput = L(
    "input",
    {
      id: `iti-${e(this, v)}__search-input`,
      // Chrome says inputs need either a name or an id
      type: "search",
      class: `iti__search-input ${s}`,
      placeholder: t.searchPlaceholder,
      // role=combobox + aria-autocomplete=list + aria-activedescendant allows maintaining focus on the search input while allowing users to navigate search results with up/down keyboard keys
      role: "combobox",
      [w.EXPANDED]: "true",
      [w.LABEL]: t.searchPlaceholder,
      [w.CONTROLS]: `iti-${e(this, v)}__country-listbox`,
      [w.AUTOCOMPLETE]: "list",
      autocomplete: "off"
    },
    i
  ), this.searchClearButton = L(
    "button",
    {
      type: "button",
      class: `iti__search-clear ${b.HIDE}`,
      [w.LABEL]: t.clearSearchAriaLabel,
      tabindex: "-1"
    },
    i
  ), this.searchClearButton.innerHTML = _s(e(this, v)), p(this, mt, L(
    "span",
    { class: "iti__a11y-text" },
    e(this, N)
  )), p(this, Y, L(
    "div",
    {
      class: `iti__no-results ${b.HIDE}`,
      [w.HIDDEN]: "true"
      // all a11y messaging happens in this.#searchResultsA11yText
    },
    e(this, N)
  )), e(this, Y).textContent = t.searchEmptyState;
}, Se = function() {
  this.countryContainer && (r(this, g, Kt).call(this), this.countryContainer.classList.remove(b.V_HIDE));
}, Pe = function(t) {
  var i, n;
  const { hiddenInput: s } = e(this, A);
  if (s) {
    const l = this.telInput.getAttribute("name") || "", u = s(l);
    if (u.phone) {
      const c = (i = this.telInput.form) == null ? void 0 : i.querySelector(
        `input[name="${u.phone}"]`
      );
      c ? this.hiddenInput = c : (this.hiddenInput = L("input", {
        type: "hidden",
        name: u.phone
      }), t.appendChild(this.hiddenInput));
    }
    if (u.country) {
      const c = (n = this.telInput.form) == null ? void 0 : n.querySelector(
        `input[name="${u.country}"]`
      );
      c ? this.hiddenInputCountry = c : (this.hiddenInputCountry = L("input", {
        type: "hidden",
        name: u.country
      }), t.appendChild(this.hiddenInputCountry));
    }
  }
}, //* For each country: add a country list item <li> to the countryList <ul> container.
ve = function() {
  const t = document.createDocumentFragment();
  for (let s = 0; s < e(this, j).length; s++) {
    const i = e(this, j)[s], n = Gt({
      [b.COUNTRY_ITEM]: !0
    }), l = L("li", {
      id: `iti-${e(this, v)}__item-${i.iso2}`,
      class: n,
      tabindex: "-1",
      role: "option",
      [w.SELECTED]: "false"
    });
    l.dataset.dialCode = i.dialCode, l.dataset.countryCode = i.iso2, i.nodeById[e(this, v)] = l, e(this, A).showFlags && L("div", { class: `${b.FLAG} iti__${i.iso2}` }, l);
    const u = L("span", { class: "iti__country-name" }, l);
    u.textContent = `${i.name} `;
    const c = L("span", { class: "iti__dial-code" }, u);
    e(this, lt) && c.setAttribute("dir", "ltr"), c.textContent = `(+${i.dialCode})`, t.appendChild(l);
  }
  this.countryList.appendChild(t);
}, //* Update the input padding to make space for the selected country/dial code.
Kt = function() {
  if (this.selectedCountry) {
    const t = e(this, A).separateDialCode ? J.SANE_SELECTED_WITH_DIAL_WIDTH : J.SANE_SELECTED_NO_DIAL_WIDTH, i = (this.selectedCountry.offsetWidth || r(this, g, Re).call(this) || t) + J.INPUT_PADDING_EXTRA_LEFT;
    this.telInput.style.paddingLeft = `${i}px`;
  }
}, _t = new WeakSet(), zt = function() {
  let t;
  try {
    t = window.top.document.body;
  } catch {
    t = document.body;
  }
  return t;
}, //* When input is in a hidden container during init, we cannot calculate the selected country width.
//* Fix: clone the markup, make it invisible, add it to the end of the DOM, and then measure it's width.
//* To get the right styling to apply, all we need is a shallow clone of the container,
//* and then to inject a deep clone of the selectedCountry element.
Re = function() {
  var t;
  if (this.telInput.parentNode) {
    const s = r(t = Lt, _t, zt).call(t), i = this.telInput.parentNode.cloneNode(
      !1
    );
    i.style.visibility = "hidden", s.appendChild(i);
    const n = this.countryContainer.cloneNode();
    i.appendChild(n);
    const l = this.selectedCountry.cloneNode(
      !0
    );
    n.appendChild(l);
    const u = l.offsetWidth;
    return s.removeChild(i), u;
  }
  return 0;
}, // this is run before we add the dropdown to the DOM
ke = function() {
  var n;
  const t = r(n = Lt, _t, zt).call(n);
  e(this, N).classList.remove(b.HIDE);
  const s = L("div", { class: "iti iti--inline-dropdown" });
  s.appendChild(e(this, N)), s.style.visibility = "hidden", t.appendChild(s);
  const i = e(this, N).offsetHeight;
  return t.removeChild(s), s.style.visibility = "", e(this, N).classList.add(b.HIDE), i > 0 ? i : J.SANE_DROPDOWN_HEIGHT;
}, //* Update search results text (for a11y).
Xt = function() {
  const { i18n: t } = e(this, A), s = this.countryList.childElementCount;
  e(this, mt).textContent = t.searchSummaryAria(s);
}, // Search input handlers
qt = function() {
  const t = this.searchInput.value.trim();
  this.filterCountriesByQuery(t), this.searchInput.value ? this.searchClearButton.classList.remove(b.HIDE) : this.searchClearButton.classList.add(b.HIDE);
}, // Update the selected list item in the dropdown
Ue = function(t) {
  var s;
  if (e(this, H) && e(this, H).dataset.countryCode !== t && (e(this, H).setAttribute(w.SELECTED, "false"), (s = e(this, H).querySelector(".iti__country-check")) == null || s.remove(), p(this, H, null)), t && !e(this, H)) {
    const i = this.countryList.querySelector(
      `[data-country-code="${t}"]`
    );
    if (i) {
      i.setAttribute(w.SELECTED, "true");
      const n = L(
        "span",
        { class: "iti__country-check", [w.HIDDEN]: "true" },
        i
      );
      n.innerHTML = Os(), p(this, H, i);
    }
  }
}, //* Country search: Filter the country list to the given array of countries.
He = function(t) {
  this.countryList.innerHTML = "";
  let s = !0;
  for (const i of t) {
    const n = i.nodeById[e(this, v)];
    n && (this.countryList.appendChild(n), s && (this.highlightListItem(n, !1), s = !1));
  }
  s ? (this.highlightListItem(null, !1), e(this, Y) && e(this, Y).classList.remove(b.HIDE)) : e(this, Y) && e(this, Y).classList.add(b.HIDE), this.countryList.scrollTop = 0, r(this, g, Xt).call(this);
}, Qt = function() {
  if (e(this, A).dropdownAlwaysOpen)
    return !0;
  const t = this.telInput.getBoundingClientRect(), s = t.top, i = window.innerHeight - t.bottom;
  return i >= e(this, rt) || i >= s;
}, // inject dropdown into container and apply positioning styles
xe = function() {
  const { dropdownContainer: t, useFullscreenPopup: s } = e(this, A);
  if (s) {
    if (window.innerWidth >= 500) {
      const i = this.telInput.getBoundingClientRect();
      e(this, _).style.paddingLeft = `${i.left}px`, e(this, _).style.paddingRight = `${window.innerWidth - i.right}px`;
    }
  } else {
    const i = this.telInput.getBoundingClientRect();
    e(this, _).style.left = `${i.left}px`, r(this, g, Qt).call(this) ? e(this, _).style.top = `${i.bottom + J.DROPDOWN_MARGIN}px` : (e(this, _).style.top = "unset", e(this, _).style.bottom = `${window.innerHeight - i.top + J.DROPDOWN_MARGIN}px`);
  }
  t.appendChild(e(this, _));
}, I(Lt, _t);
let Wt = Lt;
const Ts = (a) => {
  const { onlyCountries: t, excludeCountries: s } = a;
  if (t != null && t.length) {
    const i = t.map(
      (n) => n.toLowerCase()
    );
    return ht.filter(
      (n) => i.includes(n.iso2)
    );
  } else if (s != null && s.length) {
    const i = s.map(
      (n) => n.toLowerCase()
    );
    return ht.filter(
      (n) => !i.includes(n.iso2)
    );
  }
  return ht;
}, Ss = (a, t) => {
  const { countryNameLocale: s, i18n: i } = t;
  let n;
  try {
    typeof Intl < "u" && typeof Intl.DisplayNames == "function" ? n = new Intl.DisplayNames(s, {
      type: "region"
    }) : n = null;
  } catch (l) {
    console.error(l), n = null;
  }
  for (const l of a)
    l.name = i[l.iso2] || (n == null ? void 0 : n.of(l.iso2.toUpperCase())) || "";
}, Ps = (a) => {
  const t = /* @__PURE__ */ new Set();
  let s = 0;
  const i = {}, n = (u, c) => {
    if (!u || !c)
      return;
    c.length > s && (s = c.length), i.hasOwnProperty(c) || (i[c] = []);
    const d = i[c];
    d.includes(u) || d.push(u);
  }, l = [...a].sort((u, c) => u.priority - c.priority);
  for (const u of l) {
    t.has(u.dialCode) || t.add(u.dialCode);
    for (let c = 1; c < u.dialCode.length; c++) {
      const d = u.dialCode.substring(0, c);
      n(u.iso2, d);
    }
    if (n(u.iso2, u.dialCode), u.areaCodes) {
      const c = i[u.dialCode][0];
      for (const d of u.areaCodes) {
        for (let f = 1; f < d.length; f++) {
          const m = d.substring(0, f), D = u.dialCode + m;
          n(c, D), n(u.iso2, D);
        }
        n(u.iso2, u.dialCode + d);
      }
    }
  }
  return { dialCodes: t, dialCodeMaxLen: s, dialCodeToIso2Map: i };
}, vs = (a, t) => {
  t.countryOrder && (t.countryOrder = t.countryOrder.map(
    (s) => s.toLowerCase()
  )), a.sort((s, i) => {
    const { countryOrder: n } = t;
    if (n) {
      const l = n.indexOf(s.iso2), u = n.indexOf(i.iso2), c = l > -1, d = u > -1;
      if (c || d)
        return c && d ? l - u : c ? -1 : 1;
    }
    return s.name.localeCompare(i.name);
  });
}, Rs = (a) => {
  for (const t of a)
    t.normalisedName = De(t.name), t.initials = t.normalisedName.split(/[^a-z]/).map((s) => s[0]).join(""), t.dialCodePlus = `+${t.dialCode}`;
}, ks = /* @__PURE__ */ new Set([
  "800",
  "808",
  "870",
  "881",
  "882",
  "883",
  "888",
  "979"
]), me = (a) => {
  const t = pt(a).slice(0, 3);
  return a.startsWith("+") && ks.has(t);
}, ge = (a, t, s, i) => {
  let n = a;
  if (s && t) {
    const l = `+${i.dialCode}`, u = n[l.length] === " " || n[l.length] === "-" ? l.length + 1 : l.length;
    n = n.substring(u);
  }
  return n;
}, Us = (a, t, s, i, n) => {
  const l = s ? s.formatNumberAsYouType(a, i.iso2) : a, { dialCode: u } = i;
  return n && t.charAt(0) !== "+" && l.includes(`+${u}`) ? (l.split(`+${u}`)[1] || "").trim() : l;
}, Hs = (a, t, s, i) => {
  if (s === 0 && !i)
    return 0;
  let n = 0;
  for (let l = 0; l < t.length; l++) {
    if (/[+0-9]/.test(t[l]) && n++, n === a && !i)
      return l + 1;
    if (i && n === a + 1)
      return l;
  }
  return t.length;
}, xs = [
  "800",
  "822",
  "833",
  "844",
  "855",
  "866",
  "877",
  "880",
  "881",
  "882",
  "883",
  "884",
  "885",
  "886",
  "887",
  "888",
  "889"
], Ie = (a) => {
  const t = pt(a);
  if (t.startsWith(be.NANP) && t.length >= 4) {
    const s = t.substring(1, 4);
    return xs.includes(s);
  }
  return !1;
};
var k, Ot, Jt;
class Ms {
  constructor() {
    I(this, Ot);
    I(this, k);
  }
  // Denormalise ASCII 0-9 to the user's numeral set, if known. If not known, return the string as-is.
  denormalise(t, s) {
    if (e(this, k) || r(this, Ot, Jt).call(this, s), e(this, k) === "ascii")
      return t;
    const i = e(this, k) === "arabic-indic" ? 1632 : 1776;
    return t.replace(/[0-9]/g, (n) => String.fromCharCode(i + Number(n)));
  }
  // Normalize Eastern Arabic (U+0660-0669) and Persian/Extended Arabic-Indic (U+06F0-06F9) numerals to ASCII 0-9
  normalise(t) {
    if (!t)
      return "";
    if (r(this, Ot, Jt).call(this, t), e(this, k) === "ascii")
      return t;
    const s = e(this, k) === "arabic-indic" ? 1632 : 1776, i = e(this, k) === "arabic-indic" ? /[\u0660-\u0669]/g : /[\u06F0-\u06F9]/g;
    return t.replace(i, (n) => String.fromCharCode(48 + (n.charCodeAt(0) - s)));
  }
  isAscii() {
    return e(this, k) === "ascii";
  }
}
k = new WeakMap(), Ot = new WeakSet(), // If any Arabic-Indic digits, then label it as that set. Same for Persian. Otherwise assume ASCII.
Jt = function(t) {
  /[\u0660-\u0669]/.test(t) ? p(this, k, "arabic-indic") : /[\u06F0-\u06F9]/.test(t) ? p(this, k, "persian") : p(this, k, "ascii");
};
let Ws = 0;
const Bs = new Set(ht.map((a) => a.iso2)), Vt = (a) => Bs.has(a);
var h, E, Tt, B, St, gt, Pt, vt, y, $, F, x, et, U, st, ut, it, ct, o, W, nt, Me, We, Be, Zt, $e, Fe, Ge, Ve, je, te, Ye, Ke, ze, Xe, qe, Et, bt, Qe, Je, Ze, ts, es, ss, is, ns, dt, Ut, os, ee, Z, se, Ht, ie, ot, ne, wt, K, oe, Nt, le, ls, rs, as, us, re, cs;
const ft = class ft {
  constructor(t, s = {}) {
    I(this, o);
    //* PRIVATE FIELDS
    I(this, h);
    I(this, E);
    I(this, Tt);
    // country data
    I(this, B);
    I(this, St);
    I(this, gt);
    I(this, Pt);
    I(this, vt);
    I(this, y);
    I(this, $);
    I(this, F);
    I(this, x);
    I(this, et);
    I(this, U);
    I(this, st);
    I(this, ut);
    I(this, it);
    I(this, ct);
    this.id = Ws++, Wt.validateInput(t);
    const i = bs(s);
    p(this, E, { ...ae, ...i }), ws(e(this, E)), p(this, h, new Wt(t, e(this, E), this.id)), p(this, Tt, Ns()), p(this, U, new Ms()), this.promise = r(this, o, Me).call(this, e(this, E)), p(this, B, Ts(e(this, E)));
    const { dialCodes: n, dialCodeMaxLen: l, dialCodeToIso2Map: u } = Ps(
      e(this, B)
    );
    p(this, Pt, n), p(this, St, l), p(this, gt, u), p(this, vt, new Map(e(this, B).map((c) => [c.iso2, c]))), r(this, o, We).call(this);
  }
  //********************
  //*  PUBLIC METHODS
  //********************
  //* Remove plugin.
  destroy() {
    e(this, h).telInput && (e(this, E).allowDropdown && r(this, o, ot).call(this, !0), e(this, x).abort(), p(this, x, null), e(this, h).destroy(), C.instances instanceof Map ? C.instances.delete(this.id) : delete C.instances[this.id]);
  }
  // check if the instance is still valid (not destroyed/unmounted)
  isActive() {
    var t;
    return !!((t = e(this, h)) != null && t.telInput);
  }
  //* Get the extension from the current number.
  getExtension() {
    return C.utils && e(this, h).telInput ? C.utils.getExtension(
      r(this, o, K).call(this),
      e(this, y).iso2
    ) : "";
  }
  //* Format the number to the given format.
  getNumber(t) {
    if (C.utils && e(this, h).telInput) {
      const { iso2: s } = e(this, y), i = r(this, o, K).call(this), n = C.utils.formatNumber(
        i,
        s,
        t
      ), l = e(this, h).telInput.value;
      return e(this, U).denormalise(n, l);
    }
    return "";
  }
  //* Get the type of the entered number e.g. landline/mobile.
  getNumberType() {
    return C.utils && e(this, h).telInput ? C.utils.getNumberType(
      r(this, o, K).call(this),
      e(this, y).iso2
    ) : he.UNKNOWN_NUMBER_TYPE;
  }
  //* Get the country data for the currently selected country.
  getSelectedCountryData() {
    return e(this, y);
  }
  //* Get the validation error.
  getValidationError() {
    if (C.utils && e(this, h).telInput) {
      const { iso2: t } = e(this, y);
      return C.utils.getValidationError(r(this, o, K).call(this), t);
    }
    return he.UNKNOWN_VALIDATION_ERROR;
  }
  //* Validate the input val using number length only
  isValidNumber() {
    const { dialCode: t, iso2: s } = e(this, y);
    if (C.utils && e(this, h).telInput) {
      const i = r(this, o, K).call(this), n = C.utils.getCoreNumber(i, s);
      if (n) {
        if (t === Ft.DIAL_CODE && n[0] === Ft.MOBILE_PREFIX && n.length !== Ft.MOBILE_CORE_LENGTH)
          return !1;
        if (!Q.ALPHA_UNICODE.test(i) && t) {
          const u = i.startsWith("+") ? i.slice(1 + t.length) : i, c = pt(u).length;
          if (n.length > c)
            return !1;
        }
      }
    }
    return r(this, o, re).call(this, !1);
  }
  //* Validate the input val with precise validation
  isValidNumberPrecise() {
    return r(this, o, re).call(this, !0);
  }
  //* Update the selected country, and update the input val accordingly.
  setCountry(t) {
    if (!e(this, h).telInput)
      return;
    const s = t == null ? void 0 : t.toLowerCase();
    if (!Vt(s))
      throw new Error(`Invalid country code: '${s}'`);
    const i = e(this, y).iso2;
    if (t && s !== i || !t && i) {
      if (r(this, o, Z).call(this, s), r(this, o, ne).call(this, e(this, y).dialCode), e(this, E).formatOnDisplay) {
        const l = r(this, o, W).call(this);
        r(this, o, dt).call(this, l);
      }
      r(this, o, Nt).call(this);
    }
  }
  //* Set the input value and update the country.
  setNumber(t) {
    if (!e(this, h).telInput)
      return;
    const s = e(this, U).normalise(t), i = r(this, o, Ut).call(this, s);
    r(this, o, dt).call(this, s), i && r(this, o, Nt).call(this), r(this, o, Et).call(this, Rt.INPUT, { isSetNumber: !0 });
  }
  //* Set the placeholder number typ
  setPlaceholderNumberType(t) {
    e(this, h).telInput && (e(this, E).placeholderNumberType = t, r(this, o, Ht).call(this));
  }
  setDisabled(t) {
    e(this, h).telInput && (e(this, h).telInput.disabled = t, t ? e(this, h).selectedCountry.setAttribute("disabled", "true") : e(this, h).selectedCountry.removeAttribute("disabled"));
  }
  //********************
  //*  STATIC METHODS
  //********************
  // Internal instance notification used by utils/geoip loaders.
  // Kept public so module-level helpers (e.g. attachUtils) can call it, while still allowing
  // access to private instance methods.
  static forEachInstance(t, ...s) {
    const i = C.instances, n = i instanceof Map ? Array.from(i.values()) : Object.values(i), l = s[0];
    n.forEach((u) => {
      var c, d, f, m;
      if (u instanceof ft)
        switch (t) {
          case "handleUtils":
            r(c = u, o, rs).call(c);
            break;
          case "handleUtilsFailure":
            r(d = u, o, as).call(d, l);
            break;
          case "handleAutoCountry":
            r(f = u, o, le).call(f);
            break;
          case "handleAutoCountryFailure":
            r(m = u, o, ls).call(m);
            break;
        }
    });
  }
};
h = new WeakMap(), E = new WeakMap(), Tt = new WeakMap(), B = new WeakMap(), St = new WeakMap(), gt = new WeakMap(), Pt = new WeakMap(), vt = new WeakMap(), y = new WeakMap(), $ = new WeakMap(), F = new WeakMap(), x = new WeakMap(), et = new WeakMap(), U = new WeakMap(), st = new WeakMap(), ut = new WeakMap(), it = new WeakMap(), ct = new WeakMap(), o = new WeakSet(), W = function() {
  const t = e(this, h).telInput.value.trim();
  return e(this, U).normalise(t);
}, nt = function(t) {
  const s = e(this, h).telInput.value;
  e(this, h).telInput.value = e(this, U).denormalise(t, s);
}, Me = function(t) {
  const { initialCountry: s, geoIpLookup: i, loadUtils: n } = t, l = s === It.AUTO && !!i, u = !!n && !C.utils;
  let c;
  l ? c = new Promise((f, m) => {
    p(this, st, f), p(this, ut, m);
  }) : (c = Promise.resolve(void 0), p(this, st, () => {
  }), p(this, ut, () => {
  }));
  let d;
  return u ? d = new Promise((f, m) => {
    p(this, it, f), p(this, ct, m);
  }) : (d = Promise.resolve(void 0), p(this, it, () => {
  }), p(this, ct, () => {
  })), Promise.all([c, d]);
}, We = function() {
  p(this, y, {}), p(this, x, new AbortController()), r(this, o, Be).call(this), e(this, h).generateMarkup(e(this, B)), r(this, o, Zt).call(this), r(this, o, $e).call(this), r(this, o, Ve).call(this), e(this, E).dropdownAlwaysOpen && r(this, o, bt).call(this);
}, //********************
//*  PRIVATE METHODS
//********************
//* Prepare all of the country data, including onlyCountries, excludeCountries, countryOrder options.
Be = function() {
  Ss(e(this, B), e(this, E)), vs(e(this, B), e(this, E)), Rs(e(this, B));
}, //* Set the initial state of the input value and the selected country by:
//* 1. Extracting a dial code from the given number
//* 2. Using explicit initialCountry
Zt = function(t = !1) {
  const s = e(this, h).telInput.getAttribute("value"), i = e(this, U).normalise(s), n = r(this, o, W).call(this), u = i && i.startsWith("+") && (!n || !n.startsWith("+")) ? i : n, c = r(this, o, wt).call(this, u), d = Ie(u), { initialCountry: f, geoIpLookup: m } = e(this, E), P = f === It.AUTO && m && !t, M = f.toLowerCase(), T = Vt(M);
  c ? d ? T ? r(this, o, Z).call(this, M) : P || r(this, o, Z).call(this, ps.ISO2) : r(this, o, Ut).call(this, u) : T ? r(this, o, Z).call(this, M) : P || r(this, o, Z).call(this, ""), u && r(this, o, dt).call(this, u);
}, //* Initialise the main event listeners: input keyup, and click selected country.
$e = function() {
  r(this, o, Ye).call(this), e(this, E).allowDropdown && r(this, o, Ge).call(this), (e(this, h).hiddenInput || e(this, h).hiddenInputCountry) && e(this, h).telInput.form && r(this, o, Fe).call(this);
}, //* Update hidden input on form submit.
Fe = function() {
  var s;
  const t = () => {
    e(this, h).hiddenInput && (e(this, h).hiddenInput.value = this.getNumber()), e(this, h).hiddenInputCountry && (e(this, h).hiddenInputCountry.value = e(this, y).iso2 || "");
  };
  (s = e(this, h).telInput.form) == null || s.addEventListener("submit", t, {
    signal: e(this, x).signal
  });
}, //* initialise the dropdown listeners.
Ge = function() {
  const t = e(this, x).signal, s = (u) => {
    e(this, h).isDropdownClosed() ? e(this, h).telInput.focus() : u.preventDefault();
  }, i = e(this, h).telInput.closest("label");
  i && i.addEventListener("click", s, { signal: t });
  const n = () => {
    e(this, h).isDropdownClosed() && !e(this, h).telInput.disabled && !e(this, h).telInput.readOnly && r(this, o, bt).call(this);
  };
  e(this, h).selectedCountry.addEventListener(
    "click",
    n,
    {
      signal: t
    }
  );
  const l = (u) => {
    const c = [S.ARROW_UP, S.ARROW_DOWN, S.SPACE, S.ENTER];
    e(this, h).isDropdownClosed() && c.includes(u.key) && (u.preventDefault(), u.stopPropagation(), r(this, o, bt).call(this)), u.key === S.TAB && r(this, o, ot).call(this);
  };
  e(this, h).countryContainer.addEventListener(
    "keydown",
    l,
    { signal: t }
  );
}, //* Init requests: utils script / geo ip lookup.
Ve = function() {
  const { loadUtils: t, initialCountry: s, geoIpLookup: i } = e(this, E);
  if (t && !C.utils) {
    const l = () => {
      var u;
      (u = C.attachUtils(t)) == null || u.catch(() => {
      });
    };
    if (C.documentReady())
      l();
    else {
      const u = () => {
        l();
      };
      window.addEventListener("load", u, {
        signal: e(this, x).signal
      });
    }
  } else
    e(this, it).call(this);
  s === It.AUTO && i && (e(this, y).iso2 ? e(this, st).call(this) : r(this, o, je).call(this));
}, //* Perform the geo ip lookup.
je = function() {
  if (C.autoCountry)
    r(this, o, le).call(this);
  else if (e(this, h).selectedCountryInner.classList.add(b.LOADING), !C.startedLoadingAutoCountry && (C.startedLoadingAutoCountry = !0, typeof e(this, E).geoIpLookup == "function")) {
    const t = (i = "") => {
      e(this, h).selectedCountryInner.classList.remove(b.LOADING);
      const n = i.toLowerCase();
      Vt(n) ? (C.autoCountry = n, setTimeout(() => ft.forEachInstance("handleAutoCountry"))) : ft.forEachInstance("handleAutoCountryFailure");
    }, s = () => {
      e(this, h).selectedCountryInner.classList.remove(b.LOADING), ft.forEachInstance("handleAutoCountryFailure");
    };
    e(this, E).geoIpLookup(t, s);
  }
}, te = function() {
  r(this, o, bt).call(this), e(this, h).searchInput.value = "+", e(this, h).filterCountriesByQuery("");
}, //* Initialize the tel input listeners.
Ye = function() {
  r(this, o, Ke).call(this), r(this, o, ze).call(this), r(this, o, Xe).call(this);
}, Ke = function() {
  const {
    strictMode: t,
    formatAsYouType: s,
    separateDialCode: i,
    allowDropdown: n,
    countrySearch: l
  } = e(this, E);
  let u = !1;
  Q.ALPHA_UNICODE.test(r(this, o, W).call(this)) && (u = !0);
  const c = (d) => {
    const f = r(this, o, W).call(this);
    if (e(this, Tt) && (d == null ? void 0 : d.data) === "+" && i && n && l) {
      const T = e(this, h).telInput.selectionStart || 0, z = f.substring(0, T - 1), G = f.substring(T);
      r(this, o, nt).call(this, z + G), r(this, o, te).call(this);
      return;
    }
    r(this, o, Ut).call(this, f) && r(this, o, Nt).call(this);
    const m = (d == null ? void 0 : d.data) && Q.NON_PLUS_NUMERIC.test(d.data), D = (d == null ? void 0 : d.inputType) === ce.PASTE && f;
    m || D && !t ? u = !0 : Q.NON_PLUS_NUMERIC.test(f) || (u = !1);
    const P = (d == null ? void 0 : d.detail) && d.detail.isSetNumber, M = e(this, U).isAscii();
    if (s && !u && !P && M) {
      const T = e(this, h).telInput.selectionStart || 0, G = f.substring(
        0,
        T
      ).replace(
        Q.NON_PLUS_NUMERIC_GLOBAL,
        ""
      ).length, X = (d == null ? void 0 : d.inputType) === ce.DELETE_FWD, O = r(this, o, K).call(this), q = Us(
        O,
        f,
        C.utils,
        e(this, y),
        i
      ), V = Hs(
        G,
        q,
        T,
        X
      );
      r(this, o, nt).call(this, q), e(this, h).telInput.setSelectionRange(V, V);
    }
    if (i && f.startsWith("+") && e(this, y).dialCode) {
      const T = ge(
        f,
        !0,
        i,
        e(this, y)
      );
      r(this, o, nt).call(this, T);
    }
  };
  e(this, h).telInput.addEventListener(
    "input",
    c,
    {
      signal: e(this, x).signal
    }
  );
}, ze = function() {
  const { strictMode: t, separateDialCode: s, allowDropdown: i, countrySearch: n } = e(this, E);
  if (t || s) {
    const l = (u) => {
      if (u.key && u.key.length === 1 && !u.altKey && !u.ctrlKey && !u.metaKey) {
        if (s && i && n && u.key === "+") {
          u.preventDefault(), r(this, o, te).call(this);
          return;
        }
        if (t) {
          const c = r(this, o, W).call(this), f = !c.startsWith("+") && e(this, h).telInput.selectionStart === 0 && u.key === "+", m = e(this, U).normalise(u.key), D = /^[0-9]$/.test(m), P = s ? D : f || D, M = e(this, h).telInput, T = M.selectionStart, z = M.selectionEnd, G = c.slice(0, T), X = c.slice(z), O = G + u.key + X, q = r(this, o, K).call(this, O), V = C.utils.getCoreNumber(
            q,
            e(this, y).iso2
          ), Bt = e(this, $) && V.length > e(this, $), hs = r(this, o, ee).call(this, q) !== null;
          (!P || Bt && !hs && !f) && u.preventDefault();
        }
      }
    };
    e(this, h).telInput.addEventListener("keydown", l, {
      signal: e(this, x).signal
    });
  }
}, Xe = function() {
  if (e(this, E).strictMode) {
    const t = (s) => {
      s.preventDefault();
      const i = e(this, h).telInput, n = i.selectionStart, l = i.selectionEnd, u = r(this, o, W).call(this), c = u.slice(0, n), d = u.slice(l), f = e(this, y).iso2, m = s.clipboardData.getData("text"), D = e(this, U).normalise(m), P = n === 0 && l > 0, M = !u.startsWith("+") || P, T = D.replace(Q.NON_PLUS_NUMERIC_GLOBAL, ""), z = T.startsWith("+"), G = T.replace(/\+/g, ""), X = z && M ? `+${G}` : G;
      let O = c + X + d;
      if (O.length > 5) {
        let V = C.utils.getCoreNumber(O, f);
        for (; V.length === 0 && O.length > 0; )
          O = O.slice(0, -1), V = C.utils.getCoreNumber(O, f);
        if (!V)
          return;
        if (e(this, $) && V.length > e(this, $))
          if (i.selectionEnd === u.length) {
            const Bt = V.length - e(this, $);
            O = O.slice(0, O.length - Bt);
          } else
            return;
      }
      r(this, o, nt).call(this, O);
      const q = n + X.length;
      i.setSelectionRange(q, q), i.dispatchEvent(new InputEvent("input", { bubbles: !0 }));
    };
    e(this, h).telInput.addEventListener("paste", t, {
      signal: e(this, x).signal
    });
  }
}, //* Adhere to the input's maxlength attr.
qe = function(t) {
  const s = Number(e(this, h).telInput.getAttribute("maxlength"));
  return s && t.length > s ? t.substring(0, s) : t;
}, //* Trigger a custom event on the input (typed via ItiEventMap).
Et = function(t, s = {}) {
  const i = new CustomEvent(t, {
    bubbles: !0,
    cancelable: !0,
    detail: s
  });
  e(this, h).telInput.dispatchEvent(i);
}, //* Open the dropdown.
bt = function() {
  const { dropdownContainer: t, useFullscreenPopup: s } = e(this, E);
  if (p(this, et, new AbortController()), e(this, h).openDropdown(), !s && t) {
    const i = () => r(this, o, ot).call(this);
    window.addEventListener("scroll", i, {
      signal: e(this, et).signal
    });
  }
  r(this, o, Qe).call(this), r(this, o, Et).call(this, Rt.OPEN_COUNTRY_DROPDOWN);
}, //* We only bind dropdown listeners when the dropdown is open.
Qe = function() {
  const t = e(this, et).signal;
  r(this, o, Je).call(this, t), r(this, o, Ze).call(this, t), e(this, E).dropdownAlwaysOpen || r(this, o, ts).call(this, t), r(this, o, es).call(this, t), e(this, E).countrySearch && r(this, o, ss).call(this, t);
}, //* When mouse over a list item, just highlight that one
//* we add the class "highlight", so if they hit "enter" we know which one to select.
Je = function(t) {
  const s = (i) => {
    var l;
    const n = (l = i.target) == null ? void 0 : l.closest(
      `.${b.COUNTRY_ITEM}`
    );
    n && e(this, h).highlightListItem(n, !1);
  };
  e(this, h).countryList.addEventListener(
    "mouseover",
    s,
    {
      signal: t
    }
  );
}, //* Listen for country selection.
Ze = function(t) {
  const s = (i) => {
    var l;
    const n = (l = i.target) == null ? void 0 : l.closest(
      `.${b.COUNTRY_ITEM}`
    );
    n && r(this, o, ie).call(this, n);
  };
  e(this, h).countryList.addEventListener("click", s, {
    signal: t
  });
}, //* Click off to close (except when this initial opening click is bubbling up).
//* We cannot just stopPropagation as it may be needed to close another instance.
ts = function(t) {
  const s = (i) => {
    !!i.target.closest(
      `#iti-${this.id}__dropdown-content`
    ) || r(this, o, ot).call(this);
  };
  setTimeout(() => {
    document.documentElement.addEventListener(
      "click",
      s,
      { signal: t }
    );
  }, 0);
}, //* Listen for up/down scrolling, enter to select, or escape to close.
//* Use keydown as keypress doesn't fire for non-char keys and we want to catch if they
//* just hit down and hold it to scroll down (no keyup event).
//* Listen on the document because that's where key events are triggered if no input has focus.
es = function(t) {
  let s = "", i = null;
  const n = (l) => {
    [
      S.ARROW_UP,
      S.ARROW_DOWN,
      S.ENTER,
      S.ESC
    ].includes(l.key) && (l.preventDefault(), l.stopPropagation(), l.key === S.ARROW_UP || l.key === S.ARROW_DOWN ? e(this, h).handleUpDownKey(l.key) : l.key === S.ENTER ? r(this, o, ns).call(this) : l.key === S.ESC && (r(this, o, ot).call(this), e(this, h).selectedCountry.focus())), !e(this, E).countrySearch && Q.HIDDEN_SEARCH_CHAR.test(l.key) && (l.stopPropagation(), i && clearTimeout(i), s += l.key.toLowerCase(), r(this, o, is).call(this, s), i = setTimeout(() => {
      s = "";
    }, Ee.HIDDEN_SEARCH_RESET_MS));
  };
  document.addEventListener("keydown", n, { signal: t });
}, //* Search input listeners when countrySearch enabled.
ss = function(t) {
  e(this, h).searchInput.addEventListener(
    "input",
    () => e(this, h).handleSearchChange(),
    { signal: t }
  ), e(this, h).searchClearButton.addEventListener(
    "click",
    () => e(this, h).handleSearchClear(),
    { signal: t }
  );
}, //* Hidden search (countrySearch disabled): Find the first list item whose name starts with the query string.
is = function(t) {
  const s = Ds(e(this, B), t);
  if (s) {
    const i = s.nodeById[this.id];
    e(this, h).highlightListItem(i, !1), e(this, h).scrollTo(i);
  }
}, //* Select the currently highlighted item.
ns = function() {
  e(this, h).highlightedItem && r(this, o, ie).call(this, e(this, h).highlightedItem);
}, //* Update the input's value to the given val (format first if possible)
//* NOTE: this is called from _setInitialState, handleUtils and setNumber.
dt = function(t) {
  const { formatOnDisplay: s, nationalMode: i, separateDialCode: n } = e(this, E);
  let l = t;
  if (s && C.utils && e(this, y)) {
    const u = me(t), c = i && !u || !l.startsWith("+") && !n, { NATIONAL: d, INTERNATIONAL: f } = C.utils.numberFormat, m = c ? d : f;
    l = C.utils.formatNumber(
      l,
      e(this, y).iso2,
      m
    );
  }
  l = r(this, o, oe).call(this, l), r(this, o, nt).call(this, l);
}, //* Check if need to select a new country based on the given number
//* Note: called from _setInitialState, keyup handler, setNumber.
Ut = function(t) {
  const s = r(this, o, ee).call(this, t);
  return s !== null ? r(this, o, Z).call(this, s) : !1;
}, // if there is a selected country, and the number doesn't start with a dial code, then add it
os = function(t) {
  const { dialCode: s, nationalPrefix: i } = e(this, y);
  if (t.startsWith("+") || !s)
    return t;
  const u = i && t.startsWith(i) && !e(this, E).separateDialCode ? t.substring(1) : t;
  return `+${s}${u}`;
}, //* Get the new country based on the input number, or return null if no change, or empty string if should be empty (e.g. if they type an invalid dial code).
ee = function(t) {
  const s = t.indexOf("+");
  let i = s ? t.substring(s) : t;
  const n = e(this, y).iso2, l = e(this, y).dialCode;
  i = r(this, o, os).call(this, i);
  const u = r(this, o, wt).call(this, i, !0), c = pt(i);
  if (u) {
    const d = pt(u), f = e(this, gt)[d];
    if (f.length === 1)
      return f[0] === n ? null : f[0];
    if (!n && e(this, F) && f.includes(e(this, F)))
      return e(this, F);
    if (l === be.NANP && Ie(c))
      return null;
    const { areaCodes: D, priority: P } = e(this, y);
    if (D) {
      const X = D.map(
        (O) => `${l}${O}`
      );
      for (const O of X)
        if (c.startsWith(O))
          return null;
    }
    const T = D && !(P === 0) && c.length > d.length, z = n && f.includes(n) && !T, G = n === f[0];
    if (!z && !G)
      return f[0];
  } else if (i.startsWith("+") && c.length) {
    const d = e(this, y).dialCode || "";
    return d && d.startsWith(c) ? null : "";
  } else if ((!i || i === "+") && !n && e(this, F))
    return e(this, F);
  return null;
}, //* Update the selected country, dial code (if separateDialCode), placeholder, title, and selected list item.
//* Note: called from _setInitialState, _updateCountryFromNumber, _selectListItem, setCountry.
Z = function(t) {
  const s = e(this, y).iso2 || "";
  return p(this, y, t ? e(this, vt).get(t) : {}), e(this, y).iso2 && p(this, F, e(this, y).iso2), e(this, h).setCountry(e(this, y)), r(this, o, Ht).call(this), r(this, o, se).call(this), s !== t;
}, //* Update the maximum valid number length for the currently selected country.
se = function() {
  const { strictMode: t, placeholderNumberType: s, allowedNumberTypes: i } = e(this, E), { iso2: n } = e(this, y);
  if (t && C.utils)
    if (n) {
      const l = C.utils.numberType[s];
      let u = C.utils.getExampleNumber(
        n,
        !1,
        l,
        !0
      ), c = u;
      for (; C.utils.isPossibleNumber(
        u,
        n,
        i
      ); )
        c = u, u += "0";
      const d = C.utils.getCoreNumber(c, n);
      p(this, $, d.length), n === "by" && p(this, $, d.length + 1);
    } else
      p(this, $, null);
}, //* Update the input placeholder to an example number from the currently selected country.
Ht = function() {
  const {
    autoPlaceholder: t,
    placeholderNumberType: s,
    nationalMode: i,
    customPlaceholder: n
  } = e(this, E), l = t === xt.AGGRESSIVE || !e(this, h).hadInitialPlaceholder && t === xt.POLITE;
  if (C.utils && l) {
    const u = C.utils.numberType[s];
    let c = e(this, y).iso2 ? C.utils.getExampleNumber(
      e(this, y).iso2,
      i,
      u
    ) : "";
    c = r(this, o, oe).call(this, c), typeof n == "function" && (c = n(c, e(this, y))), e(this, h).telInput.setAttribute("placeholder", c);
  }
}, //* Called when the user selects a list item from the dropdown.
ie = function(t) {
  const s = t.dataset[de.COUNTRY_CODE], i = r(this, o, Z).call(this, s);
  r(this, o, ot).call(this);
  const n = t.dataset[de.DIAL_CODE];
  if (r(this, o, ne).call(this, n), e(this, E).formatOnDisplay) {
    const l = r(this, o, W).call(this);
    r(this, o, dt).call(this, l);
  }
  e(this, h).telInput.focus(), i && r(this, o, Nt).call(this);
}, //* Close the dropdown and unbind any listeners.
ot = function(t) {
  e(this, h).isDropdownClosed() || e(this, E).dropdownAlwaysOpen && !t || (e(this, h).closeDropdown(), e(this, et).abort(), p(this, et, null), r(this, o, Et).call(this, Rt.CLOSE_COUNTRY_DROPDOWN));
}, //* Replace any existing dial code with the new one
//* Note: called from _selectListItem and setCountry
ne = function(t) {
  const s = r(this, o, W).call(this), i = `+${t}`;
  let n;
  if (s.startsWith("+")) {
    const l = r(this, o, wt).call(this, s);
    l ? n = s.replace(l, i) : n = i, r(this, o, nt).call(this, n);
  }
}, //* Try and extract a valid international dial code from a full telephone number.
//* Note: returns the raw string inc plus character and any whitespace/dots etc.
wt = function(t, s) {
  let i = "";
  if (t.startsWith("+")) {
    let n = "", l = !1;
    for (let u = 0; u < t.length; u++) {
      const c = t.charAt(u);
      if (/[0-9]/.test(c)) {
        if (n += c, !!!e(this, gt)[n])
          break;
        if (e(this, Pt).has(n)) {
          if (i = t.substring(0, u + 1), l = !0, !s)
            break;
        } else s && l && (i = t.substring(0, u + 1));
        if (n.length === e(this, St))
          break;
      }
    }
  }
  return i;
}, //* Get the input val, adding the dial code if separateDialCode is enabled.
K = function(t) {
  const s = t ? e(this, U).normalise(t) : r(this, o, W).call(this), { dialCode: i } = e(this, y);
  let n;
  const l = pt(s);
  return e(this, E).separateDialCode && !s.startsWith("+") && i && l ? n = `+${i}` : n = "", n + s;
}, //* Remove the dial code if separateDialCode is enabled also cap the length if the input has a maxlength attribute
oe = function(t) {
  const s = !!r(this, o, wt).call(this, t), i = ge(
    t,
    s,
    e(this, E).separateDialCode,
    e(this, y)
  );
  return r(this, o, qe).call(this, i);
}, //* Trigger the 'countrychange' event.
Nt = function() {
  r(this, o, Et).call(this, Rt.COUNTRY_CHANGE);
}, //**************************
//*  INTERNAL METHODS
//**************************
//* Called when the geoip call returns.
le = function() {
  var t;
  if (!e(this, h).telInput) {
    (t = e(this, st)) == null || t.call(this);
    return;
  }
  e(this, E).initialCountry === It.AUTO && C.autoCountry && (p(this, F, C.autoCountry), e(this, y).iso2 || e(this, h).selectedCountryInner.classList.contains(b.GLOBE) || this.setCountry(e(this, F)), e(this, st).call(this));
}, //* Called when the geoip call fails or times out.
ls = function() {
  var t;
  if (!e(this, h).telInput) {
    (t = e(this, ut)) == null || t.call(this);
    return;
  }
  r(this, o, Zt).call(this, !0), e(this, ut).call(this);
}, //* Called when the utils request completes.
rs = function() {
  var t;
  if (!e(this, h).telInput) {
    (t = e(this, it)) == null || t.call(this);
    return;
  }
  if (C.utils) {
    const s = r(this, o, W).call(this);
    s && r(this, o, dt).call(this, s), e(this, y).iso2 && (r(this, o, Ht).call(this), r(this, o, se).call(this));
  }
  e(this, it).call(this);
}, //* Called when the utils request fails or times out.
as = function(t) {
  var s;
  if (!e(this, h).telInput) {
    (s = e(this, ct)) == null || s.call(this, t);
    return;
  }
  e(this, ct).call(this, t);
}, us = function(t) {
  return C.utils ? C.utils.isPossibleNumber(
    t,
    e(this, y).iso2,
    e(this, E).allowedNumberTypes
  ) : null;
}, //* Shared internal validation logic to handle alpha character extension rules.
re = function(t) {
  if (!C.utils || !e(this, h).telInput)
    return null;
  const { allowNumberExtensions: s, allowPhonewords: i } = e(this, E), n = (d) => t ? r(this, o, cs).call(this, d) : r(this, o, us).call(this, d), l = r(this, o, K).call(this);
  if (!e(this, y).iso2 && !me(l) || !n(l))
    return !1;
  if (l.search(Q.ALPHA_UNICODE) > -1) {
    const d = e(this, y).iso2;
    return C.utils.getExtension(l, d) ? s : i;
  }
  return !0;
}, cs = function(t) {
  return C.utils ? C.utils.isValidNumber(
    t,
    e(this, y).iso2,
    e(this, E).allowedNumberTypes
  ) : null;
};
let Dt = ft;
const $s = (a) => {
  if (!C.utils && !C.startedLoadingUtilsScript) {
    let t;
    if (typeof a == "function")
      try {
        t = Promise.resolve(a());
      } catch (s) {
        return Promise.reject(s);
      }
    else
      return Promise.reject(
        new TypeError(
          `The argument passed to attachUtils must be a function that returns a promise for the utilities module, not ${typeof a}`
        )
      );
    return C.startedLoadingUtilsScript = !0, t.then((s) => {
      const i = s == null ? void 0 : s.default;
      if (!i || typeof i != "object")
        throw new TypeError(
          "The loader function passed to attachUtils did not resolve to a module object with utils as its default export."
        );
      return C.utils = i, Dt.forEachInstance("handleUtils"), !0;
    }).catch((s) => {
      throw Dt.forEachInstance("handleUtilsFailure", s), s;
    });
  }
  return null;
}, C = Object.assign(
  (a, t) => {
    const s = new Dt(a, t);
    return C.instances[s.id] = s, a.iti = s, s;
  },
  {
    defaults: ae,
    //* Using a static var like this allows us to mock it in the tests.
    documentReady: () => document.readyState === "complete",
    //* Get the country data object.
    getCountryData: () => ht,
    //* A getter for the plugin instance.
    getInstance: (a) => {
      const t = a.dataset.intlTelInputId;
      return t ? C.instances[t] : null;
    },
    //* A map from instance ID to instance object.
    instances: {},
    attachUtils: $s,
    startedLoadingUtilsScript: !1,
    startedLoadingAutoCountry: !1,
    version: "26.8.1"
  }
);
export {
  C as i
};
