// intl-tel-input/angular/build/IntlTelInput.js
// angular/build/temp/intl-tel-input/data.js
var rawCountryData = [
  [
    "af",
    // Afghanistan
    "93",
    0,
    null,
    "0"
  ],
  [
    "ax",
    // Åland Islands
    "358",
    1,
    ["18", "4"],
    // (4 is a mobile range shared with FI)
    "0"
  ],
  [
    "al",
    // Albania
    "355",
    0,
    null,
    "0"
  ],
  [
    "dz",
    // Algeria
    "213",
    0,
    null,
    "0"
  ],
  [
    "as",
    // American Samoa
    "1",
    5,
    ["684"],
    "1"
  ],
  [
    "ad",
    // Andorra
    "376"
  ],
  [
    "ao",
    // Angola
    "244"
  ],
  [
    "ai",
    // Anguilla
    "1",
    6,
    ["264"],
    "1"
  ],
  [
    "ag",
    // Antigua and Barbuda
    "1",
    7,
    ["268"],
    "1"
  ],
  [
    "ar",
    // Argentina
    "54",
    0,
    null,
    "0"
  ],
  [
    "am",
    // Armenia
    "374",
    0,
    null,
    "0"
  ],
  [
    "aw",
    // Aruba
    "297"
  ],
  [
    "ac",
    // Ascension Island
    "247"
  ],
  [
    "au",
    // Australia
    "61",
    0,
    ["4"],
    // (mobile range shared with CX and CC)
    "0"
  ],
  [
    "at",
    // Austria
    "43",
    0,
    null,
    "0"
  ],
  [
    "az",
    // Azerbaijan
    "994",
    0,
    null,
    "0"
  ],
  [
    "bs",
    // Bahamas
    "1",
    8,
    ["242"],
    "1"
  ],
  [
    "bh",
    // Bahrain
    "973"
  ],
  [
    "bd",
    // Bangladesh
    "880",
    0,
    null,
    "0"
  ],
  [
    "bb",
    // Barbados
    "1",
    9,
    ["246"],
    "1"
  ],
  [
    "by",
    // Belarus
    "375",
    0,
    null,
    "8"
  ],
  [
    "be",
    // Belgium
    "32",
    0,
    null,
    "0"
  ],
  [
    "bz",
    // Belize
    "501"
  ],
  [
    "bj",
    // Benin
    "229"
  ],
  [
    "bm",
    // Bermuda
    "1",
    10,
    ["441"],
    "1"
  ],
  [
    "bt",
    // Bhutan
    "975"
  ],
  [
    "bo",
    // Bolivia
    "591",
    0,
    null,
    "0"
  ],
  [
    "ba",
    // Bosnia and Herzegovina
    "387",
    0,
    null,
    "0"
  ],
  [
    "bw",
    // Botswana
    "267"
  ],
  [
    "br",
    // Brazil
    "55",
    0,
    null,
    "0"
  ],
  [
    "io",
    // British Indian Ocean Territory
    "246"
  ],
  [
    "vg",
    // British Virgin Islands
    "1",
    11,
    ["284"],
    "1"
  ],
  [
    "bn",
    // Brunei
    "673"
  ],
  [
    "bg",
    // Bulgaria
    "359",
    0,
    null,
    "0"
  ],
  [
    "bf",
    // Burkina Faso
    "226"
  ],
  [
    "bi",
    // Burundi
    "257"
  ],
  [
    "kh",
    // Cambodia
    "855",
    0,
    null,
    "0"
  ],
  [
    "cm",
    // Cameroon
    "237"
  ],
  [
    "ca",
    // Canada
    "1",
    1,
    ["204", "226", "236", "249", "250", "257", "263", "289", "306", "343", "354", "365", "367", "368", "382", "403", "416", "418", "428", "431", "437", "438", "450", "468", "474", "506", "514", "519", "548", "579", "581", "584", "587", "604", "613", "639", "647", "672", "683", "705", "709", "742", "753", "778", "780", "782", "807", "819", "825", "867", "873", "879", "902", "905", "942"],
    "1"
  ],
  [
    "cv",
    // Cape Verde
    "238"
  ],
  [
    "bq",
    // Caribbean Netherlands
    "599",
    1,
    ["3", "4", "7"]
  ],
  [
    "ky",
    // Cayman Islands
    "1",
    12,
    ["345"],
    "1"
  ],
  [
    "cf",
    // Central African Republic
    "236"
  ],
  [
    "td",
    // Chad
    "235"
  ],
  [
    "cl",
    // Chile
    "56"
  ],
  [
    "cn",
    // China
    "86",
    0,
    null,
    "0"
  ],
  [
    "cx",
    // Christmas Island
    "61",
    2,
    ["4", "89164"],
    // (4 is a mobile range shared with AU and CC)
    "0"
  ],
  [
    "cc",
    // Cocos (Keeling) Islands
    "61",
    1,
    ["4", "89162"],
    // (4 is a mobile range shared with AU and CX)
    "0"
  ],
  [
    "co",
    // Colombia
    "57",
    0,
    null,
    "0"
  ],
  [
    "km",
    // Comoros
    "269"
  ],
  [
    "cg",
    // Congo (Brazzaville)
    "242"
  ],
  [
    "cd",
    // Congo (Kinshasa)
    "243",
    0,
    null,
    "0"
  ],
  [
    "ck",
    // Cook Islands
    "682"
  ],
  [
    "cr",
    // Costa Rica
    "506"
  ],
  [
    "ci",
    // Côte d'Ivoire
    "225"
  ],
  [
    "hr",
    // Croatia
    "385",
    0,
    null,
    "0"
  ],
  [
    "cu",
    // Cuba
    "53",
    0,
    null,
    "0"
  ],
  [
    "cw",
    // Curaçao
    "599",
    0
  ],
  [
    "cy",
    // Cyprus
    "357"
  ],
  [
    "cz",
    // Czech Republic
    "420"
  ],
  [
    "dk",
    // Denmark
    "45"
  ],
  [
    "dj",
    // Djibouti
    "253"
  ],
  [
    "dm",
    // Dominica
    "1",
    13,
    ["767"],
    "1"
  ],
  [
    "do",
    // Dominican Republic
    "1",
    2,
    ["809", "829", "849"],
    "1"
  ],
  [
    "ec",
    // Ecuador
    "593",
    0,
    null,
    "0"
  ],
  [
    "eg",
    // Egypt
    "20",
    0,
    null,
    "0"
  ],
  [
    "sv",
    // El Salvador
    "503"
  ],
  [
    "gq",
    // Equatorial Guinea
    "240"
  ],
  [
    "er",
    // Eritrea
    "291",
    0,
    null,
    "0"
  ],
  [
    "ee",
    // Estonia
    "372"
  ],
  [
    "sz",
    // Eswatini
    "268"
  ],
  [
    "et",
    // Ethiopia
    "251",
    0,
    null,
    "0"
  ],
  [
    "fk",
    // Falkland Islands (Malvinas)
    "500"
  ],
  [
    "fo",
    // Faroe Islands
    "298"
  ],
  [
    "fj",
    // Fiji
    "679"
  ],
  [
    "fi",
    // Finland
    "358",
    0,
    ["4"],
    // (mobile range shared with AX)
    "0"
  ],
  [
    "fr",
    // France
    "33",
    0,
    null,
    "0"
  ],
  [
    "gf",
    // French Guiana
    "594",
    0,
    null,
    "0"
  ],
  [
    "pf",
    // French Polynesia
    "689"
  ],
  [
    "ga",
    // Gabon
    "241"
  ],
  [
    "gm",
    // Gambia
    "220"
  ],
  [
    "ge",
    // Georgia
    "995",
    0,
    null,
    "0"
  ],
  [
    "de",
    // Germany
    "49",
    0,
    null,
    "0"
  ],
  [
    "gh",
    // Ghana
    "233",
    0,
    null,
    "0"
  ],
  [
    "gi",
    // Gibraltar
    "350"
  ],
  [
    "gr",
    // Greece
    "30"
  ],
  [
    "gl",
    // Greenland
    "299"
  ],
  [
    "gd",
    // Grenada
    "1",
    14,
    ["473"],
    "1"
  ],
  [
    "gp",
    // Guadeloupe
    "590",
    0,
    null,
    "0"
  ],
  [
    "gu",
    // Guam
    "1",
    15,
    ["671"],
    "1"
  ],
  [
    "gt",
    // Guatemala
    "502"
  ],
  [
    "gg",
    // Guernsey
    "44",
    1,
    ["1481", "7781", "7839", "7911"],
    "0"
  ],
  [
    "gn",
    // Guinea
    "224"
  ],
  [
    "gw",
    // Guinea-Bissau
    "245"
  ],
  [
    "gy",
    // Guyana
    "592"
  ],
  [
    "ht",
    // Haiti
    "509"
  ],
  [
    "hn",
    // Honduras
    "504"
  ],
  [
    "hk",
    // Hong Kong SAR China
    "852"
  ],
  [
    "hu",
    // Hungary
    "36",
    0,
    null,
    "06"
  ],
  [
    "is",
    // Iceland
    "354"
  ],
  [
    "in",
    // India
    "91",
    0,
    null,
    "0"
  ],
  [
    "id",
    // Indonesia
    "62",
    0,
    null,
    "0"
  ],
  [
    "ir",
    // Iran
    "98",
    0,
    null,
    "0"
  ],
  [
    "iq",
    // Iraq
    "964",
    0,
    null,
    "0"
  ],
  [
    "ie",
    // Ireland
    "353",
    0,
    null,
    "0"
  ],
  [
    "im",
    // Isle of Man
    "44",
    2,
    ["1624", "74576", "7524", "7624", "7924"],
    "0"
  ],
  [
    "il",
    // Israel
    "972",
    0,
    null,
    "0"
  ],
  [
    "it",
    // Italy
    "39",
    0,
    ["3"]
    // (mobile range shared with VA)
  ],
  [
    "jm",
    // Jamaica
    "1",
    4,
    ["658", "876"],
    "1"
  ],
  [
    "jp",
    // Japan
    "81",
    0,
    null,
    "0"
  ],
  [
    "je",
    // Jersey
    "44",
    3,
    ["1534", "7509", "7700", "7797", "7829", "7937"],
    "0"
  ],
  [
    "jo",
    // Jordan
    "962",
    0,
    null,
    "0"
  ],
  [
    "kz",
    // Kazakhstan
    "7",
    1,
    ["33", "7"],
    // (33 is shared with RU)
    "8"
  ],
  [
    "ke",
    // Kenya
    "254",
    0,
    null,
    "0"
  ],
  [
    "ki",
    // Kiribati
    "686",
    0,
    null,
    "0"
  ],
  [
    "xk",
    // Kosovo
    "383",
    0,
    null,
    "0"
  ],
  [
    "kw",
    // Kuwait
    "965"
  ],
  [
    "kg",
    // Kyrgyzstan
    "996",
    0,
    null,
    "0"
  ],
  [
    "la",
    // Laos
    "856",
    0,
    null,
    "0"
  ],
  [
    "lv",
    // Latvia
    "371"
  ],
  [
    "lb",
    // Lebanon
    "961",
    0,
    null,
    "0"
  ],
  [
    "ls",
    // Lesotho
    "266"
  ],
  [
    "lr",
    // Liberia
    "231",
    0,
    null,
    "0"
  ],
  [
    "ly",
    // Libya
    "218",
    0,
    null,
    "0"
  ],
  [
    "li",
    // Liechtenstein
    "423",
    0,
    null,
    "0"
  ],
  [
    "lt",
    // Lithuania
    "370",
    0,
    null,
    "0"
  ],
  [
    "lu",
    // Luxembourg
    "352"
  ],
  [
    "mo",
    // Macao SAR China
    "853"
  ],
  [
    "mg",
    // Madagascar
    "261",
    0,
    null,
    "0"
  ],
  [
    "mw",
    // Malawi
    "265",
    0,
    null,
    "0"
  ],
  [
    "my",
    // Malaysia
    "60",
    0,
    null,
    "0"
  ],
  [
    "mv",
    // Maldives
    "960"
  ],
  [
    "ml",
    // Mali
    "223"
  ],
  [
    "mt",
    // Malta
    "356"
  ],
  [
    "mh",
    // Marshall Islands
    "692",
    0,
    null,
    "1"
  ],
  [
    "mq",
    // Martinique
    "596",
    0,
    null,
    "0"
  ],
  [
    "mr",
    // Mauritania
    "222"
  ],
  [
    "mu",
    // Mauritius
    "230"
  ],
  [
    "yt",
    // Mayotte
    "262",
    1,
    ["269", "639"],
    "0"
  ],
  [
    "mx",
    // Mexico
    "52"
  ],
  [
    "fm",
    // Micronesia
    "691"
  ],
  [
    "md",
    // Moldova
    "373",
    0,
    null,
    "0"
  ],
  [
    "mc",
    // Monaco
    "377",
    0,
    null,
    "0"
  ],
  [
    "mn",
    // Mongolia
    "976",
    0,
    null,
    "0"
  ],
  [
    "me",
    // Montenegro
    "382",
    0,
    null,
    "0"
  ],
  [
    "ms",
    // Montserrat
    "1",
    16,
    ["664"],
    "1"
  ],
  [
    "ma",
    // Morocco
    "212",
    0,
    ["6", "7"],
    // (mobile ranges shared with EH)
    "0"
  ],
  [
    "mz",
    // Mozambique
    "258"
  ],
  [
    "mm",
    // Myanmar (Burma)
    "95",
    0,
    null,
    "0"
  ],
  [
    "na",
    // Namibia
    "264",
    0,
    null,
    "0"
  ],
  [
    "nr",
    // Nauru
    "674"
  ],
  [
    "np",
    // Nepal
    "977",
    0,
    null,
    "0"
  ],
  [
    "nl",
    // Netherlands
    "31",
    0,
    null,
    "0"
  ],
  [
    "nc",
    // New Caledonia
    "687"
  ],
  [
    "nz",
    // New Zealand
    "64",
    0,
    null,
    "0"
  ],
  [
    "ni",
    // Nicaragua
    "505"
  ],
  [
    "ne",
    // Niger
    "227"
  ],
  [
    "ng",
    // Nigeria
    "234",
    0,
    null,
    "0"
  ],
  [
    "nu",
    // Niue
    "683"
  ],
  [
    "nf",
    // Norfolk Island
    "672"
  ],
  [
    "kp",
    // North Korea
    "850",
    0,
    null,
    "0"
  ],
  [
    "mk",
    // North Macedonia
    "389",
    0,
    null,
    "0"
  ],
  [
    "mp",
    // Northern Mariana Islands
    "1",
    17,
    ["670"],
    "1"
  ],
  [
    "no",
    // Norway
    "47",
    0,
    ["4", "9"]
    // (mobile ranges shared with SJ)
  ],
  [
    "om",
    // Oman
    "968"
  ],
  [
    "pk",
    // Pakistan
    "92",
    0,
    null,
    "0"
  ],
  [
    "pw",
    // Palau
    "680"
  ],
  [
    "ps",
    // Palestinian Territories
    "970",
    0,
    null,
    "0"
  ],
  [
    "pa",
    // Panama
    "507"
  ],
  [
    "pg",
    // Papua New Guinea
    "675"
  ],
  [
    "py",
    // Paraguay
    "595",
    0,
    null,
    "0"
  ],
  [
    "pe",
    // Peru
    "51",
    0,
    null,
    "0"
  ],
  [
    "ph",
    // Philippines
    "63",
    0,
    null,
    "0"
  ],
  [
    "pl",
    // Poland
    "48"
  ],
  [
    "pt",
    // Portugal
    "351"
  ],
  [
    "pr",
    // Puerto Rico
    "1",
    3,
    ["787", "939"],
    "1"
  ],
  [
    "qa",
    // Qatar
    "974"
  ],
  [
    "re",
    // Réunion
    "262",
    0,
    null,
    "0"
  ],
  [
    "ro",
    // Romania
    "40",
    0,
    null,
    "0"
  ],
  [
    "ru",
    // Russia
    "7",
    0,
    ["33"],
    // (shared with KZ)
    "8"
  ],
  [
    "rw",
    // Rwanda
    "250",
    0,
    null,
    "0"
  ],
  [
    "ws",
    // Samoa
    "685"
  ],
  [
    "sm",
    // San Marino
    "378"
  ],
  [
    "st",
    // São Tomé & Príncipe
    "239"
  ],
  [
    "sa",
    // Saudi Arabia
    "966",
    0,
    null,
    "0"
  ],
  [
    "sn",
    // Senegal
    "221"
  ],
  [
    "rs",
    // Serbia
    "381",
    0,
    null,
    "0"
  ],
  [
    "sc",
    // Seychelles
    "248"
  ],
  [
    "sl",
    // Sierra Leone
    "232",
    0,
    null,
    "0"
  ],
  [
    "sg",
    // Singapore
    "65"
  ],
  [
    "sx",
    // Sint Maarten
    "1",
    21,
    ["721"],
    "1"
  ],
  [
    "sk",
    // Slovakia
    "421",
    0,
    null,
    "0"
  ],
  [
    "si",
    // Slovenia
    "386",
    0,
    null,
    "0"
  ],
  [
    "sb",
    // Solomon Islands
    "677"
  ],
  [
    "so",
    // Somalia
    "252",
    0,
    null,
    "0"
  ],
  [
    "za",
    // South Africa
    "27",
    0,
    null,
    "0"
  ],
  [
    "kr",
    // South Korea
    "82",
    0,
    null,
    "0"
  ],
  [
    "ss",
    // South Sudan
    "211",
    0,
    null,
    "0"
  ],
  [
    "es",
    // Spain
    "34"
  ],
  [
    "lk",
    // Sri Lanka
    "94",
    0,
    null,
    "0"
  ],
  [
    "bl",
    // St. Barthélemy
    "590",
    1,
    null,
    "0"
  ],
  [
    "sh",
    // St. Helena
    "290"
  ],
  [
    "kn",
    // St. Kitts & Nevis
    "1",
    18,
    ["869"],
    "1"
  ],
  [
    "lc",
    // St. Lucia
    "1",
    19,
    ["758"],
    "1"
  ],
  [
    "mf",
    // St. Martin
    "590",
    2,
    null,
    "0"
  ],
  [
    "pm",
    // St. Pierre & Miquelon
    "508",
    0,
    null,
    "0"
  ],
  [
    "vc",
    // St. Vincent & Grenadines
    "1",
    20,
    ["784"],
    "1"
  ],
  [
    "sd",
    // Sudan
    "249",
    0,
    null,
    "0"
  ],
  [
    "sr",
    // Suriname
    "597"
  ],
  [
    "sj",
    // Svalbard & Jan Mayen
    "47",
    1,
    ["4", "79", "9"]
    // (4 and 9 are mobile ranges shared with NO)
  ],
  [
    "se",
    // Sweden
    "46",
    0,
    null,
    "0"
  ],
  [
    "ch",
    // Switzerland
    "41",
    0,
    null,
    "0"
  ],
  [
    "sy",
    // Syria
    "963",
    0,
    null,
    "0"
  ],
  [
    "tw",
    // Taiwan
    "886",
    0,
    null,
    "0"
  ],
  [
    "tj",
    // Tajikistan
    "992"
  ],
  [
    "tz",
    // Tanzania
    "255",
    0,
    null,
    "0"
  ],
  [
    "th",
    // Thailand
    "66",
    0,
    null,
    "0"
  ],
  [
    "tl",
    // Timor-Leste
    "670"
  ],
  [
    "tg",
    // Togo
    "228"
  ],
  [
    "tk",
    // Tokelau
    "690"
  ],
  [
    "to",
    // Tonga
    "676"
  ],
  [
    "tt",
    // Trinidad & Tobago
    "1",
    22,
    ["868"],
    "1"
  ],
  [
    "tn",
    // Tunisia
    "216"
  ],
  [
    "tr",
    // Turkey
    "90",
    0,
    null,
    "0"
  ],
  [
    "tm",
    // Turkmenistan
    "993",
    0,
    null,
    "8"
  ],
  [
    "tc",
    // Turks & Caicos Islands
    "1",
    23,
    ["649"],
    "1"
  ],
  [
    "tv",
    // Tuvalu
    "688"
  ],
  [
    "vi",
    // U.S. Virgin Islands
    "1",
    24,
    ["340"],
    "1"
  ],
  [
    "ug",
    // Uganda
    "256",
    0,
    null,
    "0"
  ],
  [
    "ua",
    // Ukraine
    "380",
    0,
    null,
    "0"
  ],
  [
    "ae",
    // United Arab Emirates
    "971",
    0,
    null,
    "0"
  ],
  [
    "gb",
    // United Kingdom
    "44",
    0,
    null,
    "0"
  ],
  [
    "us",
    // United States
    "1",
    0,
    null,
    "1"
  ],
  [
    "uy",
    // Uruguay
    "598",
    0,
    null,
    "0"
  ],
  [
    "uz",
    // Uzbekistan
    "998"
  ],
  [
    "vu",
    // Vanuatu
    "678"
  ],
  [
    "va",
    // Vatican City
    "39",
    1,
    ["06698", "3"]
    // (3 is a mobile range shared with IT)
  ],
  [
    "ve",
    // Venezuela
    "58",
    0,
    null,
    "0"
  ],
  [
    "vn",
    // Vietnam
    "84",
    0,
    null,
    "0"
  ],
  [
    "wf",
    // Wallis & Futuna
    "681"
  ],
  [
    "eh",
    // Western Sahara
    "212",
    1,
    ["5288", "5289", "6", "7"],
    // (6 and 7 are mobile ranges shared with MA)
    "0"
  ],
  [
    "ye",
    // Yemen
    "967",
    0,
    null,
    "0"
  ],
  [
    "zm",
    // Zambia
    "260",
    0,
    null,
    "0"
  ],
  [
    "zw",
    // Zimbabwe
    "263",
    0,
    null,
    "0"
  ]
];
var allCountries = [];
for (const c of rawCountryData) {
  allCountries.push({
    name: "",
    // populated in the plugin
    iso2: c[0],
    dialCode: c[1],
    priority: c[2] || 0,
    areaCodes: c[3] || null,
    nodeById: {},
    // populated by the plugin
    nationalPrefix: c[4] || null,
    normalisedName: "",
    // populated in the plugin
    initials: "",
    // populated in the plugin
    dialCodePlus: ""
    // populated in the plugin
  });
}
var data_default = allCountries;

// angular/build/temp/modules/constants.js
var EVENTS = {
  OPEN_COUNTRY_DROPDOWN: "open:countrydropdown",
  CLOSE_COUNTRY_DROPDOWN: "close:countrydropdown",
  COUNTRY_CHANGE: "countrychange",
  INPUT: "input"
  // used for synthetic input trigger
};
var CLASSES = {
  HIDE: "iti__hide",
  V_HIDE: "iti__v-hide",
  ARROW_UP: "iti__arrow--up",
  GLOBE: "iti__globe",
  FLAG: "iti__flag",
  LOADING: "iti__loading",
  COUNTRY_ITEM: "iti__country",
  HIGHLIGHT: "iti__highlight"
};
var KEYS = {
  ARROW_UP: "ArrowUp",
  ARROW_DOWN: "ArrowDown",
  SPACE: " ",
  ENTER: "Enter",
  ESC: "Escape",
  TAB: "Tab"
};
var INPUT_TYPES = {
  PASTE: "insertFromPaste",
  DELETE_FWD: "deleteContentForward"
};
var REGEX = {
  ALPHA_UNICODE: /\p{L}/u,
  // any kind of letter from any language
  NON_PLUS_NUMERIC: /[^+0-9]/,
  // chars that are NOT + or digit
  NON_PLUS_NUMERIC_GLOBAL: /[^+0-9]/g,
  // chars that are NOT + or digit (global)
  HIDDEN_SEARCH_CHAR: /^[a-zA-ZÀ-ÿа-яА-Я ]$/
  // single acceptable hidden-search char
};
var TIMINGS = {
  SEARCH_DEBOUNCE_MS: 100,
  HIDDEN_SEARCH_RESET_MS: 1e3,
  NEXT_TICK: 0
};
var SENTINELS = {
  UNKNOWN_NUMBER_TYPE: -99,
  UNKNOWN_VALIDATION_ERROR: -99
};
var LAYOUT = {
  NARROW_VIEWPORT_WIDTH: 500,
  // keep in sync with .iti__country-list CSS media query
  SANE_SELECTED_WITH_DIAL_WIDTH: 78,
  // px width fallback when separateDialCode enabled
  SANE_SELECTED_NO_DIAL_WIDTH: 42,
  // px width fallback when no separate dial code
  INPUT_PADDING_EXTRA_LEFT: 6,
  // px gap between selected country container and input text
  DROPDOWN_MARGIN: 3,
  // px margin between dropdown and tel input
  SANE_DROPDOWN_HEIGHT: 200
  // px height fallback for dropdown
};
var DIAL = {
  PLUS: "+",
  NANP: "1"
  // North American Numbering Plan
};
var UK = {
  ISO2: "gb",
  DIAL_CODE: "44",
  // +44 United Kingdom
  MOBILE_PREFIX: "7",
  // UK mobile numbers start with 7 after national trunk (0) or core section
  MOBILE_CORE_LENGTH: 10
  // core number length (excluding dial code / national prefix) for mobiles
};
var US = {
  ISO2: "us",
  DIAL_CODE: "1"
  // +1 United States
};
var PLACEHOLDER_MODES = {
  AGGRESSIVE: "aggressive",
  POLITE: "polite",
  OFF: "off"
};
var INITIAL_COUNTRY = {
  AUTO: "auto"
};
var NUMBER_TYPES = [
  "FIXED_LINE",
  "MOBILE",
  "FIXED_LINE_OR_MOBILE",
  "TOLL_FREE",
  "PREMIUM_RATE",
  "SHARED_COST",
  "VOIP",
  "PERSONAL_NUMBER",
  "PAGER",
  "UAN",
  "VOICEMAIL",
  "UNKNOWN"
];
var NUMBER_TYPE_SET = new Set(NUMBER_TYPES);
var DATA_KEYS = {
  COUNTRY_CODE: "countryCode",
  DIAL_CODE: "dialCode"
};
var ARIA = {
  EXPANDED: "aria-expanded",
  LABEL: "aria-label",
  SELECTED: "aria-selected",
  ACTIVE_DESCENDANT: "aria-activedescendant",
  HASPOPUP: "aria-haspopup",
  CONTROLS: "aria-controls",
  HIDDEN: "aria-hidden",
  AUTOCOMPLETE: "aria-autocomplete",
  MODAL: "aria-modal"
};

// angular/build/temp/intl-tel-input/i18n/en/index.js
var interfaceTranslations = {
  selectedCountryAriaLabel: "Change country, selected ${countryName} (${dialCode})",
  noCountrySelected: "Select country",
  countryListAriaLabel: "List of countries",
  searchPlaceholder: "Search",
  clearSearchAriaLabel: "Clear search",
  searchEmptyState: "No results found",
  searchSummaryAria(count) {
    if (count === 0) {
      return "No results found";
    }
    if (count === 1) {
      return "1 result found";
    }
    return `${count} results found`;
  }
};
var en_default = interfaceTranslations;

// angular/build/temp/modules/core/options.js
var mq = (q) => typeof window !== "undefined" && typeof window.matchMedia === "function" && window.matchMedia(q).matches;
var isNarrowViewport = () => mq(`(max-width: ${LAYOUT.NARROW_VIEWPORT_WIDTH}px)`);
var computeDefaultUseFullscreenPopup = () => {
  if (typeof navigator !== "undefined" && typeof window !== "undefined") {
    const isShortViewport = mq("(max-height: 600px)");
    const isCoarsePointer = mq("(pointer: coarse)");
    return isNarrowViewport() || isCoarsePointer && isShortViewport;
  }
  return false;
};
var defaults = {
  //* Whether or not to allow the dropdown.
  allowDropdown: true,
  //* The number type to enforce during validation.
  allowedNumberTypes: ["MOBILE", "FIXED_LINE"],
  //* Whether or not to allow extensions after the main number.
  allowNumberExtensions: false,
  // Allow alphanumeric "phonewords" (e.g. +1 800 FLOWERS) as valid numbers
  allowPhonewords: false,
  //* Add a placeholder in the input with an example number for the selected country.
  autoPlaceholder: PLACEHOLDER_MODES.POLITE,
  //* Add a custom class to the (injected) container element.
  containerClass: "",
  //* Locale for localising country names via Intl.DisplayNames.
  countryNameLocale: "en",
  //* The order of the countries in the dropdown. Defaults to alphabetical.
  countryOrder: null,
  //* Add a country search input at the top of the dropdown.
  countrySearch: true,
  //* Modify the auto placeholder.
  customPlaceholder: null,
  //* Always show the dropdown
  dropdownAlwaysOpen: false,
  //* Append menu to specified element.
  dropdownContainer: null,
  //* Don't display these countries.
  excludeCountries: [],
  //* Fix the dropdown width to the input width (rather than being as wide as the longest country name).
  fixDropdownWidth: true,
  //* Format the number as the user types
  formatAsYouType: true,
  //* Format the input value during initialisation and on setNumber.
  formatOnDisplay: true,
  //* geoIp lookup function.
  geoIpLookup: null,
  //* Inject a hidden input with the name returned from this function, and on submit, populate it with the result of getNumber.
  hiddenInput: null,
  //* Internationalise the plugin text e.g. search input placeholder, country names.
  i18n: {},
  //* Initial country.
  initialCountry: "",
  //* A function to load the utils script.
  loadUtils: null,
  //* National vs international formatting for numbers e.g. placeholders and displaying existing numbers.
  nationalMode: true,
  //* Display only these countries.
  onlyCountries: [],
  //* Number type to use for placeholders.
  placeholderNumberType: "MOBILE",
  //* Add custom classes to the search input element.
  searchInputClass: "",
  //* Display the international dial code next to the selected flag.
  separateDialCode: false,
  //* Show flags - for both the selected country, and in the country dropdown
  showFlags: true,
  //* Only allow certain chars e.g. a plus followed by numeric digits, and cap at max valid length.
  strictMode: false,
  //* Use full screen popup instead of dropdown for country list.
  useFullscreenPopup: computeDefaultUseFullscreenPopup()
};
var toString = (val) => JSON.stringify(val);
var isPlainObject = (val) => Boolean(val) && typeof val === "object" && !Array.isArray(val);
var isFn = (val) => typeof val === "function";
var isElLike = (val) => {
  if (!val || typeof val !== "object")
    return false;
  const v = val;
  return v.nodeType === 1 && typeof v.tagName === "string" && typeof v.appendChild === "function";
};
var iso2Set = new Set(data_default.map((c) => c.iso2));
var isIso2 = (val) => iso2Set.has(val);
var placeholderModeSet = new Set(Object.values(PLACEHOLDER_MODES));
var warn = (message) => {
  console.warn(`[intl-tel-input] ${message}`);
};
var warnOption = (optionName, expectedType, actualValue) => {
  warn(`Option '${optionName}' must be ${expectedType}; got ${toString(actualValue)}. Ignoring.`);
};
var hasOwn = (obj, key) => Object.prototype.hasOwnProperty.call(obj, key);
var validateIso2Array = (key, value) => {
  const expectedType = "an array of ISO2 country code strings";
  if (!Array.isArray(value)) {
    warnOption(key, expectedType, value);
    return false;
  }
  for (const v of value) {
    if (typeof v !== "string") {
      warnOption(key, expectedType, value);
      return false;
    }
    const lower = v.toLowerCase();
    if (!isIso2(lower)) {
      warn(`Invalid country code in '${key}': '${v}'. Ignoring.`);
      return false;
    }
  }
  return true;
};
var validateOptions = (customOptions) => {
  if (customOptions === void 0) {
    return {};
  }
  if (!isPlainObject(customOptions)) {
    const error = `The second argument must be an options object; got ${toString(customOptions)}. Using defaults.`;
    warn(error);
    return {};
  }
  const validatedOptions = {};
  for (const [key, value] of Object.entries(customOptions)) {
    if (!hasOwn(defaults, key)) {
      warn(`Unknown option '${key}'. Ignoring.`);
      continue;
    }
    switch (key) {
      case "allowDropdown":
      case "allowNumberExtensions":
      case "allowPhonewords":
      case "countrySearch":
      case "dropdownAlwaysOpen":
      case "fixDropdownWidth":
      case "formatAsYouType":
      case "formatOnDisplay":
      case "nationalMode":
      case "showFlags":
      case "separateDialCode":
      case "strictMode":
      case "useFullscreenPopup":
        if (typeof value !== "boolean") {
          warnOption(key, "a boolean", value);
          break;
        }
        validatedOptions[key] = value;
        break;
      case "autoPlaceholder":
        if (typeof value !== "string" || !placeholderModeSet.has(value)) {
          const validModes = Array.from(placeholderModeSet).join(", ");
          warnOption("autoPlaceholder", `one of ${validModes}`, value);
          break;
        }
        validatedOptions[key] = value;
        break;
      case "containerClass":
      case "searchInputClass":
      case "countryNameLocale":
        if (typeof value !== "string") {
          warnOption(key, "a string", value);
          break;
        }
        validatedOptions[key] = value;
        break;
      case "countryOrder":
        if (value === null || validateIso2Array(key, value)) {
          validatedOptions[key] = value;
        }
        break;
      case "customPlaceholder":
      case "geoIpLookup":
      case "hiddenInput":
      case "loadUtils":
        if (value !== null && !isFn(value)) {
          warnOption(key, "a function or null", value);
          break;
        }
        validatedOptions[key] = value;
        break;
      case "dropdownContainer":
        if (value !== null && !isElLike(value)) {
          warnOption("dropdownContainer", "an HTMLElement or null", value);
          break;
        }
        validatedOptions[key] = value;
        break;
      case "excludeCountries":
      case "onlyCountries":
        if (validateIso2Array(key, value)) {
          validatedOptions[key] = value;
        }
        break;
      case "i18n":
        if (value && !isPlainObject(value)) {
          warnOption("i18n", "an object", value);
          break;
        }
        validatedOptions[key] = value;
        break;
      case "initialCountry": {
        if (typeof value !== "string") {
          warnOption("initialCountry", "a string", value);
          break;
        }
        const lower = value.toLowerCase();
        if (lower && (lower !== INITIAL_COUNTRY.AUTO && !isIso2(lower))) {
          warnOption("initialCountry", "a valid ISO2 country code or 'auto'", value);
          break;
        }
        validatedOptions[key] = value;
        break;
      }
      case "placeholderNumberType":
        if (typeof value !== "string" || !NUMBER_TYPE_SET.has(value)) {
          const validTypes = Array.from(NUMBER_TYPE_SET).join(", ");
          warnOption("placeholderNumberType", `one of ${validTypes}`, value);
          break;
        }
        validatedOptions[key] = value;
        break;
      case "allowedNumberTypes":
        if (value !== null) {
          if (!Array.isArray(value)) {
            warnOption("allowedNumberTypes", "an array of number types or null", value);
            break;
          }
          let allValid = true;
          for (const v of value) {
            if (typeof v !== "string" || !NUMBER_TYPE_SET.has(v)) {
              const validTypes = Array.from(NUMBER_TYPE_SET).join(", ");
              warnOption("allowedNumberTypes", `an array of valid number types (${validTypes})`, v);
              allValid = false;
              break;
            }
          }
          if (allValid) {
            validatedOptions[key] = value;
          }
        } else {
          validatedOptions[key] = null;
        }
        break;
    }
  }
  return validatedOptions;
};
var applyOptionSideEffects = (o) => {
  if (o.dropdownAlwaysOpen) {
    o.useFullscreenPopup = false;
    o.allowDropdown = true;
  }
  if (o.useFullscreenPopup) {
    o.fixDropdownWidth = false;
  } else {
    if (isNarrowViewport()) {
      o.fixDropdownWidth = true;
    }
  }
  if (o.onlyCountries.length === 1) {
    o.initialCountry = o.onlyCountries[0];
  }
  if (o.separateDialCode) {
    o.nationalMode = false;
  }
  if (o.allowDropdown && !o.showFlags && !o.separateDialCode) {
    o.nationalMode = false;
  }
  if (o.useFullscreenPopup && !o.dropdownContainer) {
    o.dropdownContainer = document.body;
  }
  o.i18n = Object.assign(Object.assign({}, en_default), o.i18n);
};

// angular/build/temp/modules/utils/string.js
var getNumeric = (s) => s.replace(/\D/g, "");
var normaliseString = (s = "") => s.normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase();

// angular/build/temp/modules/utils/isAndroid.js
var getIsAndroid = () => typeof navigator !== "undefined" ? /Android/i.test(navigator.userAgent) : false;

// angular/build/temp/modules/core/countrySearch.js
var getMatchedCountries = (countries, query) => {
  const normalisedQuery = normaliseString(query);
  const iso2Matches = [];
  const nameStartWith = [];
  const nameContains = [];
  const dialCodeMatches = [];
  const dialCodeContains = [];
  const initialsMatches = [];
  for (const c of countries) {
    if (c.iso2 === normalisedQuery) {
      iso2Matches.push(c);
    } else if (c.normalisedName.startsWith(normalisedQuery)) {
      nameStartWith.push(c);
    } else if (c.normalisedName.includes(normalisedQuery)) {
      nameContains.push(c);
    } else if (normalisedQuery === c.dialCode || normalisedQuery === c.dialCodePlus) {
      dialCodeMatches.push(c);
    } else if (c.dialCodePlus.includes(normalisedQuery)) {
      dialCodeContains.push(c);
    } else if (c.initials.includes(normalisedQuery)) {
      initialsMatches.push(c);
    }
  }
  const sortByPriority = (a, b) => a.priority - b.priority;
  return [
    ...iso2Matches.sort(sortByPriority),
    ...nameStartWith.sort(sortByPriority),
    ...nameContains.sort(sortByPriority),
    ...dialCodeMatches.sort(sortByPriority),
    ...dialCodeContains.sort(sortByPriority),
    ...initialsMatches.sort(sortByPriority)
  ];
};
var findFirstCountryStartingWith = (countries, query) => {
  const lowerQuery = query.toLowerCase();
  for (const c of countries) {
    const lowerName = c.name.toLowerCase();
    if (lowerName.startsWith(lowerQuery)) {
      return c;
    }
  }
  return null;
};

// angular/build/temp/modules/utils/dom.js
var buildClassNames = (flags) => Object.keys(flags).filter((k) => Boolean(flags[k])).join(" ");
var createEl = (tagName, attrs, container) => {
  const el = document.createElement(tagName);
  if (attrs) {
    Object.entries(attrs).forEach(([key, value]) => el.setAttribute(key, value));
  }
  if (container) {
    container.appendChild(el);
  }
  return el;
};

// angular/build/temp/modules/core/icons.js
var buildSearchIcon = () => `
  <svg class="iti__search-icon-svg" width="14" height="14" viewBox="0 0 24 24" focusable="false" ${ARIA.HIDDEN}="true">
    <circle cx="11" cy="11" r="7" />
    <line x1="21" y1="21" x2="16.65" y2="16.65" />
  </svg>`;
var buildClearIcon = (id2) => {
  const maskId = `iti-${id2}-clear-mask`;
  return `
    <svg class="iti__search-clear-svg" width="12" height="12" viewBox="0 0 16 16" ${ARIA.HIDDEN}="true" focusable="false">
      <mask id="${maskId}" maskUnits="userSpaceOnUse">
        <rect width="16" height="16" fill="white" />
        <path d="M5.2 5.2 L10.8 10.8 M10.8 5.2 L5.2 10.8" stroke="black" stroke-linecap="round" class="iti__search-clear-x" />
      </mask>
      <circle cx="8" cy="8" r="8" class="iti__search-clear-bg" mask="url(#${maskId})" />
    </svg>`;
};
var buildCheckIcon = () => `
  <svg class="iti__country-check-svg" width="14" height="14" viewBox="0 0 16 16" fill="currentColor" focusable="false" ${ARIA.HIDDEN}="true">
    <path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0m-3.97-3.03a.75.75 0 0 0-1.08.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-.01-1.05z"/>
  </svg>`;
var buildGlobeIcon = () => `
  <svg width="256" height="256" viewBox="0 0 512 512" class="iti__globe-svg">
    <path d="M508 213a240 240 0 0 0-449-87l-2 5-2 5c-8 14-13 30-17 46a65 65 0 0 1 56 4c16-10 35-19 56-27l9-3c-6 23-10 48-10 74h-16l4 6c3 4 5 8 6 13h6c0 22 3 44 8 65l2 10-25-10-4 5 12 18 9 3 6 2 8 3 9 26 1 2 16-7h1l-5-13-1-2c24 6 49 9 75 10v26l11 10 7 7v-30l1-13c22 0 44-3 65-8l10-2-21 48-1 1a317 317 0 0 1-14 23l-21 5h-2c6 16 7 33 1 50a240 240 0 0 0 211-265m-401-56-11 6c19-44 54-79 98-98-11 20-21 44-29 69-21 6-40 15-58 23m154 182v4c-29-1-57-6-81-13-7-25-12-52-13-81h94zm0-109h-94c1-29 6-56 13-81 24-7 52-12 81-13zm0-112c-22 1-44 4-65 8l-10 2 12-30 9-17 1-2a332 332 0 0 1 13-23c13-4 26-6 40-7zm187 69 6 4c4 12 6 25 6 38v1h-68c-1-26-4-51-10-74l48 20 1 1 14 8zm-14-44 10 20c-20-11-43-21-68-29-8-25-18-49-29-69 37 16 67 44 87 78M279 49h1c13 1 27 3 39 7l14 23 1 2a343 343 0 0 1 12 26l2 5 6 16c-23-6-48-9-74-10h-1zm0 87h1c29 1 56 6 81 13 7 24 12 51 12 80v1h-94zm2 207h-2v-94h95c-1 29-6 56-13 81-24 7-51 12-80 13m86 60-20 10c11-20 21-43 29-68 25-8 48-18 68-29-16 37-43 67-77 87m87-115-7 5-16 9-2 1a337 337 0 0 1-47 21c6-24 9-49 10-75h68c0 13-2 27-6 39"/>
    <path d="m261 428-2-2-22-21a40 40 0 0 0-32-11h-1a37 37 0 0 0-18 8l-1 1-4 2-2 2-5 4c-9-3-36-31-47-44s-32-45-34-55l3-2a151 151 0 0 0 11-9v-1a39 39 0 0 0 5-48l-3-3-11-19-3-4-5-7h-1l-3-3-4-3-5-2a35 35 0 0 0-16-3h-5c-4 1-14 5-24 11l-4 2-4 3-4 2c-9 8-17 17-18 27a380 380 0 0 0 212 259h3c12 0 25-10 36-21l10-12 6-11a39 39 0 0 0-8-40"/>
  </svg>`;

// angular/build/temp/modules/core/ui.js
var __classPrivateFieldSet = function(receiver, state, value, kind, f) {
  if (kind === "m") throw new TypeError("Private method is not writable");
  if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
  if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
  return kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value), value;
};
var __classPrivateFieldGet = function(receiver, state, kind, f) {
  if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
  if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
  return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _UI_instances;
var _a;
var _UI_options;
var _UI_id;
var _UI_isRTL;
var _UI_originalPaddingLeft;
var _UI_countries;
var _UI_searchKeyupTimer;
var _UI_inlineDropdownHeight;
var _UI_selectedDialCode;
var _UI_dropdownArrow;
var _UI_dropdownContent;
var _UI_searchIcon;
var _UI_searchNoResults;
var _UI_searchResultsA11yText;
var _UI_dropdownForContainer;
var _UI_selectedItem;
var _UI_createWrapperAndInsert;
var _UI_maybeBuildCountryContainer;
var _UI_maybeEnsureDropdownWidthSet;
var _UI_buildDropdownContent;
var _UI_buildSearchUI;
var _UI_maybeUpdateInputPaddingAndReveal;
var _UI_maybeBuildHiddenInputs;
var _UI_appendListItems;
var _UI_updateInputPadding;
var _UI_getBody;
var _UI_getHiddenSelectedCountryWidth;
var _UI_getHiddenInlineDropdownHeight;
var _UI_updateSearchResultsA11yText;
var _UI_doFilter;
var _UI_updateSelectedItem;
var _UI_filterCountries;
var _UI_shouldPositionInlineDropdownBelowInput;
var _UI_handleDropdownContainer;
var UI = class {
  constructor(input, options, id2) {
    _UI_instances.add(this);
    _UI_options.set(this, void 0);
    _UI_id.set(this, void 0);
    _UI_isRTL.set(this, void 0);
    _UI_originalPaddingLeft.set(this, void 0);
    _UI_countries.set(this, void 0);
    _UI_searchKeyupTimer.set(this, null);
    _UI_inlineDropdownHeight.set(this, null);
    _UI_selectedDialCode.set(this, void 0);
    _UI_dropdownArrow.set(this, void 0);
    _UI_dropdownContent.set(this, void 0);
    _UI_searchIcon.set(this, void 0);
    _UI_searchNoResults.set(this, void 0);
    _UI_searchResultsA11yText.set(this, void 0);
    _UI_dropdownForContainer.set(this, null);
    _UI_selectedItem.set(this, null);
    this.highlightedItem = null;
    input.dataset.intlTelInputId = id2.toString();
    this.telInput = input;
    __classPrivateFieldSet(this, _UI_options, options, "f");
    __classPrivateFieldSet(this, _UI_id, id2, "f");
    this.hadInitialPlaceholder = Boolean(input.getAttribute("placeholder"));
    __classPrivateFieldSet(this, _UI_isRTL, !!this.telInput.closest("[dir=rtl]"), "f");
    if (__classPrivateFieldGet(this, _UI_options, "f").separateDialCode) {
      __classPrivateFieldSet(this, _UI_originalPaddingLeft, this.telInput.style.paddingLeft, "f");
    }
  }
  // Validate that the provided element is an HTMLInputElement.
  static validateInput(input) {
    const tagName = input === null || input === void 0 ? void 0 : input.tagName;
    const isInputEl = Boolean(input) && typeof input === "object" && tagName === "INPUT" && typeof input.setAttribute === "function";
    if (!isInputEl) {
      const type = Object.prototype.toString.call(input);
      throw new TypeError(`The first argument must be an HTMLInputElement, not ${type}`);
    }
  }
  //* Generate all of the markup for the plugin: the selected country overlay, and the dropdown.
  generateMarkup(countries) {
    __classPrivateFieldSet(this, _UI_countries, countries, "f");
    this.telInput.classList.add("iti__tel-input");
    if (!this.telInput.hasAttribute("autocomplete")) {
      this.telInput.setAttribute("autocomplete", "tel");
    }
    if (!this.telInput.hasAttribute("inputmode")) {
      this.telInput.setAttribute("inputmode", "tel");
    }
    const wrapper = __classPrivateFieldGet(this, _UI_instances, "m", _UI_createWrapperAndInsert).call(this);
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_maybeBuildCountryContainer).call(this, wrapper);
    wrapper.appendChild(this.telInput);
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_maybeUpdateInputPaddingAndReveal).call(this);
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_maybeBuildHiddenInputs).call(this, wrapper);
  }
  //* Country search: Filter the countries according to the search query.
  filterCountriesByQuery(query) {
    let matchedCountries;
    if (query === "") {
      matchedCountries = __classPrivateFieldGet(this, _UI_countries, "f");
    } else {
      matchedCountries = getMatchedCountries(__classPrivateFieldGet(this, _UI_countries, "f"), query);
    }
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_filterCountries).call(this, matchedCountries);
  }
  handleSearchChange() {
    if (__classPrivateFieldGet(this, _UI_searchKeyupTimer, "f")) {
      clearTimeout(__classPrivateFieldGet(this, _UI_searchKeyupTimer, "f"));
    }
    __classPrivateFieldSet(this, _UI_searchKeyupTimer, setTimeout(() => {
      __classPrivateFieldGet(this, _UI_instances, "m", _UI_doFilter).call(this);
      __classPrivateFieldSet(this, _UI_searchKeyupTimer, null, "f");
    }, TIMINGS.SEARCH_DEBOUNCE_MS), "f");
  }
  handleSearchClear() {
    this.searchInput.value = "";
    this.searchInput.focus();
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_doFilter).call(this);
  }
  //* Check if an element is visible within it's container, else scroll until it is.
  scrollTo(element) {
    const container = this.countryList;
    const scrollTop = document.documentElement.scrollTop;
    const containerHeight = container.offsetHeight;
    const containerTop = container.getBoundingClientRect().top + scrollTop;
    const containerBottom = containerTop + containerHeight;
    const elementHeight = element.offsetHeight;
    const elementTop = element.getBoundingClientRect().top + scrollTop;
    const elementBottom = elementTop + elementHeight;
    const newScrollTop = elementTop - containerTop + container.scrollTop;
    if (elementTop < containerTop) {
      container.scrollTop = newScrollTop;
    } else if (elementBottom > containerBottom) {
      const heightDifference = containerHeight - elementHeight;
      container.scrollTop = newScrollTop - heightDifference;
    }
  }
  //* Remove highlighting from the previous list item and highlight the new one.
  highlightListItem(listItem, shouldFocus) {
    const prevItem = this.highlightedItem;
    if (prevItem) {
      prevItem.classList.remove(CLASSES.HIGHLIGHT);
    }
    this.highlightedItem = listItem;
    if (this.highlightedItem) {
      this.highlightedItem.classList.add(CLASSES.HIGHLIGHT);
      if (__classPrivateFieldGet(this, _UI_options, "f").countrySearch) {
        const activeDescendant = this.highlightedItem.getAttribute("id") || "";
        this.searchInput.setAttribute(ARIA.ACTIVE_DESCENDANT, activeDescendant);
      }
    }
    if (shouldFocus) {
      this.highlightedItem.focus();
    }
  }
  //* Highlight the next/prev item in the list (and ensure it is visible).
  handleUpDownKey(key) {
    var _b, _c;
    let next = key === KEYS.ARROW_UP ? (_b = this.highlightedItem) === null || _b === void 0 ? void 0 : _b.previousElementSibling : (_c = this.highlightedItem) === null || _c === void 0 ? void 0 : _c.nextElementSibling;
    if (!next && this.countryList.childElementCount > 1) {
      next = key === KEYS.ARROW_UP ? this.countryList.lastElementChild : this.countryList.firstElementChild;
    }
    if (next) {
      this.scrollTo(next);
      this.highlightListItem(next, false);
    }
  }
  destroy() {
    this.telInput.iti = void 0;
    delete this.telInput.dataset.intlTelInputId;
    if (__classPrivateFieldGet(this, _UI_options, "f").separateDialCode) {
      this.telInput.style.paddingLeft = __classPrivateFieldGet(this, _UI_originalPaddingLeft, "f");
    }
    const wrapper = this.telInput.parentNode;
    wrapper.before(this.telInput);
    wrapper.remove();
    this.telInput = null;
    this.countryContainer = null;
    this.selectedCountry = null;
    this.selectedCountryInner = null;
    this.searchInput = null;
    this.searchClearButton = null;
    this.countryList = null;
    this.hiddenInput = null;
    this.hiddenInputCountry = null;
    this.highlightedItem = null;
    __classPrivateFieldSet(this, _UI_selectedDialCode, null, "f");
    __classPrivateFieldSet(this, _UI_dropdownArrow, null, "f");
    __classPrivateFieldSet(this, _UI_dropdownContent, null, "f");
    __classPrivateFieldSet(this, _UI_searchIcon, null, "f");
    __classPrivateFieldSet(this, _UI_searchNoResults, null, "f");
    __classPrivateFieldSet(this, _UI_searchResultsA11yText, null, "f");
    __classPrivateFieldSet(this, _UI_dropdownForContainer, null, "f");
    __classPrivateFieldSet(this, _UI_selectedItem, null, "f");
    for (const c of __classPrivateFieldGet(this, _UI_countries, "f")) {
      delete c.nodeById[__classPrivateFieldGet(this, _UI_id, "f")];
    }
    __classPrivateFieldSet(this, _UI_countries, null, "f");
  }
  // UI: Open the dropdown (DOM only).
  openDropdown() {
    const { countrySearch, dropdownAlwaysOpen, dropdownContainer } = __classPrivateFieldGet(this, _UI_options, "f");
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_maybeEnsureDropdownWidthSet).call(this);
    if (dropdownContainer) {
      __classPrivateFieldGet(this, _UI_instances, "m", _UI_handleDropdownContainer).call(this);
    } else {
      const positionBelow = __classPrivateFieldGet(this, _UI_instances, "m", _UI_shouldPositionInlineDropdownBelowInput).call(this);
      const distance = this.telInput.offsetHeight + LAYOUT.DROPDOWN_MARGIN;
      if (positionBelow) {
        __classPrivateFieldGet(this, _UI_dropdownContent, "f").style.top = `${distance}px`;
      } else {
        __classPrivateFieldGet(this, _UI_dropdownContent, "f").style.bottom = `${distance}px`;
      }
    }
    __classPrivateFieldGet(this, _UI_dropdownContent, "f").classList.remove(CLASSES.HIDE);
    this.selectedCountry.setAttribute(ARIA.EXPANDED, "true");
    if (countrySearch) {
      const firstCountryItem = this.countryList.firstElementChild;
      if (firstCountryItem) {
        this.highlightListItem(firstCountryItem, false);
        this.countryList.scrollTop = 0;
      }
      if (!dropdownAlwaysOpen) {
        this.searchInput.focus();
      }
    }
    __classPrivateFieldGet(this, _UI_dropdownArrow, "f").classList.add(CLASSES.ARROW_UP);
  }
  // UI: Close the dropdown (DOM only).
  closeDropdown() {
    const { countrySearch, dropdownContainer } = __classPrivateFieldGet(this, _UI_options, "f");
    __classPrivateFieldGet(this, _UI_dropdownContent, "f").classList.add(CLASSES.HIDE);
    this.selectedCountry.setAttribute(ARIA.EXPANDED, "false");
    if (countrySearch) {
      this.searchInput.removeAttribute(ARIA.ACTIVE_DESCENDANT);
      if (this.highlightedItem) {
        this.highlightedItem.classList.remove(CLASSES.HIGHLIGHT);
        this.highlightedItem = null;
      }
    }
    __classPrivateFieldGet(this, _UI_dropdownArrow, "f").classList.remove(CLASSES.ARROW_UP);
    if (dropdownContainer) {
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").remove();
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.top = "";
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.bottom = "";
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.paddingLeft = "";
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.paddingRight = "";
    } else {
      __classPrivateFieldGet(this, _UI_dropdownContent, "f").style.top = "";
      __classPrivateFieldGet(this, _UI_dropdownContent, "f").style.bottom = "";
    }
  }
  // UI: Whether the dropdown is currently closed (hidden).
  isDropdownClosed() {
    return __classPrivateFieldGet(this, _UI_dropdownContent, "f").classList.contains(CLASSES.HIDE);
  }
  setCountry(selectedCountryData) {
    const { allowDropdown, showFlags, separateDialCode, i18n } = __classPrivateFieldGet(this, _UI_options, "f");
    const { name, dialCode, iso2 = "" } = selectedCountryData;
    if (allowDropdown) {
      __classPrivateFieldGet(this, _UI_instances, "m", _UI_updateSelectedItem).call(this, iso2);
    }
    if (this.selectedCountry) {
      const flagClass = iso2 && showFlags ? `${CLASSES.FLAG} iti__${iso2}` : `${CLASSES.FLAG} ${CLASSES.GLOBE}`;
      let ariaLabel, title, selectedCountryInner;
      if (iso2) {
        title = name;
        ariaLabel = i18n.selectedCountryAriaLabel.replace("${countryName}", name).replace("${dialCode}", `+${dialCode}`);
        selectedCountryInner = showFlags ? "" : buildGlobeIcon();
      } else {
        title = i18n.noCountrySelected;
        ariaLabel = i18n.noCountrySelected;
        selectedCountryInner = buildGlobeIcon();
      }
      this.selectedCountryInner.className = flagClass;
      this.selectedCountry.setAttribute("title", title);
      this.selectedCountry.setAttribute(ARIA.LABEL, ariaLabel);
      this.selectedCountryInner.innerHTML = selectedCountryInner;
    }
    if (separateDialCode) {
      const fullDialCode = dialCode ? `+${dialCode}` : "";
      __classPrivateFieldGet(this, _UI_selectedDialCode, "f").textContent = fullDialCode;
      __classPrivateFieldGet(this, _UI_instances, "m", _UI_updateInputPadding).call(this);
    }
  }
};
_a = UI, _UI_options = /* @__PURE__ */ new WeakMap(), _UI_id = /* @__PURE__ */ new WeakMap(), _UI_isRTL = /* @__PURE__ */ new WeakMap(), _UI_originalPaddingLeft = /* @__PURE__ */ new WeakMap(), _UI_countries = /* @__PURE__ */ new WeakMap(), _UI_searchKeyupTimer = /* @__PURE__ */ new WeakMap(), _UI_inlineDropdownHeight = /* @__PURE__ */ new WeakMap(), _UI_selectedDialCode = /* @__PURE__ */ new WeakMap(), _UI_dropdownArrow = /* @__PURE__ */ new WeakMap(), _UI_dropdownContent = /* @__PURE__ */ new WeakMap(), _UI_searchIcon = /* @__PURE__ */ new WeakMap(), _UI_searchNoResults = /* @__PURE__ */ new WeakMap(), _UI_searchResultsA11yText = /* @__PURE__ */ new WeakMap(), _UI_dropdownForContainer = /* @__PURE__ */ new WeakMap(), _UI_selectedItem = /* @__PURE__ */ new WeakMap(), _UI_instances = /* @__PURE__ */ new WeakSet(), _UI_createWrapperAndInsert = function _UI_createWrapperAndInsert2() {
  const { allowDropdown, showFlags, containerClass, useFullscreenPopup } = __classPrivateFieldGet(this, _UI_options, "f");
  const parentClasses = buildClassNames({
    iti: true,
    "iti--allow-dropdown": allowDropdown,
    "iti--show-flags": showFlags,
    "iti--inline-dropdown": !useFullscreenPopup,
    [containerClass]: Boolean(containerClass)
  });
  const wrapper = createEl("div", { class: parentClasses });
  if (__classPrivateFieldGet(this, _UI_isRTL, "f")) {
    wrapper.setAttribute("dir", "ltr");
  }
  this.telInput.before(wrapper);
  return wrapper;
}, _UI_maybeBuildCountryContainer = function _UI_maybeBuildCountryContainer2(wrapper) {
  const { allowDropdown, separateDialCode, showFlags } = __classPrivateFieldGet(this, _UI_options, "f");
  if (allowDropdown || showFlags || separateDialCode) {
    this.countryContainer = createEl(
      "div",
      // visibly hidden until we measure it's width to set the input padding correctly
      { class: `iti__country-container ${CLASSES.V_HIDE}` },
      wrapper
    );
    if (allowDropdown) {
      this.selectedCountry = createEl("button", {
        type: "button",
        class: "iti__selected-country",
        [ARIA.EXPANDED]: "false",
        [ARIA.LABEL]: __classPrivateFieldGet(this, _UI_options, "f").i18n.noCountrySelected,
        [ARIA.HASPOPUP]: "dialog",
        [ARIA.CONTROLS]: `iti-${__classPrivateFieldGet(this, _UI_id, "f")}__dropdown-content`
      }, this.countryContainer);
      if (this.telInput.disabled) {
        this.selectedCountry.setAttribute("disabled", "true");
      }
    } else {
      this.selectedCountry = createEl("div", { class: "iti__selected-country" }, this.countryContainer);
    }
    const selectedCountryPrimary = createEl("div", { class: "iti__selected-country-primary" }, this.selectedCountry);
    this.selectedCountryInner = createEl("div", { class: CLASSES.FLAG }, selectedCountryPrimary);
    if (allowDropdown) {
      __classPrivateFieldSet(this, _UI_dropdownArrow, createEl("div", { class: "iti__arrow", [ARIA.HIDDEN]: "true" }, selectedCountryPrimary), "f");
    }
    if (separateDialCode) {
      __classPrivateFieldSet(this, _UI_selectedDialCode, createEl("div", { class: "iti__selected-dial-code" }, this.selectedCountry), "f");
    }
    if (allowDropdown) {
      __classPrivateFieldGet(this, _UI_instances, "m", _UI_buildDropdownContent).call(this);
    }
  }
}, _UI_maybeEnsureDropdownWidthSet = function _UI_maybeEnsureDropdownWidthSet2() {
  const { fixDropdownWidth } = __classPrivateFieldGet(this, _UI_options, "f");
  if (fixDropdownWidth && !__classPrivateFieldGet(this, _UI_dropdownContent, "f").style.width) {
    const inputWidth = this.telInput.offsetWidth;
    if (inputWidth > 0) {
      __classPrivateFieldGet(this, _UI_dropdownContent, "f").style.width = `${inputWidth}px`;
    }
  }
}, _UI_buildDropdownContent = function _UI_buildDropdownContent2() {
  const { fixDropdownWidth, useFullscreenPopup, countrySearch, i18n, dropdownContainer, containerClass } = __classPrivateFieldGet(this, _UI_options, "f");
  const extraClasses = fixDropdownWidth ? "" : "iti--flexible-dropdown-width";
  __classPrivateFieldSet(this, _UI_dropdownContent, createEl("div", {
    id: `iti-${__classPrivateFieldGet(this, _UI_id, "f")}__dropdown-content`,
    class: `iti__dropdown-content ${CLASSES.HIDE} ${extraClasses}`,
    role: "dialog",
    [ARIA.MODAL]: "true"
  }), "f");
  if (__classPrivateFieldGet(this, _UI_isRTL, "f")) {
    __classPrivateFieldGet(this, _UI_dropdownContent, "f").setAttribute("dir", "rtl");
  }
  if (countrySearch) {
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_buildSearchUI).call(this);
  }
  this.countryList = createEl("ul", {
    class: "iti__country-list",
    id: `iti-${__classPrivateFieldGet(this, _UI_id, "f")}__country-listbox`,
    role: "listbox",
    [ARIA.LABEL]: i18n.countryListAriaLabel
  }, __classPrivateFieldGet(this, _UI_dropdownContent, "f"));
  __classPrivateFieldGet(this, _UI_instances, "m", _UI_appendListItems).call(this);
  if (countrySearch) {
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_updateSearchResultsA11yText).call(this);
  }
  if (!useFullscreenPopup) {
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_maybeEnsureDropdownWidthSet).call(this);
    __classPrivateFieldSet(this, _UI_inlineDropdownHeight, __classPrivateFieldGet(this, _UI_instances, "m", _UI_getHiddenInlineDropdownHeight).call(this), "f");
    if (countrySearch) {
      __classPrivateFieldGet(this, _UI_dropdownContent, "f").style.height = `${__classPrivateFieldGet(this, _UI_inlineDropdownHeight, "f")}px`;
    }
  }
  if (dropdownContainer) {
    const dropdownClasses = buildClassNames({
      iti: true,
      "iti--container": true,
      "iti--fullscreen-popup": useFullscreenPopup,
      "iti--inline-dropdown": !useFullscreenPopup,
      [containerClass]: Boolean(containerClass)
    });
    __classPrivateFieldSet(this, _UI_dropdownForContainer, createEl("div", { class: dropdownClasses }), "f");
    __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").appendChild(__classPrivateFieldGet(this, _UI_dropdownContent, "f"));
  } else {
    this.countryContainer.appendChild(__classPrivateFieldGet(this, _UI_dropdownContent, "f"));
  }
}, _UI_buildSearchUI = function _UI_buildSearchUI2() {
  const { i18n, searchInputClass } = __classPrivateFieldGet(this, _UI_options, "f");
  const searchWrapper = createEl("div", { class: "iti__search-input-wrapper" }, __classPrivateFieldGet(this, _UI_dropdownContent, "f"));
  __classPrivateFieldSet(this, _UI_searchIcon, createEl("span", {
    class: "iti__search-icon",
    [ARIA.HIDDEN]: "true"
  }, searchWrapper), "f");
  __classPrivateFieldGet(this, _UI_searchIcon, "f").innerHTML = buildSearchIcon();
  this.searchInput = createEl("input", {
    id: `iti-${__classPrivateFieldGet(this, _UI_id, "f")}__search-input`,
    // Chrome says inputs need either a name or an id
    type: "search",
    class: `iti__search-input ${searchInputClass}`,
    placeholder: i18n.searchPlaceholder,
    // role=combobox + aria-autocomplete=list + aria-activedescendant allows maintaining focus on the search input while allowing users to navigate search results with up/down keyboard keys
    role: "combobox",
    [ARIA.EXPANDED]: "true",
    [ARIA.LABEL]: i18n.searchPlaceholder,
    [ARIA.CONTROLS]: `iti-${__classPrivateFieldGet(this, _UI_id, "f")}__country-listbox`,
    [ARIA.AUTOCOMPLETE]: "list",
    autocomplete: "off"
  }, searchWrapper);
  this.searchClearButton = createEl("button", {
    type: "button",
    class: `iti__search-clear ${CLASSES.HIDE}`,
    [ARIA.LABEL]: i18n.clearSearchAriaLabel,
    tabindex: "-1"
  }, searchWrapper);
  this.searchClearButton.innerHTML = buildClearIcon(__classPrivateFieldGet(this, _UI_id, "f"));
  __classPrivateFieldSet(this, _UI_searchResultsA11yText, createEl("span", { class: "iti__a11y-text" }, __classPrivateFieldGet(this, _UI_dropdownContent, "f")), "f");
  __classPrivateFieldSet(this, _UI_searchNoResults, createEl("div", {
    class: `iti__no-results ${CLASSES.HIDE}`,
    [ARIA.HIDDEN]: "true"
    // all a11y messaging happens in this.#searchResultsA11yText
  }, __classPrivateFieldGet(this, _UI_dropdownContent, "f")), "f");
  __classPrivateFieldGet(this, _UI_searchNoResults, "f").textContent = i18n.searchEmptyState;
}, _UI_maybeUpdateInputPaddingAndReveal = function _UI_maybeUpdateInputPaddingAndReveal2() {
  if (this.countryContainer) {
    __classPrivateFieldGet(this, _UI_instances, "m", _UI_updateInputPadding).call(this);
    this.countryContainer.classList.remove(CLASSES.V_HIDE);
  }
}, _UI_maybeBuildHiddenInputs = function _UI_maybeBuildHiddenInputs2(wrapper) {
  var _b, _c;
  const { hiddenInput } = __classPrivateFieldGet(this, _UI_options, "f");
  if (hiddenInput) {
    const telInputName = this.telInput.getAttribute("name") || "";
    const names = hiddenInput(telInputName);
    if (names.phone) {
      const existingInput = (_b = this.telInput.form) === null || _b === void 0 ? void 0 : _b.querySelector(`input[name="${names.phone}"]`);
      if (existingInput) {
        this.hiddenInput = existingInput;
      } else {
        this.hiddenInput = createEl("input", {
          type: "hidden",
          name: names.phone
        });
        wrapper.appendChild(this.hiddenInput);
      }
    }
    if (names.country) {
      const existingInput = (_c = this.telInput.form) === null || _c === void 0 ? void 0 : _c.querySelector(`input[name="${names.country}"]`);
      if (existingInput) {
        this.hiddenInputCountry = existingInput;
      } else {
        this.hiddenInputCountry = createEl("input", {
          type: "hidden",
          name: names.country
        });
        wrapper.appendChild(this.hiddenInputCountry);
      }
    }
  }
}, _UI_appendListItems = function _UI_appendListItems2() {
  const frag = document.createDocumentFragment();
  for (let i = 0; i < __classPrivateFieldGet(this, _UI_countries, "f").length; i++) {
    const c = __classPrivateFieldGet(this, _UI_countries, "f")[i];
    const liClass = buildClassNames({
      [CLASSES.COUNTRY_ITEM]: true
    });
    const listItem = createEl("li", {
      id: `iti-${__classPrivateFieldGet(this, _UI_id, "f")}__item-${c.iso2}`,
      class: liClass,
      tabindex: "-1",
      role: "option",
      [ARIA.SELECTED]: "false"
    });
    listItem.dataset.dialCode = c.dialCode;
    listItem.dataset.countryCode = c.iso2;
    c.nodeById[__classPrivateFieldGet(this, _UI_id, "f")] = listItem;
    if (__classPrivateFieldGet(this, _UI_options, "f").showFlags) {
      createEl("div", { class: `${CLASSES.FLAG} iti__${c.iso2}` }, listItem);
    }
    const nameEl = createEl("span", { class: "iti__country-name" }, listItem);
    nameEl.textContent = `${c.name} `;
    const dialEl = createEl("span", { class: "iti__dial-code" }, nameEl);
    if (__classPrivateFieldGet(this, _UI_isRTL, "f")) {
      dialEl.setAttribute("dir", "ltr");
    }
    dialEl.textContent = `(+${c.dialCode})`;
    frag.appendChild(listItem);
  }
  this.countryList.appendChild(frag);
}, _UI_updateInputPadding = function _UI_updateInputPadding2() {
  if (this.selectedCountry) {
    const fallbackWidth = __classPrivateFieldGet(this, _UI_options, "f").separateDialCode ? LAYOUT.SANE_SELECTED_WITH_DIAL_WIDTH : LAYOUT.SANE_SELECTED_NO_DIAL_WIDTH;
    const selectedCountryWidth = this.selectedCountry.offsetWidth || __classPrivateFieldGet(this, _UI_instances, "m", _UI_getHiddenSelectedCountryWidth).call(this) || fallbackWidth;
    const inputPadding = selectedCountryWidth + LAYOUT.INPUT_PADDING_EXTRA_LEFT;
    this.telInput.style.paddingLeft = `${inputPadding}px`;
  }
}, _UI_getBody = function _UI_getBody2() {
  let body;
  try {
    body = window.top.document.body;
  } catch (e) {
    body = document.body;
  }
  return body;
}, _UI_getHiddenSelectedCountryWidth = function _UI_getHiddenSelectedCountryWidth2() {
  if (this.telInput.parentNode) {
    const body = __classPrivateFieldGet(_a, _a, "m", _UI_getBody).call(_a);
    const containerClone = this.telInput.parentNode.cloneNode(false);
    containerClone.style.visibility = "hidden";
    body.appendChild(containerClone);
    const countryContainerClone = this.countryContainer.cloneNode();
    containerClone.appendChild(countryContainerClone);
    const selectedCountryClone = this.selectedCountry.cloneNode(true);
    countryContainerClone.appendChild(selectedCountryClone);
    const width = selectedCountryClone.offsetWidth;
    body.removeChild(containerClone);
    return width;
  }
  return 0;
}, _UI_getHiddenInlineDropdownHeight = function _UI_getHiddenInlineDropdownHeight2() {
  const body = __classPrivateFieldGet(_a, _a, "m", _UI_getBody).call(_a);
  __classPrivateFieldGet(this, _UI_dropdownContent, "f").classList.remove(CLASSES.HIDE);
  const tempContainer = createEl("div", { class: "iti iti--inline-dropdown" });
  tempContainer.appendChild(__classPrivateFieldGet(this, _UI_dropdownContent, "f"));
  tempContainer.style.visibility = "hidden";
  body.appendChild(tempContainer);
  const height = __classPrivateFieldGet(this, _UI_dropdownContent, "f").offsetHeight;
  body.removeChild(tempContainer);
  tempContainer.style.visibility = "";
  __classPrivateFieldGet(this, _UI_dropdownContent, "f").classList.add(CLASSES.HIDE);
  return height > 0 ? height : LAYOUT.SANE_DROPDOWN_HEIGHT;
}, _UI_updateSearchResultsA11yText = function _UI_updateSearchResultsA11yText2() {
  const { i18n } = __classPrivateFieldGet(this, _UI_options, "f");
  const count = this.countryList.childElementCount;
  __classPrivateFieldGet(this, _UI_searchResultsA11yText, "f").textContent = i18n.searchSummaryAria(count);
}, _UI_doFilter = function _UI_doFilter2() {
  const inputQuery = this.searchInput.value.trim();
  this.filterCountriesByQuery(inputQuery);
  if (this.searchInput.value) {
    this.searchClearButton.classList.remove(CLASSES.HIDE);
  } else {
    this.searchClearButton.classList.add(CLASSES.HIDE);
  }
}, _UI_updateSelectedItem = function _UI_updateSelectedItem2(iso2) {
  var _b;
  if (__classPrivateFieldGet(this, _UI_selectedItem, "f") && __classPrivateFieldGet(this, _UI_selectedItem, "f").dataset.countryCode !== iso2) {
    __classPrivateFieldGet(this, _UI_selectedItem, "f").setAttribute(ARIA.SELECTED, "false");
    (_b = __classPrivateFieldGet(this, _UI_selectedItem, "f").querySelector(".iti__country-check")) === null || _b === void 0 ? void 0 : _b.remove();
    __classPrivateFieldSet(this, _UI_selectedItem, null, "f");
  }
  if (iso2 && !__classPrivateFieldGet(this, _UI_selectedItem, "f")) {
    const newListItem = this.countryList.querySelector(`[data-country-code="${iso2}"]`);
    if (newListItem) {
      newListItem.setAttribute(ARIA.SELECTED, "true");
      const checkIcon = createEl("span", { class: "iti__country-check", [ARIA.HIDDEN]: "true" }, newListItem);
      checkIcon.innerHTML = buildCheckIcon();
      __classPrivateFieldSet(this, _UI_selectedItem, newListItem, "f");
    }
  }
}, _UI_filterCountries = function _UI_filterCountries2(matchedCountries) {
  this.countryList.innerHTML = "";
  let noCountriesAddedYet = true;
  for (const c of matchedCountries) {
    const listItem = c.nodeById[__classPrivateFieldGet(this, _UI_id, "f")];
    if (listItem) {
      this.countryList.appendChild(listItem);
      if (noCountriesAddedYet) {
        this.highlightListItem(listItem, false);
        noCountriesAddedYet = false;
      }
    }
  }
  if (noCountriesAddedYet) {
    this.highlightListItem(null, false);
    if (__classPrivateFieldGet(this, _UI_searchNoResults, "f")) {
      __classPrivateFieldGet(this, _UI_searchNoResults, "f").classList.remove(CLASSES.HIDE);
    }
  } else if (__classPrivateFieldGet(this, _UI_searchNoResults, "f")) {
    __classPrivateFieldGet(this, _UI_searchNoResults, "f").classList.add(CLASSES.HIDE);
  }
  this.countryList.scrollTop = 0;
  __classPrivateFieldGet(this, _UI_instances, "m", _UI_updateSearchResultsA11yText).call(this);
}, _UI_shouldPositionInlineDropdownBelowInput = function _UI_shouldPositionInlineDropdownBelowInput2() {
  if (__classPrivateFieldGet(this, _UI_options, "f").dropdownAlwaysOpen) {
    return true;
  }
  const inputPos = this.telInput.getBoundingClientRect();
  const spaceAbove = inputPos.top;
  const spaceBelow = window.innerHeight - inputPos.bottom;
  return spaceBelow >= __classPrivateFieldGet(this, _UI_inlineDropdownHeight, "f") || spaceBelow >= spaceAbove;
}, _UI_handleDropdownContainer = function _UI_handleDropdownContainer2() {
  const { dropdownContainer, useFullscreenPopup } = __classPrivateFieldGet(this, _UI_options, "f");
  if (useFullscreenPopup) {
    if (window.innerWidth >= 500) {
      const inputPos = this.telInput.getBoundingClientRect();
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.paddingLeft = `${inputPos.left}px`;
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.paddingRight = `${window.innerWidth - inputPos.right}px`;
    }
  } else {
    const inputPos = this.telInput.getBoundingClientRect();
    __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.left = `${inputPos.left}px`;
    const positionBelow = __classPrivateFieldGet(this, _UI_instances, "m", _UI_shouldPositionInlineDropdownBelowInput).call(this);
    if (positionBelow) {
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.top = `${inputPos.bottom + LAYOUT.DROPDOWN_MARGIN}px`;
    } else {
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.top = "unset";
      __classPrivateFieldGet(this, _UI_dropdownForContainer, "f").style.bottom = `${window.innerHeight - inputPos.top + LAYOUT.DROPDOWN_MARGIN}px`;
    }
  }
  dropdownContainer.appendChild(__classPrivateFieldGet(this, _UI_dropdownForContainer, "f"));
};
var ui_default = UI;

// angular/build/temp/modules/data/country-data.js
var processAllCountries = (options) => {
  const { onlyCountries, excludeCountries } = options;
  if (onlyCountries === null || onlyCountries === void 0 ? void 0 : onlyCountries.length) {
    const lowerCaseOnlyCountries = onlyCountries.map((country) => country.toLowerCase());
    return data_default.filter((country) => lowerCaseOnlyCountries.includes(country.iso2));
  } else if (excludeCountries === null || excludeCountries === void 0 ? void 0 : excludeCountries.length) {
    const lowerCaseExcludeCountries = excludeCountries.map((country) => country.toLowerCase());
    return data_default.filter((country) => !lowerCaseExcludeCountries.includes(country.iso2));
  }
  return data_default;
};
var generateCountryNames = (countries, options) => {
  const { countryNameLocale, i18n } = options;
  let displayNames;
  try {
    const hasDisplayNames = typeof Intl !== "undefined" && typeof Intl.DisplayNames === "function";
    if (hasDisplayNames) {
      displayNames = new Intl.DisplayNames(countryNameLocale, {
        type: "region"
      });
    } else {
      displayNames = null;
    }
  } catch (e) {
    console.error(e);
    displayNames = null;
  }
  for (const c of countries) {
    c.name = i18n[c.iso2] || (displayNames === null || displayNames === void 0 ? void 0 : displayNames.of(c.iso2.toUpperCase())) || "";
  }
};
var processDialCodes = (countries) => {
  const dialCodes = /* @__PURE__ */ new Set();
  let dialCodeMaxLen = 0;
  const dialCodeToIso2Map = {};
  const addToDialCodeMap = (iso2, dialCode) => {
    if (!iso2 || !dialCode) {
      return;
    }
    if (dialCode.length > dialCodeMaxLen) {
      dialCodeMaxLen = dialCode.length;
    }
    if (!dialCodeToIso2Map.hasOwnProperty(dialCode)) {
      dialCodeToIso2Map[dialCode] = [];
    }
    const iso2List = dialCodeToIso2Map[dialCode];
    if (iso2List.includes(iso2)) {
      return;
    }
    iso2List.push(iso2);
  };
  const countriesSortedByPriority = [...countries].sort((a, b) => a.priority - b.priority);
  for (const c of countriesSortedByPriority) {
    if (!dialCodes.has(c.dialCode)) {
      dialCodes.add(c.dialCode);
    }
    for (let k = 1; k < c.dialCode.length; k++) {
      const partialDialCode = c.dialCode.substring(0, k);
      addToDialCodeMap(c.iso2, partialDialCode);
    }
    addToDialCodeMap(c.iso2, c.dialCode);
    if (c.areaCodes) {
      const rootIso2Code = dialCodeToIso2Map[c.dialCode][0];
      for (const areaCode of c.areaCodes) {
        for (let k = 1; k < areaCode.length; k++) {
          const partialAreaCode = areaCode.substring(0, k);
          const partialDialCode = c.dialCode + partialAreaCode;
          addToDialCodeMap(rootIso2Code, partialDialCode);
          addToDialCodeMap(c.iso2, partialDialCode);
        }
        addToDialCodeMap(c.iso2, c.dialCode + areaCode);
      }
    }
  }
  return { dialCodes, dialCodeMaxLen, dialCodeToIso2Map };
};
var sortCountries = (countries, options) => {
  if (options.countryOrder) {
    options.countryOrder = options.countryOrder.map((iso2) => iso2.toLowerCase());
  }
  countries.sort((a, b) => {
    const { countryOrder } = options;
    if (countryOrder) {
      const aIndex = countryOrder.indexOf(a.iso2);
      const bIndex = countryOrder.indexOf(b.iso2);
      const aIndexExists = aIndex > -1;
      const bIndexExists = bIndex > -1;
      if (aIndexExists || bIndexExists) {
        if (aIndexExists && bIndexExists) {
          return aIndex - bIndex;
        }
        return aIndexExists ? -1 : 1;
      }
    }
    return a.name.localeCompare(b.name);
  });
};
var cacheSearchTokens = (countries) => {
  for (const c of countries) {
    c.normalisedName = normaliseString(c.name);
    c.initials = c.normalisedName.split(/[^a-z]/).map((word) => word[0]).join("");
    c.dialCodePlus = `+${c.dialCode}`;
  }
};

// angular/build/temp/modules/data/intl-regionless.js
var REGIONLESS_DIAL_CODES = /* @__PURE__ */ new Set([
  "800",
  "808",
  "870",
  "881",
  "882",
  "883",
  "888",
  "979"
]);
var hasRegionlessDialCode = (number) => {
  const dialCode = getNumeric(number).slice(0, 3);
  return number.startsWith("+") && REGIONLESS_DIAL_CODES.has(dialCode);
};

// angular/build/temp/modules/format/formatting.js
var beforeSetNumber = (fullNumber, hasValidDialCode, separateDialCode, selectedCountryData) => {
  let number = fullNumber;
  if (separateDialCode) {
    if (hasValidDialCode) {
      const dialCode = `+${selectedCountryData.dialCode}`;
      const start = number[dialCode.length] === " " || number[dialCode.length] === "-" ? dialCode.length + 1 : dialCode.length;
      number = number.substring(start);
    }
  }
  return number;
};
var formatNumberAsYouType = (fullNumber, telInputValue, utils, selectedCountryData, separateDialCode) => {
  const result = utils ? utils.formatNumberAsYouType(fullNumber, selectedCountryData.iso2) : fullNumber;
  const { dialCode } = selectedCountryData;
  if (separateDialCode && telInputValue.charAt(0) !== "+" && result.includes(`+${dialCode}`)) {
    const afterDialCode = result.split(`+${dialCode}`)[1] || "";
    return afterDialCode.trim();
  }
  return result;
};

// angular/build/temp/modules/format/caret.js
var translateCursorPosition = (relevantChars, formattedValue, prevCaretPos, isDeleteForwards) => {
  if (prevCaretPos === 0 && !isDeleteForwards) {
    return 0;
  }
  let relevantCharCount = 0;
  for (let i = 0; i < formattedValue.length; i++) {
    if (/[+0-9]/.test(formattedValue[i])) {
      relevantCharCount++;
    }
    if (relevantCharCount === relevantChars && !isDeleteForwards) {
      return i + 1;
    }
    if (isDeleteForwards && relevantCharCount === relevantChars + 1) {
      return i;
    }
  }
  return formattedValue.length;
};

// angular/build/temp/modules/data/nanp-regionless.js
var regionlessNanpNumbers = [
  "800",
  "822",
  "833",
  "844",
  "855",
  "866",
  "877",
  "880",
  "881",
  "882",
  "883",
  "884",
  "885",
  "886",
  "887",
  "888",
  "889"
];
var isRegionlessNanp = (number) => {
  const numeric = getNumeric(number);
  if (numeric.startsWith(DIAL.NANP) && numeric.length >= 4) {
    const areaCode = numeric.substring(1, 4);
    return regionlessNanpNumbers.includes(areaCode);
  }
  return false;
};

// angular/build/temp/modules/core/numerals.js
var __classPrivateFieldSet2 = function(receiver, state, value, kind, f) {
  if (kind === "m") throw new TypeError("Private method is not writable");
  if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
  if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
  return kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value), value;
};
var __classPrivateFieldGet2 = function(receiver, state, kind, f) {
  if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
  if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
  return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _Numerals_instances;
var _Numerals_userNumeralSet;
var _Numerals_updateNumeralSet;
var Numerals = class {
  constructor() {
    _Numerals_instances.add(this);
    _Numerals_userNumeralSet.set(this, void 0);
  }
  // Denormalise ASCII 0-9 to the user's numeral set, if known. If not known, return the string as-is.
  denormalise(str, currentInputValue) {
    if (!__classPrivateFieldGet2(this, _Numerals_userNumeralSet, "f")) {
      __classPrivateFieldGet2(this, _Numerals_instances, "m", _Numerals_updateNumeralSet).call(this, currentInputValue);
    }
    if (__classPrivateFieldGet2(this, _Numerals_userNumeralSet, "f") === "ascii") {
      return str;
    }
    const base = __classPrivateFieldGet2(this, _Numerals_userNumeralSet, "f") === "arabic-indic" ? 1632 : 1776;
    return str.replace(/[0-9]/g, (d) => String.fromCharCode(base + Number(d)));
  }
  // Normalize Eastern Arabic (U+0660-0669) and Persian/Extended Arabic-Indic (U+06F0-06F9) numerals to ASCII 0-9
  normalise(str) {
    if (!str) {
      return "";
    }
    __classPrivateFieldGet2(this, _Numerals_instances, "m", _Numerals_updateNumeralSet).call(this, str);
    if (__classPrivateFieldGet2(this, _Numerals_userNumeralSet, "f") === "ascii") {
      return str;
    }
    const base = __classPrivateFieldGet2(this, _Numerals_userNumeralSet, "f") === "arabic-indic" ? 1632 : 1776;
    const regex = __classPrivateFieldGet2(this, _Numerals_userNumeralSet, "f") === "arabic-indic" ? /[\u0660-\u0669]/g : /[\u06F0-\u06F9]/g;
    return str.replace(regex, (ch) => String.fromCharCode(48 + (ch.charCodeAt(0) - base)));
  }
  isAscii() {
    return __classPrivateFieldGet2(this, _Numerals_userNumeralSet, "f") === "ascii";
  }
};
_Numerals_userNumeralSet = /* @__PURE__ */ new WeakMap(), _Numerals_instances = /* @__PURE__ */ new WeakSet(), _Numerals_updateNumeralSet = function _Numerals_updateNumeralSet2(str) {
  if (/[\u0660-\u0669]/.test(str)) {
    __classPrivateFieldSet2(this, _Numerals_userNumeralSet, "arabic-indic", "f");
  } else if (/[\u06F0-\u06F9]/.test(str)) {
    __classPrivateFieldSet2(this, _Numerals_userNumeralSet, "persian", "f");
  } else {
    __classPrivateFieldSet2(this, _Numerals_userNumeralSet, "ascii", "f");
  }
};

// angular/build/temp/intl-tel-input.js
var __classPrivateFieldSet3 = function(receiver, state, value, kind, f) {
  if (kind === "m") throw new TypeError("Private method is not writable");
  if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
  if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
  return kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value), value;
};
var __classPrivateFieldGet3 = function(receiver, state, kind, f) {
  if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
  if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
  return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _Iti_instances;
var _a2;
var _Iti_ui;
var _Iti_options;
var _Iti_isAndroid;
var _Iti_countries;
var _Iti_dialCodeMaxLen;
var _Iti_dialCodeToIso2Map;
var _Iti_dialCodes;
var _Iti_countryByIso2;
var _Iti_selectedCountryData;
var _Iti_maxCoreNumberLength;
var _Iti_defaultCountry;
var _Iti_abortController;
var _Iti_dropdownAbortController;
var _Iti_numerals;
var _Iti_resolveAutoCountryPromise;
var _Iti_rejectAutoCountryPromise;
var _Iti_resolveUtilsScriptPromise;
var _Iti_rejectUtilsScriptPromise;
var _Iti_getTelInputValue;
var _Iti_setTelInputValue;
var _Iti_createInitPromises;
var _Iti_init;
var _Iti_processCountryData;
var _Iti_setInitialState;
var _Iti_initListeners;
var _Iti_initHiddenInputListener;
var _Iti_initDropdownListeners;
var _Iti_initRequests;
var _Iti_loadAutoCountry;
var _Iti_openDropdownWithPlus;
var _Iti_initTelInputListeners;
var _Iti_bindInputListener;
var _Iti_maybeBindKeydownListener;
var _Iti_maybeBindPasteListener;
var _Iti_cap;
var _Iti_trigger;
var _Iti_openDropdown;
var _Iti_bindDropdownListeners;
var _Iti_bindDropdownMouseoverListener;
var _Iti_bindDropdownCountryClickListener;
var _Iti_bindDropdownClickOffListener;
var _Iti_bindDropdownKeydownListener;
var _Iti_bindDropdownSearchListeners;
var _Iti_searchForCountry;
var _Iti_handleEnterKey;
var _Iti_updateValFromNumber;
var _Iti_updateCountryFromNumber;
var _Iti_ensureHasDialCode;
var _Iti_getNewCountryFromNumber;
var _Iti_setCountry;
var _Iti_updateMaxLength;
var _Iti_updatePlaceholder;
var _Iti_selectListItem;
var _Iti_closeDropdown;
var _Iti_updateDialCode;
var _Iti_getDialCode;
var _Iti_getFullNumber;
var _Iti_beforeSetNumber;
var _Iti_triggerCountryChange;
var _Iti_handleAutoCountry;
var _Iti_handleAutoCountryFailure;
var _Iti_handleUtils;
var _Iti_handleUtilsFailure;
var _Iti_utilsIsPossibleNumber;
var _Iti_validateNumber;
var _Iti_utilsIsValidNumber;
var id = 0;
var iso2Set2 = new Set(data_default.map((c) => c.iso2));
var isIso22 = (val) => iso2Set2.has(val);
var Iti = class {
  constructor(input, customOptions = {}) {
    _Iti_instances.add(this);
    _Iti_ui.set(this, void 0);
    _Iti_options.set(this, void 0);
    _Iti_isAndroid.set(this, void 0);
    _Iti_countries.set(this, void 0);
    _Iti_dialCodeMaxLen.set(this, void 0);
    _Iti_dialCodeToIso2Map.set(this, void 0);
    _Iti_dialCodes.set(this, void 0);
    _Iti_countryByIso2.set(this, void 0);
    _Iti_selectedCountryData.set(this, void 0);
    _Iti_maxCoreNumberLength.set(this, void 0);
    _Iti_defaultCountry.set(this, void 0);
    _Iti_abortController.set(this, void 0);
    _Iti_dropdownAbortController.set(this, void 0);
    _Iti_numerals.set(this, void 0);
    _Iti_resolveAutoCountryPromise.set(this, void 0);
    _Iti_rejectAutoCountryPromise.set(this, void 0);
    _Iti_resolveUtilsScriptPromise.set(this, void 0);
    _Iti_rejectUtilsScriptPromise.set(this, void 0);
    this.id = id++;
    ui_default.validateInput(input);
    const validatedOptions = validateOptions(customOptions);
    __classPrivateFieldSet3(this, _Iti_options, Object.assign(Object.assign({}, defaults), validatedOptions), "f");
    applyOptionSideEffects(__classPrivateFieldGet3(this, _Iti_options, "f"));
    __classPrivateFieldSet3(this, _Iti_ui, new ui_default(input, __classPrivateFieldGet3(this, _Iti_options, "f"), this.id), "f");
    __classPrivateFieldSet3(this, _Iti_isAndroid, getIsAndroid(), "f");
    __classPrivateFieldSet3(this, _Iti_numerals, new Numerals(), "f");
    this.promise = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_createInitPromises).call(this, __classPrivateFieldGet3(this, _Iti_options, "f"));
    __classPrivateFieldSet3(this, _Iti_countries, processAllCountries(__classPrivateFieldGet3(this, _Iti_options, "f")), "f");
    const { dialCodes, dialCodeMaxLen, dialCodeToIso2Map } = processDialCodes(__classPrivateFieldGet3(this, _Iti_countries, "f"));
    __classPrivateFieldSet3(this, _Iti_dialCodes, dialCodes, "f");
    __classPrivateFieldSet3(this, _Iti_dialCodeMaxLen, dialCodeMaxLen, "f");
    __classPrivateFieldSet3(this, _Iti_dialCodeToIso2Map, dialCodeToIso2Map, "f");
    __classPrivateFieldSet3(this, _Iti_countryByIso2, new Map(__classPrivateFieldGet3(this, _Iti_countries, "f").map((c) => [c.iso2, c])), "f");
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_init).call(this);
  }
  //********************
  //*  PUBLIC METHODS
  //********************
  //* Remove plugin.
  destroy() {
    if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      return;
    }
    if (__classPrivateFieldGet3(this, _Iti_options, "f").allowDropdown) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_closeDropdown).call(this, true);
    }
    __classPrivateFieldGet3(this, _Iti_abortController, "f").abort();
    __classPrivateFieldSet3(this, _Iti_abortController, null, "f");
    __classPrivateFieldGet3(this, _Iti_ui, "f").destroy();
    if (intlTelInput.instances instanceof Map) {
      intlTelInput.instances.delete(this.id);
    } else {
      delete intlTelInput.instances[this.id];
    }
  }
  // check if the instance is still valid (not destroyed/unmounted)
  isActive() {
    var _b;
    return !!((_b = __classPrivateFieldGet3(this, _Iti_ui, "f")) === null || _b === void 0 ? void 0 : _b.telInput);
  }
  //* Get the extension from the current number.
  getExtension() {
    if (intlTelInput.utils && __classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      return intlTelInput.utils.getExtension(__classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getFullNumber).call(this), __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2);
    }
    return "";
  }
  //* Format the number to the given format.
  getNumber(format) {
    if (intlTelInput.utils && __classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      const { iso2 } = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f");
      const fullNumber = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getFullNumber).call(this);
      const formattedNumber = intlTelInput.utils.formatNumber(fullNumber, iso2, format);
      const currentVal = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.value;
      return __classPrivateFieldGet3(this, _Iti_numerals, "f").denormalise(formattedNumber, currentVal);
    }
    return "";
  }
  //* Get the type of the entered number e.g. landline/mobile.
  getNumberType() {
    if (intlTelInput.utils && __classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      return intlTelInput.utils.getNumberType(__classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getFullNumber).call(this), __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2);
    }
    return SENTINELS.UNKNOWN_NUMBER_TYPE;
  }
  //* Get the country data for the currently selected country.
  getSelectedCountryData() {
    return __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f");
  }
  //* Get the validation error.
  getValidationError() {
    if (intlTelInput.utils && __classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      const { iso2 } = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f");
      return intlTelInput.utils.getValidationError(__classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getFullNumber).call(this), iso2);
    }
    return SENTINELS.UNKNOWN_VALIDATION_ERROR;
  }
  //* Validate the input val using number length only
  isValidNumber() {
    const { dialCode, iso2 } = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f");
    if (intlTelInput.utils && __classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      const number = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getFullNumber).call(this);
      const coreNumber = intlTelInput.utils.getCoreNumber(number, iso2);
      if (coreNumber) {
        if (dialCode === UK.DIAL_CODE) {
          if (coreNumber[0] === UK.MOBILE_PREFIX && coreNumber.length !== UK.MOBILE_CORE_LENGTH) {
            return false;
          }
        }
        const hasAlphaChar = REGEX.ALPHA_UNICODE.test(number);
        if (!hasAlphaChar && dialCode) {
          const nationalPortion = number.startsWith("+") ? number.slice(1 + dialCode.length) : number;
          const nationalDigitCount = getNumeric(nationalPortion).length;
          if (coreNumber.length > nationalDigitCount) {
            return false;
          }
        }
      }
    }
    return __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_validateNumber).call(this, false);
  }
  //* Validate the input val with precise validation
  isValidNumberPrecise() {
    return __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_validateNumber).call(this, true);
  }
  //* Update the selected country, and update the input val accordingly.
  setCountry(iso2) {
    if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      return;
    }
    const iso2Lower = iso2 === null || iso2 === void 0 ? void 0 : iso2.toLowerCase();
    if (!isIso22(iso2Lower)) {
      throw new Error(`Invalid country code: '${iso2Lower}'`);
    }
    const currentCountry = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2;
    const isCountryChange = iso2 && iso2Lower !== currentCountry || !iso2 && currentCountry;
    if (isCountryChange) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setCountry).call(this, iso2Lower);
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateDialCode).call(this, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").dialCode);
      if (__classPrivateFieldGet3(this, _Iti_options, "f").formatOnDisplay) {
        const inputValue = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
        __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateValFromNumber).call(this, inputValue);
      }
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_triggerCountryChange).call(this);
    }
  }
  //* Set the input value and update the country.
  setNumber(number) {
    if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      return;
    }
    const normalisedNumber = __classPrivateFieldGet3(this, _Iti_numerals, "f").normalise(number);
    const countryChanged = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateCountryFromNumber).call(this, normalisedNumber);
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateValFromNumber).call(this, normalisedNumber);
    if (countryChanged) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_triggerCountryChange).call(this);
    }
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_trigger).call(this, EVENTS.INPUT, { isSetNumber: true });
  }
  //* Set the placeholder number typ
  setPlaceholderNumberType(type) {
    if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      return;
    }
    __classPrivateFieldGet3(this, _Iti_options, "f").placeholderNumberType = type;
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updatePlaceholder).call(this);
  }
  setDisabled(disabled) {
    if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
      return;
    }
    __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.disabled = disabled;
    if (disabled) {
      __classPrivateFieldGet3(this, _Iti_ui, "f").selectedCountry.setAttribute("disabled", "true");
    } else {
      __classPrivateFieldGet3(this, _Iti_ui, "f").selectedCountry.removeAttribute("disabled");
    }
  }
  //********************
  //*  STATIC METHODS
  //********************
  // Internal instance notification used by utils/geoip loaders.
  // Kept public so module-level helpers (e.g. attachUtils) can call it, while still allowing
  // access to private instance methods.
  static forEachInstance(method, ...args) {
    const instances = intlTelInput.instances;
    const values = instances instanceof Map ? Array.from(instances.values()) : Object.values(instances);
    const arg = args[0];
    values.forEach((instance) => {
      if (!(instance instanceof _a2)) {
        return;
      }
      switch (method) {
        case "handleUtils":
          __classPrivateFieldGet3(instance, _Iti_instances, "m", _Iti_handleUtils).call(instance);
          break;
        case "handleUtilsFailure":
          __classPrivateFieldGet3(instance, _Iti_instances, "m", _Iti_handleUtilsFailure).call(instance, arg);
          break;
        case "handleAutoCountry":
          __classPrivateFieldGet3(instance, _Iti_instances, "m", _Iti_handleAutoCountry).call(instance);
          break;
        case "handleAutoCountryFailure":
          __classPrivateFieldGet3(instance, _Iti_instances, "m", _Iti_handleAutoCountryFailure).call(instance);
          break;
      }
    });
  }
};
_a2 = Iti, _Iti_ui = /* @__PURE__ */ new WeakMap(), _Iti_options = /* @__PURE__ */ new WeakMap(), _Iti_isAndroid = /* @__PURE__ */ new WeakMap(), _Iti_countries = /* @__PURE__ */ new WeakMap(), _Iti_dialCodeMaxLen = /* @__PURE__ */ new WeakMap(), _Iti_dialCodeToIso2Map = /* @__PURE__ */ new WeakMap(), _Iti_dialCodes = /* @__PURE__ */ new WeakMap(), _Iti_countryByIso2 = /* @__PURE__ */ new WeakMap(), _Iti_selectedCountryData = /* @__PURE__ */ new WeakMap(), _Iti_maxCoreNumberLength = /* @__PURE__ */ new WeakMap(), _Iti_defaultCountry = /* @__PURE__ */ new WeakMap(), _Iti_abortController = /* @__PURE__ */ new WeakMap(), _Iti_dropdownAbortController = /* @__PURE__ */ new WeakMap(), _Iti_numerals = /* @__PURE__ */ new WeakMap(), _Iti_resolveAutoCountryPromise = /* @__PURE__ */ new WeakMap(), _Iti_rejectAutoCountryPromise = /* @__PURE__ */ new WeakMap(), _Iti_resolveUtilsScriptPromise = /* @__PURE__ */ new WeakMap(), _Iti_rejectUtilsScriptPromise = /* @__PURE__ */ new WeakMap(), _Iti_instances = /* @__PURE__ */ new WeakSet(), _Iti_getTelInputValue = function _Iti_getTelInputValue2() {
  const inputValue = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.value.trim();
  return __classPrivateFieldGet3(this, _Iti_numerals, "f").normalise(inputValue);
}, _Iti_setTelInputValue = function _Iti_setTelInputValue2(asciiValue) {
  const currentValue = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.value;
  __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.value = __classPrivateFieldGet3(this, _Iti_numerals, "f").denormalise(asciiValue, currentValue);
}, _Iti_createInitPromises = function _Iti_createInitPromises2(options) {
  const { initialCountry, geoIpLookup, loadUtils } = options;
  const needsAutoCountryPromise = initialCountry === INITIAL_COUNTRY.AUTO && Boolean(geoIpLookup);
  const needsUtilsScriptPromise = Boolean(loadUtils) && !intlTelInput.utils;
  let autoCountryPromise;
  if (needsAutoCountryPromise) {
    autoCountryPromise = new Promise((resolve, reject) => {
      __classPrivateFieldSet3(this, _Iti_resolveAutoCountryPromise, resolve, "f");
      __classPrivateFieldSet3(this, _Iti_rejectAutoCountryPromise, reject, "f");
    });
  } else {
    autoCountryPromise = Promise.resolve(void 0);
    __classPrivateFieldSet3(this, _Iti_resolveAutoCountryPromise, () => {
    }, "f");
    __classPrivateFieldSet3(this, _Iti_rejectAutoCountryPromise, () => {
    }, "f");
  }
  let utilsScriptPromise;
  if (needsUtilsScriptPromise) {
    utilsScriptPromise = new Promise((resolve, reject) => {
      __classPrivateFieldSet3(this, _Iti_resolveUtilsScriptPromise, resolve, "f");
      __classPrivateFieldSet3(this, _Iti_rejectUtilsScriptPromise, reject, "f");
    });
  } else {
    utilsScriptPromise = Promise.resolve(void 0);
    __classPrivateFieldSet3(this, _Iti_resolveUtilsScriptPromise, () => {
    }, "f");
    __classPrivateFieldSet3(this, _Iti_rejectUtilsScriptPromise, () => {
    }, "f");
  }
  return Promise.all([autoCountryPromise, utilsScriptPromise]);
}, _Iti_init = function _Iti_init2() {
  __classPrivateFieldSet3(this, _Iti_selectedCountryData, {}, "f");
  __classPrivateFieldSet3(this, _Iti_abortController, new AbortController(), "f");
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_processCountryData).call(this);
  __classPrivateFieldGet3(this, _Iti_ui, "f").generateMarkup(__classPrivateFieldGet3(this, _Iti_countries, "f"));
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setInitialState).call(this);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_initListeners).call(this);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_initRequests).call(this);
  if (__classPrivateFieldGet3(this, _Iti_options, "f").dropdownAlwaysOpen) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_openDropdown).call(this);
  }
}, _Iti_processCountryData = function _Iti_processCountryData2() {
  generateCountryNames(__classPrivateFieldGet3(this, _Iti_countries, "f"), __classPrivateFieldGet3(this, _Iti_options, "f"));
  sortCountries(__classPrivateFieldGet3(this, _Iti_countries, "f"), __classPrivateFieldGet3(this, _Iti_options, "f"));
  cacheSearchTokens(__classPrivateFieldGet3(this, _Iti_countries, "f"));
}, _Iti_setInitialState = function _Iti_setInitialState2(overrideAutoCountry = false) {
  const attributeValueRaw = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.getAttribute("value");
  const attributeValue = __classPrivateFieldGet3(this, _Iti_numerals, "f").normalise(attributeValueRaw);
  const inputValue = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
  const useAttribute = attributeValue && attributeValue.startsWith("+") && (!inputValue || !inputValue.startsWith("+"));
  const val = useAttribute ? attributeValue : inputValue;
  const dialCode = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getDialCode).call(this, val);
  const isRegionlessNanpNumber = isRegionlessNanp(val);
  const { initialCountry, geoIpLookup } = __classPrivateFieldGet3(this, _Iti_options, "f");
  const isAutoCountry = initialCountry === INITIAL_COUNTRY.AUTO && geoIpLookup;
  const doingAutoCountryLookup = isAutoCountry && !overrideAutoCountry;
  const initialCountryLower = initialCountry.toLowerCase();
  const isValidInitialCountry = isIso22(initialCountryLower);
  if (dialCode) {
    if (isRegionlessNanpNumber) {
      if (isValidInitialCountry) {
        __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setCountry).call(this, initialCountryLower);
      } else if (!doingAutoCountryLookup) {
        __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setCountry).call(this, US.ISO2);
      }
    } else {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateCountryFromNumber).call(this, val);
    }
  } else if (isValidInitialCountry) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setCountry).call(this, initialCountryLower);
  } else if (!doingAutoCountryLookup) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setCountry).call(this, "");
  }
  if (val) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateValFromNumber).call(this, val);
  }
}, _Iti_initListeners = function _Iti_initListeners2() {
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_initTelInputListeners).call(this);
  if (__classPrivateFieldGet3(this, _Iti_options, "f").allowDropdown) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_initDropdownListeners).call(this);
  }
  if ((__classPrivateFieldGet3(this, _Iti_ui, "f").hiddenInput || __classPrivateFieldGet3(this, _Iti_ui, "f").hiddenInputCountry) && __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.form) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_initHiddenInputListener).call(this);
  }
}, _Iti_initHiddenInputListener = function _Iti_initHiddenInputListener2() {
  var _b;
  const handleHiddenInputSubmit = () => {
    if (__classPrivateFieldGet3(this, _Iti_ui, "f").hiddenInput) {
      __classPrivateFieldGet3(this, _Iti_ui, "f").hiddenInput.value = this.getNumber();
    }
    if (__classPrivateFieldGet3(this, _Iti_ui, "f").hiddenInputCountry) {
      __classPrivateFieldGet3(this, _Iti_ui, "f").hiddenInputCountry.value = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2 || "";
    }
  };
  (_b = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.form) === null || _b === void 0 ? void 0 : _b.addEventListener("submit", handleHiddenInputSubmit, {
    signal: __classPrivateFieldGet3(this, _Iti_abortController, "f").signal
  });
}, _Iti_initDropdownListeners = function _Iti_initDropdownListeners2() {
  const signal = __classPrivateFieldGet3(this, _Iti_abortController, "f").signal;
  const handleLabelClick = (e) => {
    if (__classPrivateFieldGet3(this, _Iti_ui, "f").isDropdownClosed()) {
      __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.focus();
    } else {
      e.preventDefault();
    }
  };
  const label = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.closest("label");
  if (label) {
    label.addEventListener("click", handleLabelClick, { signal });
  }
  const handleClickSelectedCountry = () => {
    if (__classPrivateFieldGet3(this, _Iti_ui, "f").isDropdownClosed() && !__classPrivateFieldGet3(this, _Iti_ui, "f").telInput.disabled && !__classPrivateFieldGet3(this, _Iti_ui, "f").telInput.readOnly) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_openDropdown).call(this);
    }
  };
  __classPrivateFieldGet3(this, _Iti_ui, "f").selectedCountry.addEventListener("click", handleClickSelectedCountry, {
    signal
  });
  const handleCountryContainerKeydown = (e) => {
    const allowedKeys = [KEYS.ARROW_UP, KEYS.ARROW_DOWN, KEYS.SPACE, KEYS.ENTER];
    if (__classPrivateFieldGet3(this, _Iti_ui, "f").isDropdownClosed() && allowedKeys.includes(e.key)) {
      e.preventDefault();
      e.stopPropagation();
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_openDropdown).call(this);
    }
    if (e.key === KEYS.TAB) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_closeDropdown).call(this);
    }
  };
  __classPrivateFieldGet3(this, _Iti_ui, "f").countryContainer.addEventListener("keydown", handleCountryContainerKeydown, { signal });
}, _Iti_initRequests = function _Iti_initRequests2() {
  const { loadUtils, initialCountry, geoIpLookup } = __classPrivateFieldGet3(this, _Iti_options, "f");
  if (loadUtils && !intlTelInput.utils) {
    const doAttachUtils = () => {
      var _b;
      (_b = intlTelInput.attachUtils(loadUtils)) === null || _b === void 0 ? void 0 : _b.catch(() => {
      });
    };
    if (intlTelInput.documentReady()) {
      doAttachUtils();
    } else {
      const handlePageLoad = () => {
        doAttachUtils();
      };
      window.addEventListener("load", handlePageLoad, {
        signal: __classPrivateFieldGet3(this, _Iti_abortController, "f").signal
      });
    }
  } else {
    __classPrivateFieldGet3(this, _Iti_resolveUtilsScriptPromise, "f").call(this);
  }
  const isAutoCountry = initialCountry === INITIAL_COUNTRY.AUTO && geoIpLookup;
  if (isAutoCountry) {
    if (__classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2) {
      __classPrivateFieldGet3(this, _Iti_resolveAutoCountryPromise, "f").call(this);
    } else {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_loadAutoCountry).call(this);
    }
  }
}, _Iti_loadAutoCountry = function _Iti_loadAutoCountry2() {
  if (intlTelInput.autoCountry) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_handleAutoCountry).call(this);
  } else {
    __classPrivateFieldGet3(this, _Iti_ui, "f").selectedCountryInner.classList.add(CLASSES.LOADING);
    if (!intlTelInput.startedLoadingAutoCountry) {
      intlTelInput.startedLoadingAutoCountry = true;
      if (typeof __classPrivateFieldGet3(this, _Iti_options, "f").geoIpLookup === "function") {
        const successCallback = (iso2 = "") => {
          __classPrivateFieldGet3(this, _Iti_ui, "f").selectedCountryInner.classList.remove(CLASSES.LOADING);
          const iso2Lower = iso2.toLowerCase();
          if (isIso22(iso2Lower)) {
            intlTelInput.autoCountry = iso2Lower;
            setTimeout(() => _a2.forEachInstance("handleAutoCountry"));
          } else {
            _a2.forEachInstance("handleAutoCountryFailure");
          }
        };
        const failureCallback = () => {
          __classPrivateFieldGet3(this, _Iti_ui, "f").selectedCountryInner.classList.remove(CLASSES.LOADING);
          _a2.forEachInstance("handleAutoCountryFailure");
        };
        __classPrivateFieldGet3(this, _Iti_options, "f").geoIpLookup(successCallback, failureCallback);
      }
    }
  }
}, _Iti_openDropdownWithPlus = function _Iti_openDropdownWithPlus2() {
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_openDropdown).call(this);
  __classPrivateFieldGet3(this, _Iti_ui, "f").searchInput.value = "+";
  __classPrivateFieldGet3(this, _Iti_ui, "f").filterCountriesByQuery("");
}, _Iti_initTelInputListeners = function _Iti_initTelInputListeners2() {
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_bindInputListener).call(this);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_maybeBindKeydownListener).call(this);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_maybeBindPasteListener).call(this);
}, _Iti_bindInputListener = function _Iti_bindInputListener2() {
  const { strictMode, formatAsYouType, separateDialCode, allowDropdown, countrySearch } = __classPrivateFieldGet3(this, _Iti_options, "f");
  let userOverrideFormatting = false;
  if (REGEX.ALPHA_UNICODE.test(__classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this))) {
    userOverrideFormatting = true;
  }
  const handleInputEvent = (e) => {
    const inputValue = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
    if (__classPrivateFieldGet3(this, _Iti_isAndroid, "f") && (e === null || e === void 0 ? void 0 : e.data) === "+" && separateDialCode && allowDropdown && countrySearch) {
      const currentCaretPos = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.selectionStart || 0;
      const valueBeforeCaret = inputValue.substring(0, currentCaretPos - 1);
      const valueAfterCaret = inputValue.substring(currentCaretPos);
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setTelInputValue).call(this, valueBeforeCaret + valueAfterCaret);
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_openDropdownWithPlus).call(this);
      return;
    }
    if (__classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateCountryFromNumber).call(this, inputValue)) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_triggerCountryChange).call(this);
    }
    const isFormattingChar = (e === null || e === void 0 ? void 0 : e.data) && REGEX.NON_PLUS_NUMERIC.test(e.data);
    const isPaste = (e === null || e === void 0 ? void 0 : e.inputType) === INPUT_TYPES.PASTE && inputValue;
    if (isFormattingChar || isPaste && !strictMode) {
      userOverrideFormatting = true;
    } else if (!REGEX.NON_PLUS_NUMERIC.test(inputValue)) {
      userOverrideFormatting = false;
    }
    const isSetNumber = (e === null || e === void 0 ? void 0 : e.detail) && e.detail["isSetNumber"];
    const isAscii = __classPrivateFieldGet3(this, _Iti_numerals, "f").isAscii();
    if (formatAsYouType && !userOverrideFormatting && !isSetNumber && isAscii) {
      const currentCaretPos = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.selectionStart || 0;
      const valueBeforeCaret = inputValue.substring(0, currentCaretPos);
      const relevantCharsBeforeCaret = valueBeforeCaret.replace(REGEX.NON_PLUS_NUMERIC_GLOBAL, "").length;
      const isDeleteForwards = (e === null || e === void 0 ? void 0 : e.inputType) === INPUT_TYPES.DELETE_FWD;
      const fullNumber = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getFullNumber).call(this);
      const formattedValue = formatNumberAsYouType(fullNumber, inputValue, intlTelInput.utils, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f"), separateDialCode);
      const newCaretPos = translateCursorPosition(relevantCharsBeforeCaret, formattedValue, currentCaretPos, isDeleteForwards);
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setTelInputValue).call(this, formattedValue);
      __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.setSelectionRange(newCaretPos, newCaretPos);
    }
    if (separateDialCode && inputValue.startsWith("+") && __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").dialCode) {
      const cleanNumber = beforeSetNumber(inputValue, true, separateDialCode, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f"));
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setTelInputValue).call(this, cleanNumber);
    }
  };
  __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.addEventListener("input", handleInputEvent, {
    signal: __classPrivateFieldGet3(this, _Iti_abortController, "f").signal
  });
}, _Iti_maybeBindKeydownListener = function _Iti_maybeBindKeydownListener2() {
  const { strictMode, separateDialCode, allowDropdown, countrySearch } = __classPrivateFieldGet3(this, _Iti_options, "f");
  if (strictMode || separateDialCode) {
    const handleKeydownEvent = (e) => {
      if (e.key && e.key.length === 1 && !e.altKey && !e.ctrlKey && !e.metaKey) {
        if (separateDialCode && allowDropdown && countrySearch && e.key === "+") {
          e.preventDefault();
          __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_openDropdownWithPlus).call(this);
          return;
        }
        if (strictMode) {
          const inputValue = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
          const alreadyHasPlus = inputValue.startsWith("+");
          const isInitialPlus = !alreadyHasPlus && __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.selectionStart === 0 && e.key === "+";
          const normalisedKey = __classPrivateFieldGet3(this, _Iti_numerals, "f").normalise(e.key);
          const isNumeric = /^[0-9]$/.test(normalisedKey);
          const isAllowedChar = separateDialCode ? isNumeric : isInitialPlus || isNumeric;
          const input = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput;
          const selStart = input.selectionStart;
          const selEnd = input.selectionEnd;
          const before = inputValue.slice(0, selStart);
          const after = inputValue.slice(selEnd);
          const newValue = before + e.key + after;
          const newFullNumber = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getFullNumber).call(this, newValue);
          const coreNumber = intlTelInput.utils.getCoreNumber(newFullNumber, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2);
          const hasExceededMaxLength = __classPrivateFieldGet3(this, _Iti_maxCoreNumberLength, "f") && coreNumber.length > __classPrivateFieldGet3(this, _Iti_maxCoreNumberLength, "f");
          const newCountry = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getNewCountryFromNumber).call(this, newFullNumber);
          const isChangingDialCode = newCountry !== null;
          if (!isAllowedChar || hasExceededMaxLength && !isChangingDialCode && !isInitialPlus) {
            e.preventDefault();
          }
        }
      }
    };
    __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.addEventListener("keydown", handleKeydownEvent, {
      signal: __classPrivateFieldGet3(this, _Iti_abortController, "f").signal
    });
  }
}, _Iti_maybeBindPasteListener = function _Iti_maybeBindPasteListener2() {
  if (__classPrivateFieldGet3(this, _Iti_options, "f").strictMode) {
    const handlePasteEvent = (e) => {
      e.preventDefault();
      const input = __classPrivateFieldGet3(this, _Iti_ui, "f").telInput;
      const selStart = input.selectionStart;
      const selEnd = input.selectionEnd;
      const inputValue = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
      const before = inputValue.slice(0, selStart);
      const after = inputValue.slice(selEnd);
      const iso2 = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2;
      const pastedRaw = e.clipboardData.getData("text");
      const pasted = __classPrivateFieldGet3(this, _Iti_numerals, "f").normalise(pastedRaw);
      const initialCharSelected = selStart === 0 && selEnd > 0;
      const allowLeadingPlus = !inputValue.startsWith("+") || initialCharSelected;
      const allowedChars = pasted.replace(REGEX.NON_PLUS_NUMERIC_GLOBAL, "");
      const hasLeadingPlus = allowedChars.startsWith("+");
      const numerics = allowedChars.replace(/\+/g, "");
      const sanitised = hasLeadingPlus && allowLeadingPlus ? `+${numerics}` : numerics;
      let newVal = before + sanitised + after;
      if (newVal.length > 5) {
        let coreNumber = intlTelInput.utils.getCoreNumber(newVal, iso2);
        while (coreNumber.length === 0 && newVal.length > 0) {
          newVal = newVal.slice(0, -1);
          coreNumber = intlTelInput.utils.getCoreNumber(newVal, iso2);
        }
        if (!coreNumber) {
          return;
        }
        if (__classPrivateFieldGet3(this, _Iti_maxCoreNumberLength, "f") && coreNumber.length > __classPrivateFieldGet3(this, _Iti_maxCoreNumberLength, "f")) {
          if (input.selectionEnd === inputValue.length) {
            const trimLength = coreNumber.length - __classPrivateFieldGet3(this, _Iti_maxCoreNumberLength, "f");
            newVal = newVal.slice(0, newVal.length - trimLength);
          } else {
            return;
          }
        }
      }
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setTelInputValue).call(this, newVal);
      const caretPos = selStart + sanitised.length;
      input.setSelectionRange(caretPos, caretPos);
      input.dispatchEvent(new InputEvent("input", { bubbles: true }));
    };
    __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.addEventListener("paste", handlePasteEvent, {
      signal: __classPrivateFieldGet3(this, _Iti_abortController, "f").signal
    });
  }
}, _Iti_cap = function _Iti_cap2(number) {
  const max = Number(__classPrivateFieldGet3(this, _Iti_ui, "f").telInput.getAttribute("maxlength"));
  return max && number.length > max ? number.substring(0, max) : number;
}, _Iti_trigger = function _Iti_trigger2(name, detailProps = {}) {
  const e = new CustomEvent(name, {
    bubbles: true,
    cancelable: true,
    detail: detailProps
  });
  __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.dispatchEvent(e);
}, _Iti_openDropdown = function _Iti_openDropdown2() {
  const { dropdownContainer, useFullscreenPopup } = __classPrivateFieldGet3(this, _Iti_options, "f");
  __classPrivateFieldSet3(this, _Iti_dropdownAbortController, new AbortController(), "f");
  __classPrivateFieldGet3(this, _Iti_ui, "f").openDropdown();
  if (!useFullscreenPopup && dropdownContainer) {
    const handleWindowScroll = () => __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_closeDropdown).call(this);
    window.addEventListener("scroll", handleWindowScroll, {
      signal: __classPrivateFieldGet3(this, _Iti_dropdownAbortController, "f").signal
    });
  }
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_bindDropdownListeners).call(this);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_trigger).call(this, EVENTS.OPEN_COUNTRY_DROPDOWN);
}, _Iti_bindDropdownListeners = function _Iti_bindDropdownListeners2() {
  const signal = __classPrivateFieldGet3(this, _Iti_dropdownAbortController, "f").signal;
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_bindDropdownMouseoverListener).call(this, signal);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_bindDropdownCountryClickListener).call(this, signal);
  if (!__classPrivateFieldGet3(this, _Iti_options, "f").dropdownAlwaysOpen) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_bindDropdownClickOffListener).call(this, signal);
  }
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_bindDropdownKeydownListener).call(this, signal);
  if (__classPrivateFieldGet3(this, _Iti_options, "f").countrySearch) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_bindDropdownSearchListeners).call(this, signal);
  }
}, _Iti_bindDropdownMouseoverListener = function _Iti_bindDropdownMouseoverListener2(signal) {
  const handleMouseoverCountryList = (e) => {
    var _b;
    const listItem = (_b = e.target) === null || _b === void 0 ? void 0 : _b.closest(`.${CLASSES.COUNTRY_ITEM}`);
    if (listItem) {
      __classPrivateFieldGet3(this, _Iti_ui, "f").highlightListItem(listItem, false);
    }
  };
  __classPrivateFieldGet3(this, _Iti_ui, "f").countryList.addEventListener("mouseover", handleMouseoverCountryList, {
    signal
  });
}, _Iti_bindDropdownCountryClickListener = function _Iti_bindDropdownCountryClickListener2(signal) {
  const handleClickCountryList = (e) => {
    var _b;
    const listItem = (_b = e.target) === null || _b === void 0 ? void 0 : _b.closest(`.${CLASSES.COUNTRY_ITEM}`);
    if (listItem) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_selectListItem).call(this, listItem);
    }
  };
  __classPrivateFieldGet3(this, _Iti_ui, "f").countryList.addEventListener("click", handleClickCountryList, {
    signal
  });
}, _Iti_bindDropdownClickOffListener = function _Iti_bindDropdownClickOffListener2(signal) {
  const handleClickOffToClose = (e) => {
    const target = e.target;
    const clickedInsideDropdown = !!target.closest(`#iti-${this.id}__dropdown-content`);
    if (!clickedInsideDropdown) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_closeDropdown).call(this);
    }
  };
  setTimeout(() => {
    document.documentElement.addEventListener("click", handleClickOffToClose, { signal });
  }, 0);
}, _Iti_bindDropdownKeydownListener = function _Iti_bindDropdownKeydownListener2(signal) {
  let query = "";
  let queryTimer = null;
  const handleKeydownOnDropdown = (e) => {
    const allowedKeys = [
      KEYS.ARROW_UP,
      KEYS.ARROW_DOWN,
      KEYS.ENTER,
      KEYS.ESC
    ];
    if (allowedKeys.includes(e.key)) {
      e.preventDefault();
      e.stopPropagation();
      if (e.key === KEYS.ARROW_UP || e.key === KEYS.ARROW_DOWN) {
        __classPrivateFieldGet3(this, _Iti_ui, "f").handleUpDownKey(e.key);
      } else if (e.key === KEYS.ENTER) {
        __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_handleEnterKey).call(this);
      } else if (e.key === KEYS.ESC) {
        __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_closeDropdown).call(this);
        __classPrivateFieldGet3(this, _Iti_ui, "f").selectedCountry.focus();
      }
    }
    if (!__classPrivateFieldGet3(this, _Iti_options, "f").countrySearch && REGEX.HIDDEN_SEARCH_CHAR.test(e.key)) {
      e.stopPropagation();
      if (queryTimer) {
        clearTimeout(queryTimer);
      }
      query += e.key.toLowerCase();
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_searchForCountry).call(this, query);
      queryTimer = setTimeout(() => {
        query = "";
      }, TIMINGS.HIDDEN_SEARCH_RESET_MS);
    }
  };
  document.addEventListener("keydown", handleKeydownOnDropdown, { signal });
}, _Iti_bindDropdownSearchListeners = function _Iti_bindDropdownSearchListeners2(signal) {
  __classPrivateFieldGet3(this, _Iti_ui, "f").searchInput.addEventListener("input", () => __classPrivateFieldGet3(this, _Iti_ui, "f").handleSearchChange(), { signal });
  __classPrivateFieldGet3(this, _Iti_ui, "f").searchClearButton.addEventListener("click", () => __classPrivateFieldGet3(this, _Iti_ui, "f").handleSearchClear(), { signal });
}, _Iti_searchForCountry = function _Iti_searchForCountry2(query) {
  const match = findFirstCountryStartingWith(__classPrivateFieldGet3(this, _Iti_countries, "f"), query);
  if (match) {
    const listItem = match.nodeById[this.id];
    __classPrivateFieldGet3(this, _Iti_ui, "f").highlightListItem(listItem, false);
    __classPrivateFieldGet3(this, _Iti_ui, "f").scrollTo(listItem);
  }
}, _Iti_handleEnterKey = function _Iti_handleEnterKey2() {
  if (__classPrivateFieldGet3(this, _Iti_ui, "f").highlightedItem) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_selectListItem).call(this, __classPrivateFieldGet3(this, _Iti_ui, "f").highlightedItem);
  }
}, _Iti_updateValFromNumber = function _Iti_updateValFromNumber2(fullNumber) {
  const { formatOnDisplay, nationalMode, separateDialCode } = __classPrivateFieldGet3(this, _Iti_options, "f");
  let number = fullNumber;
  if (formatOnDisplay && intlTelInput.utils && __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f")) {
    const isRegionless = hasRegionlessDialCode(fullNumber);
    const useNational = nationalMode && !isRegionless || !number.startsWith("+") && !separateDialCode;
    const { NATIONAL, INTERNATIONAL } = intlTelInput.utils.numberFormat;
    const format = useNational ? NATIONAL : INTERNATIONAL;
    number = intlTelInput.utils.formatNumber(number, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2, format);
  }
  number = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_beforeSetNumber).call(this, number);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setTelInputValue).call(this, number);
}, _Iti_updateCountryFromNumber = function _Iti_updateCountryFromNumber2(fullNumber) {
  const iso2 = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getNewCountryFromNumber).call(this, fullNumber);
  if (iso2 !== null) {
    return __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setCountry).call(this, iso2);
  }
  return false;
}, _Iti_ensureHasDialCode = function _Iti_ensureHasDialCode2(number) {
  const { dialCode, nationalPrefix } = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f");
  const alreadyHasPlus = number.startsWith("+");
  if (alreadyHasPlus || !dialCode) {
    return number;
  }
  const hasPrefix = nationalPrefix && number.startsWith(nationalPrefix) && !__classPrivateFieldGet3(this, _Iti_options, "f").separateDialCode;
  const cleanNumber = hasPrefix ? number.substring(1) : number;
  return `+${dialCode}${cleanNumber}`;
}, _Iti_getNewCountryFromNumber = function _Iti_getNewCountryFromNumber2(fullNumber) {
  const plusIndex = fullNumber.indexOf("+");
  let number = plusIndex ? fullNumber.substring(plusIndex) : fullNumber;
  const selectedIso2 = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2;
  const selectedDialCode = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").dialCode;
  number = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_ensureHasDialCode).call(this, number);
  const dialCodeMatch = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getDialCode).call(this, number, true);
  const numeric = getNumeric(number);
  if (dialCodeMatch) {
    const dialCodeMatchNumeric = getNumeric(dialCodeMatch);
    const iso2Codes = __classPrivateFieldGet3(this, _Iti_dialCodeToIso2Map, "f")[dialCodeMatchNumeric];
    if (iso2Codes.length === 1) {
      if (iso2Codes[0] === selectedIso2) {
        return null;
      }
      return iso2Codes[0];
    }
    if (!selectedIso2 && __classPrivateFieldGet3(this, _Iti_defaultCountry, "f") && iso2Codes.includes(__classPrivateFieldGet3(this, _Iti_defaultCountry, "f"))) {
      return __classPrivateFieldGet3(this, _Iti_defaultCountry, "f");
    }
    const isRegionlessNanpNumber = selectedDialCode === DIAL.NANP && isRegionlessNanp(numeric);
    if (isRegionlessNanpNumber) {
      return null;
    }
    const { areaCodes, priority } = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f");
    if (areaCodes) {
      const dialCodeAreaCodes = areaCodes.map((areaCode) => `${selectedDialCode}${areaCode}`);
      for (const dialCodeAreaCode of dialCodeAreaCodes) {
        if (numeric.startsWith(dialCodeAreaCode)) {
          return null;
        }
      }
    }
    const isMainCountry = priority === 0;
    const hasAreaCodesButNoneMatched = areaCodes && !isMainCountry && numeric.length > dialCodeMatchNumeric.length;
    const isValidSelection = selectedIso2 && iso2Codes.includes(selectedIso2) && !hasAreaCodesButNoneMatched;
    const alreadySelected = selectedIso2 === iso2Codes[0];
    if (!isValidSelection && !alreadySelected) {
      return iso2Codes[0];
    }
  } else if (number.startsWith("+") && numeric.length) {
    const currentDial = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").dialCode || "";
    if (currentDial && currentDial.startsWith(numeric)) {
      return null;
    }
    return "";
  } else if ((!number || number === "+") && !selectedIso2 && __classPrivateFieldGet3(this, _Iti_defaultCountry, "f")) {
    return __classPrivateFieldGet3(this, _Iti_defaultCountry, "f");
  }
  return null;
}, _Iti_setCountry = function _Iti_setCountry2(iso2) {
  const prevIso2 = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2 || "";
  __classPrivateFieldSet3(this, _Iti_selectedCountryData, iso2 ? __classPrivateFieldGet3(this, _Iti_countryByIso2, "f").get(iso2) : {}, "f");
  if (__classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2) {
    __classPrivateFieldSet3(this, _Iti_defaultCountry, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2, "f");
  }
  __classPrivateFieldGet3(this, _Iti_ui, "f").setCountry(__classPrivateFieldGet3(this, _Iti_selectedCountryData, "f"));
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updatePlaceholder).call(this);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateMaxLength).call(this);
  return prevIso2 !== iso2;
}, _Iti_updateMaxLength = function _Iti_updateMaxLength2() {
  const { strictMode, placeholderNumberType, allowedNumberTypes } = __classPrivateFieldGet3(this, _Iti_options, "f");
  const { iso2 } = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f");
  if (strictMode && intlTelInput.utils) {
    if (iso2) {
      const numberType = intlTelInput.utils.numberType[placeholderNumberType];
      let exampleNumber = intlTelInput.utils.getExampleNumber(iso2, false, numberType, true);
      let validNumber = exampleNumber;
      while (intlTelInput.utils.isPossibleNumber(exampleNumber, iso2, allowedNumberTypes)) {
        validNumber = exampleNumber;
        exampleNumber += "0";
      }
      const coreNumber = intlTelInput.utils.getCoreNumber(validNumber, iso2);
      __classPrivateFieldSet3(this, _Iti_maxCoreNumberLength, coreNumber.length, "f");
      if (iso2 === "by") {
        __classPrivateFieldSet3(this, _Iti_maxCoreNumberLength, coreNumber.length + 1, "f");
      }
    } else {
      __classPrivateFieldSet3(this, _Iti_maxCoreNumberLength, null, "f");
    }
  }
}, _Iti_updatePlaceholder = function _Iti_updatePlaceholder2() {
  const { autoPlaceholder, placeholderNumberType, nationalMode, customPlaceholder } = __classPrivateFieldGet3(this, _Iti_options, "f");
  const shouldSetPlaceholder = autoPlaceholder === PLACEHOLDER_MODES.AGGRESSIVE || !__classPrivateFieldGet3(this, _Iti_ui, "f").hadInitialPlaceholder && autoPlaceholder === PLACEHOLDER_MODES.POLITE;
  if (intlTelInput.utils && shouldSetPlaceholder) {
    const numberType = intlTelInput.utils.numberType[placeholderNumberType];
    let placeholder = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2 ? intlTelInput.utils.getExampleNumber(__classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2, nationalMode, numberType) : "";
    placeholder = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_beforeSetNumber).call(this, placeholder);
    if (typeof customPlaceholder === "function") {
      placeholder = customPlaceholder(placeholder, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f"));
    }
    __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.setAttribute("placeholder", placeholder);
  }
}, _Iti_selectListItem = function _Iti_selectListItem2(listItem) {
  const iso2 = listItem.dataset[DATA_KEYS.COUNTRY_CODE];
  const countryChanged = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setCountry).call(this, iso2);
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_closeDropdown).call(this);
  const dialCode = listItem.dataset[DATA_KEYS.DIAL_CODE];
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateDialCode).call(this, dialCode);
  if (__classPrivateFieldGet3(this, _Iti_options, "f").formatOnDisplay) {
    const inputValue = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateValFromNumber).call(this, inputValue);
  }
  __classPrivateFieldGet3(this, _Iti_ui, "f").telInput.focus();
  if (countryChanged) {
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_triggerCountryChange).call(this);
  }
}, _Iti_closeDropdown = function _Iti_closeDropdown2(isDestroy) {
  if (__classPrivateFieldGet3(this, _Iti_ui, "f").isDropdownClosed() || __classPrivateFieldGet3(this, _Iti_options, "f").dropdownAlwaysOpen && !isDestroy) {
    return;
  }
  __classPrivateFieldGet3(this, _Iti_ui, "f").closeDropdown();
  __classPrivateFieldGet3(this, _Iti_dropdownAbortController, "f").abort();
  __classPrivateFieldSet3(this, _Iti_dropdownAbortController, null, "f");
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_trigger).call(this, EVENTS.CLOSE_COUNTRY_DROPDOWN);
}, _Iti_updateDialCode = function _Iti_updateDialCode2(newDialCodeBare) {
  const inputVal = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
  const newDialCode = `+${newDialCodeBare}`;
  let newNumber;
  if (inputVal.startsWith("+")) {
    const prevDialCode = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getDialCode).call(this, inputVal);
    if (prevDialCode) {
      newNumber = inputVal.replace(prevDialCode, newDialCode);
    } else {
      newNumber = newDialCode;
    }
    __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setTelInputValue).call(this, newNumber);
  }
}, _Iti_getDialCode = function _Iti_getDialCode2(number, includeAreaCode) {
  let dialCode = "";
  if (number.startsWith("+")) {
    let numericChars = "";
    let foundBaseDialCode = false;
    for (let i = 0; i < number.length; i++) {
      const c = number.charAt(i);
      if (/[0-9]/.test(c)) {
        numericChars += c;
        const hasMapEntry = Boolean(__classPrivateFieldGet3(this, _Iti_dialCodeToIso2Map, "f")[numericChars]);
        if (!hasMapEntry) {
          break;
        }
        if (__classPrivateFieldGet3(this, _Iti_dialCodes, "f").has(numericChars)) {
          dialCode = number.substring(0, i + 1);
          foundBaseDialCode = true;
          if (!includeAreaCode) {
            break;
          }
        } else if (includeAreaCode && foundBaseDialCode) {
          dialCode = number.substring(0, i + 1);
        }
        if (numericChars.length === __classPrivateFieldGet3(this, _Iti_dialCodeMaxLen, "f")) {
          break;
        }
      }
    }
  }
  return dialCode;
}, _Iti_getFullNumber = function _Iti_getFullNumber2(overrideVal) {
  const val = overrideVal ? __classPrivateFieldGet3(this, _Iti_numerals, "f").normalise(overrideVal) : __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
  const { dialCode } = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f");
  let prefix;
  const numericVal = getNumeric(val);
  if (__classPrivateFieldGet3(this, _Iti_options, "f").separateDialCode && !val.startsWith("+") && dialCode && numericVal) {
    prefix = `+${dialCode}`;
  } else {
    prefix = "";
  }
  return prefix + val;
}, _Iti_beforeSetNumber = function _Iti_beforeSetNumber2(fullNumber) {
  const hasValidDialCode = Boolean(__classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getDialCode).call(this, fullNumber));
  const number = beforeSetNumber(fullNumber, hasValidDialCode, __classPrivateFieldGet3(this, _Iti_options, "f").separateDialCode, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f"));
  return __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_cap).call(this, number);
}, _Iti_triggerCountryChange = function _Iti_triggerCountryChange2() {
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_trigger).call(this, EVENTS.COUNTRY_CHANGE);
}, _Iti_handleAutoCountry = function _Iti_handleAutoCountry2() {
  var _b;
  if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
    (_b = __classPrivateFieldGet3(this, _Iti_resolveAutoCountryPromise, "f")) === null || _b === void 0 ? void 0 : _b.call(this);
    return;
  }
  if (__classPrivateFieldGet3(this, _Iti_options, "f").initialCountry === INITIAL_COUNTRY.AUTO && intlTelInput.autoCountry) {
    __classPrivateFieldSet3(this, _Iti_defaultCountry, intlTelInput.autoCountry, "f");
    const hasSelectedCountryOrGlobe = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2 || __classPrivateFieldGet3(this, _Iti_ui, "f").selectedCountryInner.classList.contains(CLASSES.GLOBE);
    if (!hasSelectedCountryOrGlobe) {
      this.setCountry(__classPrivateFieldGet3(this, _Iti_defaultCountry, "f"));
    }
    __classPrivateFieldGet3(this, _Iti_resolveAutoCountryPromise, "f").call(this);
  }
}, _Iti_handleAutoCountryFailure = function _Iti_handleAutoCountryFailure2() {
  var _b;
  if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
    (_b = __classPrivateFieldGet3(this, _Iti_rejectAutoCountryPromise, "f")) === null || _b === void 0 ? void 0 : _b.call(this);
    return;
  }
  __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_setInitialState).call(this, true);
  __classPrivateFieldGet3(this, _Iti_rejectAutoCountryPromise, "f").call(this);
}, _Iti_handleUtils = function _Iti_handleUtils2() {
  var _b;
  if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
    (_b = __classPrivateFieldGet3(this, _Iti_resolveUtilsScriptPromise, "f")) === null || _b === void 0 ? void 0 : _b.call(this);
    return;
  }
  if (intlTelInput.utils) {
    const inputValue = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getTelInputValue).call(this);
    if (inputValue) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateValFromNumber).call(this, inputValue);
    }
    if (__classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2) {
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updatePlaceholder).call(this);
      __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_updateMaxLength).call(this);
    }
  }
  __classPrivateFieldGet3(this, _Iti_resolveUtilsScriptPromise, "f").call(this);
}, _Iti_handleUtilsFailure = function _Iti_handleUtilsFailure2(error) {
  var _b;
  if (!__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
    (_b = __classPrivateFieldGet3(this, _Iti_rejectUtilsScriptPromise, "f")) === null || _b === void 0 ? void 0 : _b.call(this, error);
    return;
  }
  __classPrivateFieldGet3(this, _Iti_rejectUtilsScriptPromise, "f").call(this, error);
}, _Iti_utilsIsPossibleNumber = function _Iti_utilsIsPossibleNumber2(val) {
  return intlTelInput.utils ? intlTelInput.utils.isPossibleNumber(val, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2, __classPrivateFieldGet3(this, _Iti_options, "f").allowedNumberTypes) : null;
}, _Iti_validateNumber = function _Iti_validateNumber2(precise) {
  if (!intlTelInput.utils || !__classPrivateFieldGet3(this, _Iti_ui, "f").telInput) {
    return null;
  }
  const { allowNumberExtensions, allowPhonewords } = __classPrivateFieldGet3(this, _Iti_options, "f");
  const testValidity = (s) => precise ? __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_utilsIsValidNumber).call(this, s) : __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_utilsIsPossibleNumber).call(this, s);
  const val = __classPrivateFieldGet3(this, _Iti_instances, "m", _Iti_getFullNumber).call(this);
  if (!__classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2) {
    const isRegionlessDialCode = hasRegionlessDialCode(val);
    if (!isRegionlessDialCode) {
      return false;
    }
  }
  if (!testValidity(val)) {
    return false;
  }
  const alphaCharPosition = val.search(REGEX.ALPHA_UNICODE);
  const hasAlphaChar = alphaCharPosition > -1;
  if (hasAlphaChar) {
    const selectedIso2 = __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2;
    const hasExtension = Boolean(intlTelInput.utils.getExtension(val, selectedIso2));
    if (hasExtension) {
      return allowNumberExtensions;
    }
    return allowPhonewords;
  }
  return true;
}, _Iti_utilsIsValidNumber = function _Iti_utilsIsValidNumber2(val) {
  return intlTelInput.utils ? intlTelInput.utils.isValidNumber(val, __classPrivateFieldGet3(this, _Iti_selectedCountryData, "f").iso2, __classPrivateFieldGet3(this, _Iti_options, "f").allowedNumberTypes) : null;
};
var attachUtils = (source) => {
  if (!intlTelInput.utils && !intlTelInput.startedLoadingUtilsScript) {
    let loadCall;
    if (typeof source === "function") {
      try {
        loadCall = Promise.resolve(source());
      } catch (error) {
        return Promise.reject(error);
      }
    } else {
      return Promise.reject(new TypeError(`The argument passed to attachUtils must be a function that returns a promise for the utilities module, not ${typeof source}`));
    }
    intlTelInput.startedLoadingUtilsScript = true;
    return loadCall.then((module) => {
      const utils = module === null || module === void 0 ? void 0 : module.default;
      if (!utils || typeof utils !== "object") {
        throw new TypeError("The loader function passed to attachUtils did not resolve to a module object with utils as its default export.");
      }
      intlTelInput.utils = utils;
      Iti.forEachInstance("handleUtils");
      return true;
    }).catch((error) => {
      Iti.forEachInstance("handleUtilsFailure", error);
      throw error;
    });
  }
  return null;
};
var intlTelInput = Object.assign((input, options) => {
  const iti = new Iti(input, options);
  intlTelInput.instances[iti.id] = iti;
  input.iti = iti;
  return iti;
}, {
  defaults,
  //* Using a static var like this allows us to mock it in the tests.
  documentReady: () => document.readyState === "complete",
  //* Get the country data object.
  getCountryData: () => data_default,
  //* A getter for the plugin instance.
  getInstance: (input) => {
    const id2 = input.dataset.intlTelInputId;
    return id2 ? intlTelInput.instances[id2] : null;
  },
  //* A map from instance ID to instance object.
  instances: {},
  attachUtils,
  startedLoadingUtilsScript: false,
  startedLoadingAutoCountry: false,
  version: "26.8.1"
});
var intl_tel_input_default = intlTelInput;

// angular/build/temp/intl-tel-input/angular.js
import { Component, Input, ViewChild, ElementRef, Output, EventEmitter, forwardRef } from "@angular/core";
import { NG_VALUE_ACCESSOR, NG_VALIDATORS } from "@angular/forms";
import * as i0 from "@angular/core";
var _c0 = ["inputRef"];
var PHONE_ERROR_MESSAGES = [
  "invalid",
  "invalid-country-code",
  "too-short",
  "too-long",
  "invalid-format"
];
var IntlTelInputComponent = class {
  constructor() {
    this.usePreciseValidation = false;
    this.inputProps = {};
    this.disabled = false;
    this.numberChange = new EventEmitter();
    this.countryChange = new EventEmitter();
    this.validityChange = new EventEmitter();
    this.errorCodeChange = new EventEmitter();
    this.blur = new EventEmitter();
    this.focus = new EventEmitter();
    this.keydown = new EventEmitter();
    this.keyup = new EventEmitter();
    this.paste = new EventEmitter();
    this.click = new EventEmitter();
    this.appliedInputPropKeys = /* @__PURE__ */ new Set();
    this.countryChangeHandler = () => this.handleInput();
    this.onChange = () => {
    };
    this.onTouched = () => {
    };
    this.onValidatorChange = () => {
    };
  }
  ngAfterViewInit() {
    var _a3, _b;
    if (this.inputRef.nativeElement) {
      this.iti = intl_tel_input_default(this.inputRef.nativeElement, this.initOptions);
    }
    this.inputRef.nativeElement.addEventListener("countrychange", this.countryChangeHandler);
    this.applyInputProps();
    if (this.initialValue) {
      (_a3 = this.iti) === null || _a3 === void 0 ? void 0 : _a3.setNumber(this.initialValue);
    }
    if (this.disabled) {
      (_b = this.iti) === null || _b === void 0 ? void 0 : _b.setDisabled(this.disabled);
    }
  }
  ngOnChanges(changes) {
    var _a3;
    if (changes["disabled"]) {
      (_a3 = this.iti) === null || _a3 === void 0 ? void 0 : _a3.setDisabled(this.disabled);
    }
    if (changes["inputProps"]) {
      this.applyInputProps();
    }
  }
  handleInput() {
    if (!this.iti)
      return;
    const num = this.iti.getNumber() || "";
    const countryIso = this.iti.getSelectedCountryData().iso2 || "";
    let hasChanged = false;
    if (num !== this.lastEmittedNumber) {
      this.lastEmittedNumber = num;
      this.numberChange.emit(num);
      this.onChange(num);
      hasChanged = true;
    }
    if (countryIso !== this.lastEmittedCountry) {
      this.lastEmittedCountry = countryIso;
      this.countryChange.emit(countryIso);
      hasChanged = true;
    }
    const isValid = this.usePreciseValidation ? this.iti.isValidNumberPrecise() : this.iti.isValidNumber();
    const errorCode = isValid ? null : this.iti.getValidationError();
    if (isValid !== this.lastEmittedValidity) {
      this.lastEmittedValidity = isValid;
      this.validityChange.emit(isValid);
      hasChanged = true;
    }
    if (errorCode !== this.lastEmittedErrorCode) {
      this.lastEmittedErrorCode = errorCode;
      this.errorCodeChange.emit(errorCode);
      hasChanged = true;
    }
    if (hasChanged) {
      this.onValidatorChange();
    }
  }
  handleBlur(event) {
    this.onTouched();
    this.blur.emit(event);
  }
  handleFocus(event) {
    this.focus.emit(event);
  }
  handleKeyDown(event) {
    this.keydown.emit(event);
  }
  handleKeyUp(event) {
    this.keyup.emit(event);
  }
  handlePaste(event) {
    this.paste.emit(event);
  }
  handleClick(event) {
    this.click.emit(event);
  }
  /**
   * This method must be called in `ngAfterViewInit` or later lifecycle hooks,
   * not in `ngOnInit` or the `constructor`, as the component needs to be fully initialized.
   */
  getInstance() {
    return this.iti;
  }
  /**
   * This method must be called in `ngAfterViewInit` or later lifecycle hooks,
   * not in `ngOnInit` or the `constructor`, as the component needs to be fully initialized.
   */
  getInput() {
    return this.inputRef.nativeElement;
  }
  ngOnDestroy() {
    var _a3;
    (_a3 = this.iti) === null || _a3 === void 0 ? void 0 : _a3.destroy();
    this.inputRef.nativeElement.removeEventListener("countrychange", this.countryChangeHandler);
  }
  applyInputProps() {
    const currentKeys = /* @__PURE__ */ new Set();
    Object.entries(this.inputProps).forEach(([key, value]) => {
      currentKeys.add(key);
      this.inputRef.nativeElement.setAttribute(key, value);
    });
    this.appliedInputPropKeys.forEach((key) => {
      if (!currentKeys.has(key)) {
        this.inputRef.nativeElement.removeAttribute(key);
      }
    });
    this.appliedInputPropKeys = currentKeys;
  }
  // ============ ControlValueAccessor Implementation ============
  writeValue(value) {
    if (this.iti) {
      this.iti.setNumber(value || "");
    }
  }
  registerOnChange(fn) {
    this.onChange = fn;
  }
  registerOnTouched(fn) {
    this.onTouched = fn;
  }
  setDisabledState(isDisabled) {
    var _a3;
    this.disabled = isDisabled;
    (_a3 = this.iti) === null || _a3 === void 0 ? void 0 : _a3.setDisabled(isDisabled);
  }
  // ============ Validator Implementation ============
  validate(control) {
    var _a3;
    if (!control.value || !this.iti) {
      return null;
    }
    const isValid = this.usePreciseValidation ? this.iti.isValidNumberPrecise() : this.iti.isValidNumber();
    if (isValid) {
      return null;
    }
    const errorCode = this.iti.getValidationError();
    return {
      invalidPhone: {
        errorCode,
        errorMessage: (_a3 = PHONE_ERROR_MESSAGES[errorCode]) !== null && _a3 !== void 0 ? _a3 : "unknown"
      }
    };
  }
  registerOnValidatorChange(fn) {
    this.onValidatorChange = fn;
  }
};
IntlTelInputComponent.\u0275fac = function IntlTelInputComponent_Factory(__ngFactoryType__) {
  return new (__ngFactoryType__ || IntlTelInputComponent)();
};
IntlTelInputComponent.\u0275cmp = /* @__PURE__ */ i0.\u0275\u0275defineComponent({ type: IntlTelInputComponent, selectors: [["intl-tel-input"]], viewQuery: function IntlTelInputComponent_Query(rf, ctx) {
  if (rf & 1) {
    i0.\u0275\u0275viewQuery(_c0, 7);
  }
  if (rf & 2) {
    let _t;
    i0.\u0275\u0275queryRefresh(_t = i0.\u0275\u0275loadQuery()) && (ctx.inputRef = _t.first);
  }
}, inputs: { initialValue: "initialValue", usePreciseValidation: "usePreciseValidation", inputProps: "inputProps", disabled: "disabled", initOptions: "initOptions" }, outputs: { numberChange: "numberChange", countryChange: "countryChange", validityChange: "validityChange", errorCodeChange: "errorCodeChange", blur: "blur", focus: "focus", keydown: "keydown", keyup: "keyup", paste: "paste", click: "click" }, features: [i0.\u0275\u0275ProvidersFeature([
  {
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(() => IntlTelInputComponent),
    multi: true
  },
  {
    provide: NG_VALIDATORS,
    useExisting: forwardRef(() => IntlTelInputComponent),
    multi: true
  }
]), i0.\u0275\u0275NgOnChangesFeature], decls: 2, vars: 0, consts: [["inputRef", ""], ["type", "tel", 3, "input", "blur", "focus", "keydown", "keyup", "paste", "click"]], template: function IntlTelInputComponent_Template(rf, ctx) {
  if (rf & 1) {
    const _r1 = i0.\u0275\u0275getCurrentView();
    i0.\u0275\u0275elementStart(0, "input", 1, 0);
    i0.\u0275\u0275listener("input", function IntlTelInputComponent_Template_input_input_0_listener() {
      i0.\u0275\u0275restoreView(_r1);
      return i0.\u0275\u0275resetView(ctx.handleInput());
    })("blur", function IntlTelInputComponent_Template_input_blur_0_listener($event) {
      i0.\u0275\u0275restoreView(_r1);
      return i0.\u0275\u0275resetView(ctx.handleBlur($event));
    })("focus", function IntlTelInputComponent_Template_input_focus_0_listener($event) {
      i0.\u0275\u0275restoreView(_r1);
      return i0.\u0275\u0275resetView(ctx.handleFocus($event));
    })("keydown", function IntlTelInputComponent_Template_input_keydown_0_listener($event) {
      i0.\u0275\u0275restoreView(_r1);
      return i0.\u0275\u0275resetView(ctx.handleKeyDown($event));
    })("keyup", function IntlTelInputComponent_Template_input_keyup_0_listener($event) {
      i0.\u0275\u0275restoreView(_r1);
      return i0.\u0275\u0275resetView(ctx.handleKeyUp($event));
    })("paste", function IntlTelInputComponent_Template_input_paste_0_listener($event) {
      i0.\u0275\u0275restoreView(_r1);
      return i0.\u0275\u0275resetView(ctx.handlePaste($event));
    })("click", function IntlTelInputComponent_Template_input_click_0_listener($event) {
      i0.\u0275\u0275restoreView(_r1);
      return i0.\u0275\u0275resetView(ctx.handleClick($event));
    });
    i0.\u0275\u0275elementEnd();
  }
}, encapsulation: 2 });
(() => {
  (typeof ngDevMode === "undefined" || ngDevMode) && i0.\u0275setClassMetadata(IntlTelInputComponent, [{
    type: Component,
    args: [{
      selector: "intl-tel-input",
      standalone: true,
      template: `
    <input
      type="tel"
      #inputRef
      (input)="handleInput()"
      (blur)="handleBlur($event)"
      (focus)="handleFocus($event)"
      (keydown)="handleKeyDown($event)"
      (keyup)="handleKeyUp($event)"
      (paste)="handlePaste($event)"
      (click)="handleClick($event)"
    />
  `,
      providers: [
        {
          provide: NG_VALUE_ACCESSOR,
          useExisting: forwardRef(() => IntlTelInputComponent),
          multi: true
        },
        {
          provide: NG_VALIDATORS,
          useExisting: forwardRef(() => IntlTelInputComponent),
          multi: true
        }
      ]
    }]
  }], null, { inputRef: [{
    type: ViewChild,
    args: ["inputRef", { static: true }]
  }], initialValue: [{
    type: Input
  }], usePreciseValidation: [{
    type: Input
  }], inputProps: [{
    type: Input
  }], disabled: [{
    type: Input
  }], initOptions: [{
    type: Input
  }], numberChange: [{
    type: Output
  }], countryChange: [{
    type: Output
  }], validityChange: [{
    type: Output
  }], errorCodeChange: [{
    type: Output
  }], blur: [{
    type: Output
  }], focus: [{
    type: Output
  }], keydown: [{
    type: Output
  }], keyup: [{
    type: Output
  }], paste: [{
    type: Output
  }], click: [{
    type: Output
  }] });
})();
(() => {
  (typeof ngDevMode === "undefined" || ngDevMode) && i0.\u0275setClassDebugInfo(IntlTelInputComponent, { className: "IntlTelInputComponent", filePath: "intl-tel-input/angular.ts", lineNumber: 66 });
})();
export {
  IntlTelInputComponent,
  PHONE_ERROR_MESSAGES,
  intl_tel_input_default as intlTelInput
};
